---
title: "flowcells_dna"
author: "George"
date: "3/3/2021"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
```

## Read files

```{r input files, message=FALSE, warning=FALSE}
read_tsv('smd.tsv', col_types = cols(.default = col_character())) -> smd
read_tsv("taxonomy.tsv", col_types = cols(.default = col_character())) -> tax
read_tsv('seqtab.tsv', col_types = cols(.default = col_character(), count = col_integer())) %>%
  # Filter out samples with less than 1000 reads
  group_by(sample) %>% filter(sum(count) > 1000) %>%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %>% ungroup() %>%
  # Average the relab within an environment by dividing by the total
  inner_join(smd, by = 'sample') %>%
  group_by(env) %>% mutate(relab = relab / sum(relab)) %>% ungroup() %>%
  select(seqid, sample, count, relab) -> seqtab
```

## Rarefaction

```{r rarefaction}
seqtab %>%
  inner_join(smd, by = 'sample') %>% mutate(nature = paste(nature, origin)) %>%
  group_by(nature, seqid) %>% summarise(count = sum(count), .groups = 'drop') %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('nature') %>%
  vegan::rarecurve(sample = 100, step = 100, xlab = 'Sequencing depth', ylab = '# ASVs', label = FALSE)
```

## Alpha diversity using Shannon-Weaver index

```{r calculcate Shannon, message=FALSE, warning=FALSE}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>% gather(seqid, shannon, -1) %>%
  rename(shannon = 1) %>%
  rownames_to_column('sample') %>%
  inner_join(smd, by = "sample") %>%
  arrange(env) -> adiv
```

Plot alpha diversity

```{r plot Shannon}
adiv %>%
  mutate(env = paste(nature, desig)) %>%
  ggplot(aes(x = shannon, y = fct_relevel(env, c('Biofilm TM-448.4','Planktonic TM-448.4')))) + 
  geom_boxplot() +
  labs(x = "Alpha diversity (Shannon index)", y = "") +
  theme_bw() +
  theme(panel.grid = element_blank(), axis.ticks = element_blank(),
        aspect.ratio = 0.7,
        axis.text.x = element_text(size = 9, colour = 'black'),
        axis.text.y = element_text(size = 9, colour = 'black')) -> a
```

```{r save plot, message=FALSE, warning=FALSE}
ggsave("figures/adiv_dna_origin.pdf", width = 12, height = 10, units = "cm")
```

```{r statistics alpha diversity}
# Statistics on alpha diversity
adiv %>%
  group_by(nature, origin, filter, age) %>%
  summarise(meandiv = mean(pielou)) %>%
  # Log transform
  mutate(meandiv = log10(meandiv)) -> adiv

suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(adiv$meandiv)
# Preferably all the dots on the diagonal
qqPlot(adiv$meandiv)
# Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(meandiv ~ paste(nature, origin), data = adiv)
# Run ANOVA (null hypothesis diversity is equal)
aov(meandiv ~ paste(nature, origin), data = adiv) %>% summary()
# Post-hoc
aov(meandiv ~ paste(nature, origin), data = adiv) %>% TukeyHSD()
```

## Bar plot

```{r select 11 most abundant phyla, message=FALSE, warning=FALSE}
seqtab %>%
  inner_join(tax, by = "seqid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(relab = sum(relab)) %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & order != 'uncultured bacterium') %>%
  top_n(11, mean_relab) -> t
```

```{r add topphylum to taxonomy, message=FALSE, warning=FALSE}
# Add column to taxonomy whether or not phylum is part of selection
tax %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```

```{r alternative barplot, message=FALSE, warning=FALSE}
seqtab %>%
  inner_join(taxref, by = "seqid") %>%
  # Proteobacteria on Class level
  mutate(topphylum = if_else(topphylum == 'Proteobacteria', class, topphylum)) %>%
  filter(!is.na(topphylum)) %>%
  # Combine water type and origin
  inner_join(smd, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(env, topphylum) %>% summarise(relab = sum(relab), .groups = 'drop') %>%
  # Call plot
  ggplot(aes(x = fct_rev(env), 
             y = relab, 
    fill = fct_relevel(topphylum, c('Other', 'Alphaproteobacteria', 'Deltaproteobacteria', 'Gammaproteobacteria')))) +
  labs(x = '', y = 'Relative abundance') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_flip() + scale_y_continuous(trans = 'reverse', labels = c('1.0','0.75','0.50','0.25','0')) +
  theme(legend.position = 'bottom', 
        legend.spacing.x = unit(0.5, 'mm'),
        text = element_text(colour = 'black', size = 7), 
        axis.text.x = element_text(colour = 'black', size = 7),
        legend.text = element_text(colour = 'black', size = 7),
        legend.box.spacing = unit(1, 'mm'),
        # For margin, the order is top - right - bottom - left
        axis.text.y = element_text(margin = margin(0,-5,0,0, unit = 'mm'), colour = 'black', size = 7),
        panel.background = element_blank(), aspect.ratio = 0.5,
        legend.title = element_blank(),
        axis.ticks.y = element_blank())
```


```{r save barplot}
ggsave("figures/barplot_class.pdf", height = 12, width = 16, units = "cm")
```

## Differential abundance and heatmap

```{r ALDEx2 analysis for differential ASV abundance, message=FALSE, warning=FALSE}
seqtab %>%
  # Filter out very low abundant ASVs
  group_by(seqid) %>% filter(count > 10) %>% ungroup() %>%
  select(-relab) %>% spread(sample, count, fill = 0) %>%
  column_to_rownames('seqid') %>%
  ALDEx2::aldex(conditions = smd %>% arrange(sample) %>% pull(nature)) %>%
  rownames_to_column("seqid") %>%
  inner_join(tax) -> aldex
```

```{r pheatmap}
aldex %>%
  slice_min(we.eBH, n = 30) %>%
  # Mutate taxonomies that only say 'uncultured'
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured bacterium'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured organism'))) %>%
  # Create a label based on the highest taxonomic level
  mutate(label = coalesce(genus, family, order, class, phylum, domain)) %>%
  # Add a unique ASV labl
  mutate(label = paste("ASV", sprintf('%02d', row_number()), label)) %>%
  select(label, seqid) %>%
  # Join with seqtab to obtain abundance per sample
  inner_join(seqtab, by = 'seqid') %>%
  # Join with metadata to add categories
  inner_join(smd, by = 'sample') %>%
  select(label, env, relab) %>%
  # Sum abundance within each grouping
  group_by(env, label) %>% summarise(relab = sum(relab)) %>%
  # Transform abundance
  mutate(relab = sqrt(relab)) %>%
  spread(env, relab, fill = 0) %>% column_to_rownames('label') %>%
  # Convert to a matrix
  as.matrix() -> heatmap_matrix

library(RColorBrewer)
pheatmap::pheatmap(heatmap_matrix,
                   filename = 'figures/heatmap_d.pdf',
                   color = colorRampPalette(brewer.pal(8, "Blues"))(7),
                   legend_breaks = c(0, 0.1, 0.2, 0.3, max(heatmap_matrix)),
                   legend_labels = c('0', '0.1', '0.2', '0.3', 'Abundance'),
                   breaks = c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35),
                   border_color = 'grey40', fontsize = 9,
                   cellwidth = 15, cellheight = 15, 
                   scale = 'none', cluster_rows = T, cluster_cols = F, treeheight_row = 20,
                   clustering_distance_rows = 'euclidean',
                   legend = T)
```

## PCA on Hellinger-transformed dataset

Statistics on beta diversity: PERMANOVA

```{r Adonis}
smd %>% mutate(nature = paste(nature, desig)) %>% arrange(sample) -> df

seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  pairwiseAdonis::pairwise.adonis(factors = df$nature, sim.method = 'euclidean')
```

Transform the genecounts

```{r Run PCA}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  rownames_to_column('sample') %>%
  gather(seqid, hellinger, -1) %>%
  filter(hellinger > 0) %>%
  mutate(hellinger = as.numeric(hellinger)) -> seqtab_t
```

```{r Extract PCA values}
seqtab_t %>%
  spread(seqid, hellinger, fill = 0) %>%
  column_to_rownames('sample') %>% vegan::rda() -> pca

pca.samples <- pca$CA$u %>% data.frame() %>% tibble::rownames_to_column('sample')
pca.geneid    <- pca$CA$v %>% data.frame() %>% tibble::rownames_to_column('seqid')
pca.eigs    <- pca$CA$eig %>% data.frame() %>% tibble::rownames_to_column('pc') %>%
  rename(eigval = 2) %>%
  mutate(propexpl = eigval/sum(eigval))
```

```{r Plot PCA}
pca.samples %>%
  inner_join(smd, by = 'sample') %>%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = desig), size = 5) +
  # Labelled according to groundwater type
  scale_color_grey() +
  xlab(sprintf("PC1 (%2.1f%% explained)", pca.eigs[1,3] * 100)) +
  ylab(sprintf("PC2 (%2.1f%% explained)", pca.eigs[2,3] * 100)) +
  theme_bw() +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme(legend.title = element_blank(), 
        text = element_text(size = 9, colour = 'black'),
        axis.text.x = element_text(colour = 'black'), 
        axis.text.y = element_text(colour = 'black'),
        panel.grid = element_blank(),
        legend.position = c(0.85, 0.25),
        legend.box.background = element_rect(color = 'black', size = 0.5),
        legend.background = element_blank(),
        legend.text = element_text(size = 8),
        aspect.ratio = 1.0) -> b
```

```{r}
cowplot::plot_grid(a, b, labels = c('a','b'))
```

```{r export PCA}
ggsave("figures/fig1.pdf", width = 20, height = 12, units = "cm")
```
