<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="George Westmeijer" />


<title>Flowcells Äspö HRL</title>

<script src="flowcells_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="flowcells_files/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="flowcells_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="flowcells_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="flowcells_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="flowcells_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="flowcells_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="flowcells_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="flowcells_files/navigation-1.1/tabsets.js"></script>
<script src="flowcells_files/navigation-1.1/codefolding.js"></script>
<link href="flowcells_files/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="flowcells_files/highlightjs-9.12.0/highlight.js"></script>
<link href="flowcells_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="flowcells_files/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Flowcells Äspö HRL</h1>
<h3 class="subtitle">Biofilm formation in deep biosphere at Äspö HRL</h3>
<h4 class="author">George Westmeijer</h4>

</div>


<pre class="r"><code>library(tidyverse)</code></pre>
<div id="define-theme-and-functions" class="section level2" number="0.1">
<h2><span class="header-section-number">0.1</span> Define theme and function(s)</h2>
<pre class="r"><code>source(&quot;/Users/geweaa/Box Sync/Protocols/theme.R&quot;)
source(&quot;/Users/geweaa/Box Sync/Protocols/barplot.R&quot;)</code></pre>
</div>
<div id="read-files" class="section level2" number="0.2">
<h2><span class="header-section-number">0.2</span> Read files</h2>
<pre class="r"><code>read_tsv(&#39;smd.tsv&#39;, col_types = cols(.default = col_character(),
                                     env = col_factor(levels = 
                                                        c(&quot;Planktonic MM-171.3&quot;,&quot;Biofilm MM-171.3&quot;,
                                                          &quot;Planktonic TM-448.4&quot;,&quot;Biofilm TM-448.4&quot;)),
                                     treat = col_factor(levels = c(
                                       &quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;
                                     ))
                                     )
         ) -&gt; smd

read_tsv(&quot;taxonomy.tsv&quot;, col_types = cols(.default = col_character())) %&gt;%
  mutate(order = if_else(order == &quot;Betaproteobacteriales&quot;,&quot;Nitrosomonadales&quot;,order)) %&gt;%
  mutate(family = if_else(family == &quot;Hydrogenophilaceae&quot;,&quot;Thiobacillaceae&quot;,family)) %&gt;%
  mutate(order = if_else(family == &quot;Burkholderiaceae&quot;,&quot;Burkholderiales&quot;,order))-&gt; tax

read_tsv(&#39;seqtab.tsv&#39;, col_types = cols(.default = col_character(), count = col_integer())) %&gt;%
  # Filter out samples with less than 1000 reads
  group_by(sample) %&gt;% filter(sum(count) &gt; 1000) %&gt;%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %&gt;% ungroup() %&gt;%
  # Average the relative abundance within an environment by dividing by the total
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  mutate(env = if_else(nature == &quot;Planktonic&quot;, as.vector(env), paste(env,age))) %&gt;%
  group_by(env) %&gt;% mutate(relab = relab / sum(relab)) %&gt;% ungroup() %&gt;%
  select(sample, seqid, count, relab) -&gt; seqtab</code></pre>
</div>
<div id="rarefaction" class="section level2" number="0.3">
<h2><span class="header-section-number">0.3</span> Rarefaction</h2>
<pre class="r"><code>seqtab %&gt;%
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  group_by(env, seqid) %&gt;% summarise(count = sum(count), .groups = &#39;drop&#39;) %&gt;%
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&#39;env&#39;) -&gt; seqtab_matrix

min(rowSums(seqtab_matrix)) %&gt;%
vegan::rarecurve(seqtab_matrix, sample = ., step = 100, xlab = &#39;Sequencing depth (No. reads)&#39;, ylab = &#39;Diversity (No. ASVs)&#39;, label = TRUE)</code></pre>
<p><img src="flowcells_files/figure-html/rarefaction-1.png" width="672" /></p>
</div>
<div id="alpha-diversity-using-shannon-weaver-index" class="section level2" number="0.4">
<h2><span class="header-section-number">0.4</span> Alpha diversity using Shannon-Weaver index</h2>
<pre class="r"><code>seqtab %&gt;%
  select(-relab) %&gt;%
  spread(seqid, count, fill = 0) %&gt;% column_to_rownames(&#39;sample&#39;) %&gt;%
  vegan::diversity() %&gt;% as.data.frame() %&gt;% 
  gather(seqid, shannon, -1) %&gt;%
  rename(shannon = 1) %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) %&gt;% as_tibble() -&gt; adiv</code></pre>
<pre class="r"><code>adiv %&gt;% group_by(env,treat,age) %&gt;% summarise(meandiv = mean(shannon), .groups = &quot;drop&quot;) -&gt; stat

suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(stat$meandiv)</code></pre>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  stat$meandiv
## W = 0.94627, p-value = 0.6246</code></pre>
<pre class="r"><code># Preferably all the dots on the diagonal
qqPlot(stat$meandiv)</code></pre>
<p><img src="flowcells_files/figure-html/statistics%20alpha%20diversity-1.png" width="672" /></p>
<pre><code>## [1] 1 8</code></pre>
<pre class="r"><code># Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(meandiv ~ env, data = stat)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["F value"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"0.8002808","3":"0.5374241","_rn_":"group"},{"1":"6","2":"NA","3":"NA","_rn_":""}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># Run ANOVA (null hypothesis diversity is equal)
aov(meandiv ~ env, data = stat) %&gt;% summary()</code></pre>
<pre><code>##             Df Sum Sq Mean Sq F value Pr(&gt;F)
## env          3  1.170  0.3901   1.434  0.323
## Residuals    6  1.632  0.2721</code></pre>
<pre class="r"><code># Post-hoc
aov(meandiv ~ env, data = stat) %&gt;% TukeyHSD()</code></pre>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = meandiv ~ env, data = stat)
## 
## $env
##                                                diff       lwr      upr
## Biofilm MM-171.3-Planktonic MM-171.3    -0.50600036 -2.154258 1.142257
## Planktonic TM-448.4-Planktonic MM-171.3 -0.04962994 -1.855206 1.755946
## Biofilm TM-448.4-Planktonic MM-171.3    -0.83695149 -2.485209 0.811306
## Planktonic TM-448.4-Biofilm MM-171.3     0.45637041 -1.191887 2.104628
## Biofilm TM-448.4-Biofilm MM-171.3       -0.33095113 -1.805197 1.143295
## Biofilm TM-448.4-Planktonic TM-448.4    -0.78732154 -2.435579 0.860936
##                                             p adj
## Biofilm MM-171.3-Planktonic MM-171.3    0.7224663
## Planktonic TM-448.4-Planktonic MM-171.3 0.9996554
## Biofilm TM-448.4-Planktonic MM-171.3    0.3748675
## Planktonic TM-448.4-Biofilm MM-171.3    0.7766906
## Biofilm TM-448.4-Biofilm MM-171.3       0.8622213
## Biofilm TM-448.4-Planktonic TM-448.4    0.4196640</code></pre>
<p>Plot alpha diversity</p>
<pre class="r"><code>adiv %&gt;%
  mutate(age = if_else(nature == &quot;Planktonic&quot;, &quot;&quot;, gsub(&quot;d&quot;,&quot; d&quot;, age))) %&gt;%
  ggplot(aes(x = shannon, y = fct_rev(env))) +
  geom_boxplot() +
  geom_jitter(width = 0.01, stroke = 0.8, size = 1, fill = &quot;white&quot;, shape = 21, height = 0.1) +
  labs(x = &quot;Alpha diversity (Shannon index)&quot;, y = &quot;&quot;) +
  scale_x_continuous(limits = c(1,6), expand = c(0.004,0), breaks = ) +
  scale_y_discrete(labels = c(&quot;Biofilm TM-448.4\nn = 6&quot;,&quot;Planktonic TM-448.4\nn = 12&quot;,
                              &quot;Biofilm MM-171.3\nn = 6&quot;,&quot;Planktonic MM-171.3\nn = 12&quot;)) +
  ggrepel::geom_text_repel(aes(label = age), color = &quot;black&quot;,
                  box.padding = 0.3, 
                  max.overlaps = 10,
                  min.segment.length = 0.4,
                  size = 6/.pt) +
  # Bracket TM-448.4
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 5, xend = 5, y = 1, yend = 2) +
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 4.9, xend = 5, y = 1, yend = 1) +
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 4.9, xend = 5, y = 2, yend = 2) +
  #annotate(geom = &quot;text&quot;, x = 5.2, y = 1.5, label = &quot;ns&quot;, size = 6/.pt) +
  # Bracket MM-171.3
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 5.7, xend = 5.7, y = 3, yend = 4) +
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 5.6, xend = 5.7, y = 3, yend = 3) +
  #annotate(geom = &quot;segment&quot;, lineend = &quot;square&quot;, x = 5.6, xend = 5.7, y = 4, yend = 4) +
  #annotate(geom = &quot;text&quot;, x = 5.9, y = 3.5, label = &quot;ns&quot;, size = 6/.pt) +
  theme_clean(ratio = 1.6) +
  theme(plot.tag.position = c(.2,1.01),
        panel.grid.major.y = element_blank(),
        panel.grid.major.x = element_line(linetype = &quot;dotted&quot;, color = &quot;grey&quot;, size = 0.3),
        axis.text.y.left = element_text(face = &quot;plain&quot;, hjust = 0.5))  -&gt; a</code></pre>
</div>
<div id="fig.-1b-nmds-on-dataset-dna" class="section level2" number="0.5">
<h2><span class="header-section-number">0.5</span> Fig. 1b: NMDS on dataset DNA</h2>
<pre class="r"><code>seqtab %&gt;%
  group_by(seqid) %&gt;% filter(count &gt; 2) %&gt;% ungroup() %&gt;%
  select(seqid, sample, count) %&gt;% spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) -&gt; t

set.seed(999)
t %&gt;%  vegan::metaMDS(trymax = 50, k = 2, autotransform = T) -&gt; nmds</code></pre>
<p>Extract the scores</p>
<pre class="r"><code>vegan::scores(nmds) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) -&gt; nmds.scores</code></pre>
<p>Plot the ordination</p>
<pre class="r"><code>nmds.scores %&gt;% select(1:3) %&gt;% inner_join(smd, by = &quot;sample&quot;) -&gt; t

#getting the convex hull of each unique point set
find_hull &lt;- function(t) t[chull(t$NMDS1, t$NMDS2), ]
hulls &lt;- plyr::ddply(t, &quot;nature&quot;, find_hull)

nmds.scores %&gt;%
  mutate(age = if_else(nature == &quot;Planktonic&quot;, &quot;&quot;, gsub(&quot;d&quot;,&quot; d&quot;,age))) %&gt;%
  ggplot(aes(x = NMDS1, y = NMDS2, fill = nature)) +
  geom_polygon(data = hulls,alpha = .3) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  # Labelled according to groundwater type
  scale_color_grey() + 
  scale_fill_grey() +
  annotate(&#39;text&#39;, x = Inf, y = -Inf, size = 6/.pt, 
           label = paste(&#39;Stress = &#39;, round(nmds$stress, digits = 2)),
           hjust = 1.1, vjust = -1,
           ) +
  ggrepel::geom_text_repel(aes(label = age), color = &quot;black&quot;,
                  box.padding = 0.3, 
                  max.overlaps = 12,
                  min.segment.length = 0.5,
                  size = 6/.pt) +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;) +
  geom_hline(yintercept = 0, linetype = &#39;dotted&#39;) +
  theme_tidy() +
  theme(legend.spacing.x = unit(0.0, &quot;mm&quot;),
        plot.tag.position = c(.02,1.01)) -&gt; b</code></pre>
<pre class="r"><code>library(patchwork)
a + b + plot_annotation(tag_levels = &quot;a&quot;, tag_suffix = &quot;)&quot;) &amp;
  theme(plot.tag = element_text(size = 8, color = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/Fig.%202-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_2.pdf&quot;, width = 16, height = 10, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_2.png&quot;, width = 16, height = 10, units = &quot;cm&quot;)</code></pre>
<p>Statistics on beta diversity: PERMANOVA</p>
<pre class="r"><code>seqtab %&gt;%
  select(-relab) %&gt;%
  inner_join(smd[,c(&quot;sample&quot;,&quot;nature&quot;)], by = &quot;sample&quot;) %&gt;% 
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) -&gt; t

vegan::adonis(t[,c(-1)] ~ nature, data = t)</code></pre>
<pre><code>## 
## Call:
## vegan::adonis(formula = t[, c(-1)] ~ nature, data = t) 
## 
## Permutation: free
## Number of permutations: 999
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)    
## nature     1    1.8736 1.87356  4.9128 0.12625  0.001 ***
## Residuals 34   12.9662 0.38136         0.87375           
## Total     35   14.8397                 1.00000           
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>seqtab %&gt;%
  select(-relab) %&gt;%
  inner_join(smd[,c(&quot;sample&quot;,&quot;env&quot;)], by = &quot;sample&quot;) %&gt;% 
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) -&gt; t
  
pairwiseAdonis::pairwise.adonis(t[,c(-1)], factors = t$env)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["pairs"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Df"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["SumsOfSqs"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F.Model"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["R2"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p.value"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["p.adjusted"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["sig"],"name":[8],"type":["fct"],"align":["left"]}],"data":[{"1":"Planktonic MM-171.3 vs Planktonic TM-448.4","2":"1","3":"1.8358064","4":"5.4193887","5":"0.1976480","6":"0.001","7":"0.006","8":"*"},{"1":"Planktonic MM-171.3 vs Biofilm MM-171.3","2":"1","3":"1.4668982","4":"4.3593570","5":"0.2141206","6":"0.002","7":"0.012","8":"."},{"1":"Planktonic MM-171.3 vs Biofilm TM-448.4","2":"1","3":"1.4470373","4":"4.2663395","5":"0.2105136","6":"0.001","7":"0.006","8":"*"},{"1":"Planktonic TM-448.4 vs Biofilm MM-171.3","2":"1","3":"1.2844318","4":"3.7831242","5":"0.1912299","6":"0.001","7":"0.006","8":"*"},{"1":"Planktonic TM-448.4 vs Biofilm TM-448.4","2":"1","3":"1.1343726","4":"3.3149657","5":"0.1716268","6":"0.003","7":"0.018","8":"."},{"1":"Biofilm MM-171.3 vs Biofilm TM-448.4","2":"1","3":"0.2713187","4":"0.7964474","5":"0.0737694","6":"0.636","7":"1.000","8":""}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="fig.-2-bar-plot-dna" class="section level2" number="0.6">
<h2><span class="header-section-number">0.6</span> Fig. 2: Bar plot DNA</h2>
<pre class="r"><code>cols &lt;- c(&#39;Other&#39; = &#39;#A6CEE3&#39;, &#39;Actinobacteria&#39; = &#39;#569EA4&#39;, &#39;Alphaproteobacteria&#39; = &#39;#AADB84&#39;,
&#39;Bacteroidetes&#39; = &#39;#52AF43&#39;, &#39;Chloroflexi&#39; = &#39;#F88A89&#39;,
&#39;Deltaproteobacteria&#39; = &#39;#E73233&#39;, &#39;Epsilonbacteraeota&#39; = &#39;#fdbf6f&#39;,
&#39;Firmicutes&#39; = &#39;#fccde5&#39;, &#39;Gammaproteobacteria&#39; = &#39;#bc80bd&#39;, &quot;Omnitrophicaeota&quot; = &#39;#3D3F99&#39;,
&#39;Patescibacteria&#39; = &#39;#ffff99&#39;, &#39;Spirochaetes&#39; = &#39;#b15928&#39;)

barplot() %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;SAR324 clade(Marine group B)&quot;, &quot;SAR324 clade&quot;, topphylum)) %&gt;%
  mutate(env =  gsub(&quot; after&quot;, &quot;&quot;, env)) %&gt;%
  mutate(env =  gsub(&quot; before&quot;, &quot;&quot;, env)) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance (%)&#39;) +
  geom_col(width = 0.9) +
  scale_fill_manual(values = cols) +
  annotate(geom = &quot;segment&quot;, x = .55, xend = 6.45, y = -.01, yend = -.01) +
  annotate(geom = &quot;segment&quot;, x = 6.55, xend = 8.45, y = -.01, yend = -.01) +
  annotate(&#39;text&#39;, x = 3.5, y = -0.03, size = 6 / .pt, label = &quot;Biofilm&quot;, angle = 90, hjust = 0.5) +
  annotate(&#39;text&#39;, x = 7.5, y = -0.03, size = 6 / .pt, label = &quot;Planktonic&quot;, angle = 90, hjust = 0.5) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;100&#39;,&#39;75&#39;,&#39;50&#39;,&#39;25&#39;,&#39;0&#39;)) +
  theme_barplot(ratio = 1.4) + theme(legend.position = &quot;right&quot;)</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20DNA%20phylum-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_s2.pdf&quot;, height = 10, width = 12, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_s2.png&quot;, height = 10, width = 12, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code># Ralstonia (Burkholderiales), Immundisolibacter (Immundisolibacterales), Sulfurimonas (Campylobacterales),
# Brevundimonas (Caulobacterales), Desulfatiferula (Desulfobacterales), Methylotenera (Nitrosomonadales),
# Pseudomonas (Pseudomonadales), Hoeflea (Rhizobiales), Parvibaculum (Parvibaculales), 
# Thiobacillus (Nitrosomonadales), Acidiphilium (Acetobacterales)

cols &lt;- c(&quot;Other&quot; = &quot;#a6cee3&quot;, &quot;Ralstonia&quot; = &quot;#1f78b4&quot;,&quot;Immundisolibacter&quot; = &quot;#045a8d&quot;,
          &quot;Sulfurimonas&quot; = &quot;#33a02c&quot;, &quot;Brevundimonas&quot; = &quot;#fb9a99&quot;, &quot;Desulfatiferula&quot; = &quot;#e31a1c&quot;,
          &quot;Methylotenera&quot; = &quot;#a50f15&quot;,&quot;Thiobacillus&quot; = &quot;#fdbf6f&quot;,&quot;Parvibaculum&quot; = &quot;#ff7f00&quot;, 
          &quot;Pseudomonas&quot; = &quot;#efedf5&quot;, &quot;Hoeflea&quot; = &quot;#cab2d6&quot;,
          &quot;Acidiphilium&quot; = &quot;#807dba&quot;)

barplot(taxlevel = &quot;genus&quot;, n = 12) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;,&quot;Ralstonia&quot;,&quot;Immundisolibacter&quot;,&quot;Sulfurimonas&quot;,
                                             &quot;Brevundimonas&quot;,&quot;Desulfatiferula&quot;,&quot;Methylotenera&quot;,&quot;Thiobacillus&quot;,
                                             &quot;Parvibaculum&quot;,&quot;Pseudomonas&quot;,
                                             &quot;Hoeflea&quot;,&quot;Acidiphilium&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance (%)&#39;) +
  geom_col(width = 0.9) +
  scale_fill_manual(values = cols) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;100&#39;,&#39;75&#39;,&#39;50&#39;,&#39;25&#39;,&#39;0&#39;), expand = c(0.004,0)) + 
  theme_barplot(ratio = 2) + 
  theme(legend.key.size = unit(3, &quot;mm&quot;), legend.text = element_text(size = 6), legend.margin = margin(0,0,0,0),
        legend.box.margin = margin(0,0,0,0), legend.box.spacing = unit(0, &quot;mm&quot;),
        legend.position = &quot;left&quot;,
        axis.line.x = element_line(),
        axis.text.y = element_text(face = &quot;bold&quot;),
        plot.margin = margin(4,5,4,0),
        plot.tag.position = c(.1,1.02)) -&gt; a</code></pre>
<pre class="r"><code>seqtab %&gt;%
  inner_join(smd, by = &quot;sample&quot;) %&gt;%
  mutate(env = ifelse(nature == &quot;Planktonic&quot;, designation, paste(designation,age))) %&gt;%
  inner_join(tax, by = &quot;seqid&quot;) %&gt;%
  group_by(env, genus) %&gt;% summarise(relab = sum(relab) * 100, .groups = &quot;drop_last&quot;) %&gt;%
  filter(!is.na(genus) &amp; genus != &quot;uncultured bacterium&quot; &amp; genus != &quot;uncultured&quot;) %&gt;%
  slice_max(order_by = relab, n = 1) %&gt;% ungroup() %&gt;%
  mutate(hjust = if_else(relab &gt;= 70, 1.07, -0.09)) %&gt;%
  mutate(relab = round(relab, digits = 0)) %&gt;%
  mutate(genus = paste(genus,&quot; (&quot;,relab,&quot;%)&quot;, sep = &quot;&quot;)) %&gt;%
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)), 
             y = relab)) +
  geom_segment(aes(xend = env, y = 0, yend = relab), color = &quot;black&quot;, size = 0.3) +
  geom_point(stroke = 0.8, size = 2, fill = &quot;white&quot;, shape = 21) +
  geom_label(aes(label = genus, hjust = hjust), 
            color = &quot;black&quot;, fontface = &quot;plain&quot;, size = 1.8, label.size = 0.1, label.padding = unit(0.10, &quot;lines&quot;)) +
  coord_flip() +
  scale_y_continuous(expand = c(0.004,0), limits = c(0,100), breaks = c(0,50,100), labels = c(&quot;0&quot;,&quot;50&quot;,&quot;100&quot;)) +
  labs(x = &quot;&quot;, y = &quot;Relative abundance (%)&quot;) +
  theme_clean(ratio = 2.5) +
  theme(panel.grid.major.x = element_line(colour = &quot;grey&quot;, linetype = &quot;dotted&quot;, size = 0.4),
        panel.grid.major.y = element_blank(),
        axis.text.y = element_blank(),
        plot.margin = margin(4,5,4,0),
        plot.tag.position = c(.1,1.02)) -&gt; b</code></pre>
<pre class="r"><code>cols &lt;- c(&quot;Other&quot; = &quot;#a6cee3&quot;, &quot;Burkholderiales&quot; = &quot;#1f78b4&quot;, &quot;Ca. Falkowbacteria&quot; = &quot;#b2df8a&quot;,
          &quot;Campylobacterales&quot; = &quot;#33a02c&quot;, &quot;Caulobacterales&quot; = &quot;#fb9a99&quot;, &quot;Desulfobacterales&quot; = &quot;#e31a1c&quot;,
          &quot;Nitrosomonadales&quot; = &quot;#fdbf6f&quot;,&quot;Parvibaculales&quot; = &quot;#ff7f00&quot;, &quot;Pseudomonadales&quot; = &quot;#efedf5&quot;, &quot;Rhizobiales&quot; = &quot;#cab2d6&quot;,
          &quot;SAR324 clade&quot; = &quot;#6a3d9a&quot;, &quot;Spirochaetales&quot; = &quot;#ffff99&quot;)

barplot(taxlevel = &quot;order&quot;, n = 12) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;SAR324 clade(Marine group B)&quot;, &quot;SAR324 clade&quot;, topphylum)) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;Candidatus Falkowbacteria&quot;, &quot;Ca. Falkowbacteria&quot;, topphylum)) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance (%)&#39;) +
  geom_col(width = 0.9) +
  scale_fill_manual(values = cols) +
  annotate(geom = &quot;segment&quot;, x = .55, xend = 6.45, y = -.01, yend = -.01) +
  annotate(geom = &quot;segment&quot;, x = 6.55, xend = 8.45, y = -.01, yend = -.01) +
  annotate(&#39;text&#39;, x = 3.5, y = -0.06, size = 6 / .pt, label = &quot;Biofilm&quot;, angle = 90, hjust = 0.5) +
  annotate(&#39;text&#39;, x = 7.5, y = -0.06, size = 6 / .pt, label = &quot;Planktonic&quot;, angle = 90, hjust = 0.5) +
  coord_flip(clip = &quot;off&quot;) + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;100&#39;,&#39;75&#39;,&#39;50&#39;,&#39;25&#39;,&#39;0&#39;), expand = c(0.004,0)) + 
  theme_barplot(ratio = 2) + 
  theme(legend.key.size = unit(3, &quot;mm&quot;), legend.text = element_text(size = 6), legend.box.margin = margin(0,0,0,0),
        legend.position = &quot;right&quot;,
        axis.line.x = element_line(),
        axis.text.y = element_blank(),
        plot.margin = margin(4,0,4,0),
        plot.tag.position = c(.1,1.02)) -&gt; c</code></pre>
<pre class="r"><code>library(patchwork)

a + b + c +
  plot_annotation(tag_levels = &quot;a&quot;, tag_suffix = &quot;)&quot;) +
  plot_layout(nrow = 1) &amp;
  theme(plot.tag = element_text(size = 8, color = &quot;black&quot;), plot.tag.position = &quot;top&quot;)</code></pre>
<p><img src="flowcells_files/figure-html/Fig.%203-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_3.pdf&quot;, height = 10, width = 18, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_3.png&quot;, height = 10, width = 18, units = &quot;cm&quot;)</code></pre>
</div>
<div id="load-gene-counts-taxonomy-and-sample-metadata" class="section level2" number="0.7">
<h2><span class="header-section-number">0.7</span> Load gene counts, taxonomy and sample metadata</h2>
<pre class="r"><code>read_tsv(&#39;final.contigs.tsv&#39;, col_types = cols(.default = col_character(),
                                               count = col_integer()
                                               )
         )  -&gt; rnatab

# Read taxonomy (Prokka)
read_tsv(&#39;prokka.tsv&#39;, col_types = cols(.default = col_character(),
                                        length_bp = col_integer()
                                        )
         ) %&gt;% 
  rename(geneid = locus_tag) -&gt; prokka
read_tsv(&#39;eggnogs.tsv&#39;, col_types = cols(.default = col_character())) -&gt; eggnogs

# Taxonomy (NCBI)
read_tsv(&#39;taxa.tsv&#39;, col_types = cols(.default = col_character())) -&gt; ncbi</code></pre>
</div>
<div id="fig.-4a-nmds-on-bray-curtis-dissimilarities-rna" class="section level2" number="0.8">
<h2><span class="header-section-number">0.8</span> Fig. 4a: NMDS on Bray-Curtis dissimilarities (RNA)</h2>
<p>Transform the genecounts</p>
<pre class="r"><code>rnatab %&gt;%
  select(geneid, sample, count) %&gt;% spread(geneid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) -&gt; t

set.seed(999)
t %&gt;%  vegan::metaMDS(trymax = 50, k = 2, autotransform = T) -&gt; nmds</code></pre>
<pre><code>## Warning in postMDS(out$points, dis, plot = max(0, plot - 1), ...): skipping
## half-change scaling: too few points below threshold</code></pre>
<p>Extract the scores</p>
<pre class="r"><code>vegan::scores(nmds) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) -&gt; nmds.scores</code></pre>
<p>Plot the ordination</p>
<pre class="r"><code>nmds.scores %&gt;% select(1:3) %&gt;% inner_join(smd, by = &quot;sample&quot;) -&gt; df

#getting the convex hull of each unique point set
find_hull &lt;- function(df) df[chull(df$NMDS1, df$NMDS2), ]
hulls &lt;- plyr::ddply(df, &quot;nature&quot;, find_hull)

nmds.scores %&gt;%
  ggplot(aes(x = NMDS1, y = NMDS2, fill = nature)) +
  geom_polygon(data = hulls,alpha = .3) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  # Labelled according to groundwater type
  scale_color_grey() + 
  scale_fill_grey() +
  annotate(&#39;text&#39;, x = Inf, y = -Inf, size = 6/.pt, 
           label = paste(&#39;Stress = &#39;, round(nmds$stress, digits = 2)),
           hjust = 1.1, vjust = -1,
           ) +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;) +
  geom_hline(yintercept = 0, linetype = &#39;dotted&#39;) +
  theme_tidy() +
  theme(legend.spacing.x = unit(0.0, &quot;mm&quot;)) -&gt; a</code></pre>
</div>
<div id="fig.-4b-barplot-mrna" class="section level2" number="0.9">
<h2><span class="header-section-number">0.9</span> Fig. 4b: Barplot mRNA</h2>
<p>Transform absolute counts to counts per million (cpm) which is the relative abundance within each sample.</p>
<pre class="r"><code>rnatab %&gt;%
  filter(geneid %in% ncbi$geneid) %&gt;%
  spread(sample, count, fill = 0) %&gt;%
  column_to_rownames(&#39;geneid&#39;) %&gt;%
  edgeR::DGEList(group = na.omit(smd$treat)) %&gt;% edgeR::cpm() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&#39;geneid&#39;) %&gt;%
  gather(sample, cpm , -1) %&gt;% filter(cpm &gt; 0) %&gt;% 
  as_tibble() -&gt; seqcpm</code></pre>
<pre class="r"><code>seqcpm %&gt;%
  # Only use transcripts annotated on phylum level
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) %&gt;%
  group_by(order, env) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = &quot;drop_last&quot;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order)) %&gt;%
  top_n(12, mean_relab) -&gt; t

ncbi %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  mutate(topphylum = if_else(is.na(phylum), &quot;Unidentified&quot;, topphylum)) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;Proteobacteria&quot;, class, topphylum)) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref</code></pre>
<pre class="r"><code>cols &lt;- c(&quot;Other&quot; = &quot;#a6cee3&quot;, &quot;Unidentified&quot; = &quot;#d9d9d9&quot;,
          &quot;Acholeplasmatales&quot; = &quot;#969696&quot;, &quot;Bacillales&quot; = &quot;#b15928&quot;, &quot;Burkholderiales&quot; = &quot;#1f78b4&quot;,
          &quot;Campylobacterales&quot; = &quot;#33a02c&quot;, &quot;Clostridiales&quot; = &quot;#b2df8a&quot;, &quot;Desulfobacterales&quot; = &quot;#e31a1c&quot;,
          &quot;Enterobacterales&quot; = &quot;#fb9a99&quot;, &quot;Methanobacteriales&quot; = &quot;#ff7f00&quot;,
          &quot;Nitrosomonadales&quot; = &quot;#fdbf6f&quot;, &quot;Rhizobiales&quot; = &quot;#cab2d6&quot;,
          &quot;Rhodocyclales&quot; = &quot;#6a3d9a&quot;, &quot;Trichomonadida&quot; = &quot;#ffff99&quot;)

seqcpm %&gt;%
  inner_join(taxref, by = &quot;geneid&quot;) %&gt;%
  # Combine water type and origin
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(env, topphylum) %&gt;% summarise(cpm = sum(cpm) / 2, .groups = &quot;drop&quot;) %&gt;%
  mutate(topphylum = gsub(&quot;proteobacteria&quot;, &quot;proteob.&quot;, topphylum)) %&gt;%
  # Call plot
  ggplot(aes(x = fct_rev(env), 
             y = cpm, 
             fill = fct_relevel(topphylum, c(&#39;Other&#39;, &#39;Unidentified&#39;)))) +
  labs(y = expression(bold(&#39;Counts per million (&#39; %*% &#39;1000)&#39;)), x = &#39;&#39;) +
  scale_y_continuous(labels = c(expression(bold(&quot;1000&quot;)), expression(bold(&quot;750&quot;)), expression(bold(&quot;500&quot;)), expression(bold(&quot;250&quot;)), expression(bold(&quot;0&quot;))), 
                     trans = &#39;reverse&#39;, expand = c(0.002,0)) +
  scale_x_discrete(labels = c(&quot;Biofilm 75 d\nTM-448.4&quot;,&quot;Planktonic\nTM-448.4&quot;,
                              &quot;Biofilm 75 d\nMM-171.3&quot;,&quot;Planktonic\nMM-171.3&quot;)) +
  geom_col() +
  scale_fill_manual(values = cols) +
  coord_flip() + 
  annotate(&#39;text&#39;, x = 1, y = 430000, size = 2, label = &quot;41% Thiobacillaceae&quot;) +
  annotate(&#39;text&#39;, x = 3, y = 430000, size = 2, label = &quot;34% Desulfobacteraceae&quot;) +
  theme_barplot(ratio = 1) + 
  guides(fill = guide_legend(nrow = 5)) +
  theme(legend.key.size = unit(3.3, &quot;mm&quot;),
        axis.line.x = element_line()) -&gt; b</code></pre>
<pre class="r"><code>library(patchwork)
a + b + plot_annotation(tag_levels = &quot;a&quot;, tag_suffix = &quot;)&quot;) &amp;
  theme(plot.tag = element_text(size = 8, color = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/Fig.%204-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_4.pdf&quot;, height = 10, width = 16, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_4.png&quot;, height = 10, width = 16, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code>set.seed(999)

rnatab %&gt;%
  inner_join(smd[,c(&quot;sample&quot;,&quot;nature&quot;)], by = &quot;sample&quot;) %&gt;% 
  spread(geneid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) -&gt; t

vegan::adonis(t[,c(-1)] ~ nature, data = t)</code></pre>
<pre><code>## 
## Call:
## vegan::adonis(formula = t[, c(-1)] ~ nature, data = t) 
## 
## Permutation: free
## Number of permutations: 999
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model     R2 Pr(&gt;F)  
## nature     1   0.79295 0.79295  2.0246 0.2523  0.032 *
## Residuals  6   2.34995 0.39166         0.7477         
## Total      7   3.14290                 1.0000         
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>i &lt;- c(&quot;Translation, ribosomal structure and biogenesis&quot;,&quot;Cell wall/membrane/envelope biogenesis&quot;,
       &quot;Energy production and conversion&quot;,&quot;Cell motility&quot;,&quot;Posttranslational modification, protein turnover, chaperones&quot;,&quot;Signal transduction mechanisms&quot;,
       &quot;Amino acid transport and metabolism&quot;,
       &quot;Replication, recombination and repair&quot;,&quot;Carbohydrate transport and metabolism&quot;,
       &quot;Cytoskeleton&quot;)

cols &lt;- c(&quot;Other&quot; = &quot;#a6cee3&quot;,&quot;Unidentified&quot; = &quot;#d9d9d9&quot;, &quot;Acholeplasmatales&quot; = &quot;#969696&quot;,
          &quot;Burkholderiales&quot; = &quot;#1f78b4&quot;,
          &quot;Campylobacterales&quot; = &quot;#33a02c&quot;, &quot;Clostridiales&quot; = &quot;#b2df8a&quot;, &quot;Desulfobacterales&quot; = &quot;#e31a1c&quot;,
          &quot;Methanobacteriales&quot; = &quot;#ff7f00&quot;,
          &quot;Nitrosomonadales&quot; = &quot;#fdbf6f&quot;, &quot;Rhizobiales&quot; = &quot;#cab2d6&quot;,
          &quot;Rhodocyclales&quot; = &quot;#6a3d9a&quot;, &quot;Trichomonadida&quot; = &quot;#ffff99&quot;)

seqcpm %&gt;%
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  group_by(order, sample) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = &quot;drop_last&quot;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order)) %&gt;%
  top_n(10, mean_relab) -&gt; t

ncbi %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  mutate(topphylum = if_else(is.na(order), &quot;Unidentified&quot;, topphylum)) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) %&gt;%
  inner_join(eggnogs, by = &quot;geneid&quot;) -&gt; taxref

seqcpm %&gt;%
  inner_join(taxref, by = &#39;geneid&#39;) %&gt;%
  filter(eggnog %in% i) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) %&gt;%
  mutate(designation = paste(designation,nature)) %&gt;%
  group_by(topphylum, eggnog, designation) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  ggplot(aes(x = cpm, 
             y = fct_relevel(eggnog, i),
             fill = fct_relevel(topphylum, c(&#39;Other&#39;,&#39;Unidentified&#39;)) %&gt;% fct_rev()
             )
         ) +
  labs(x = expression(bold(&#39;Counts per million (&#39; %*% &#39;1000)&#39;)), y = &#39;&#39;) +
  facet_wrap(~ designation, nrow = 1, ncol = 4) +
  geom_col() +
  scale_fill_manual(values = cols) +
  scale_x_continuous(labels = c(&quot;0&quot;,&quot;50&quot;,&quot;100&quot;,&quot;150&quot;,&quot;&quot;), n.breaks = 5, expand = c(0.004,0)) +  
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  theme_tidy(ratio = 3.4) + 
  theme(panel.border = element_blank(),
        strip.background = element_rect(size = 0,fill = &quot;#f0f0f0&quot;),
        axis.ticks.y = element_blank(),
        legend.position = c(1,0.72),
        strip.text = element_text(face = &quot;bold&quot;),
        axis.text.y = element_text(face = &quot;plain&quot;, size = 6, colour = &quot;black&quot;),
        axis.text.x = element_text(face = &quot;bold&quot;, size = 6),
        legend.spacing.x = unit(0.5, &quot;mm&quot;),
        legend.box.margin = margin(0,0,0,0),
        legend.box.background = element_rect(size = .7, colour = &quot;black&quot;),
        legend.key.size = unit(3.3, &quot;mm&quot;),
        axis.line.x.bottom = element_line(colour = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/Fig.%20s3%20barplot%20biofilm%20order%20level%20eggnog-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_s3.pdf&quot;, height = 10, width = 16, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_s3.png&quot;, height = 10, width = 16, units = &quot;cm&quot;)</code></pre>
</div>
<div id="fig.-5-differential-gene-expression-using-edger" class="section level2" number="0.10">
<h2><span class="header-section-number">0.10</span> Fig. 5: Differential gene expression using EdgeR</h2>
<pre class="r"><code>library(edgeR)
rnatab %&gt;% spread(sample, count, fill = 0) %&gt;% column_to_rownames(&#39;geneid&#39;) -&gt; d
DGEList(d, group = smd$treat %&gt;% na.omit()) -&gt; d

# Filter based on cpm
keep &lt;- filterByExpr(d)
d &lt;- d[keep, , keep.lib.sizes = F]
# TTM normalization to correct for compositional bias
calcNormFactors(d) -&gt; d

# Design matrix based on experimental design
design.mat &lt;- model.matrix(~ 0 + smd$treat)
rownames(design.mat) &lt;- colnames(d)
colnames(design.mat) &lt;- levels(d$samples$group)

# Estimate gene expression dispersion
d %&gt;% estimateDisp(design = design.mat, robust = F) -&gt; d
# Fit a GLM and run the test
fit &lt;- glmFit(d, design = design.mat)

# Genes which baseline differences between planktonic and biofilm
i &lt;- makeContrasts(mm = A-B, tm = C-D, levels = design.mat)

de_mm &lt;- glmLRT(fit, contrast = i[, &quot;mm&quot;])
de_tm &lt;- glmLRT(fit, contrast = i[, &quot;tm&quot;])</code></pre>
<pre class="r"><code>de_mm_test &lt;- decideTestsDGE(de_mm) %&gt;% data.frame() %&gt;% rename(diffexp = 1) %&gt;% rownames_to_column(&quot;geneid&quot;)
de_tm_test &lt;- decideTestsDGE(de_tm) %&gt;% data.frame() %&gt;% rename(diffexp = 1) %&gt;% rownames_to_column(&quot;geneid&quot;)

de_mm$table %&gt;% 
  rownames_to_column(&quot;geneid&quot;) %&gt;% 
  inner_join(eggnogs, by = &quot;geneid&quot;) %&gt;% 
  mutate(aquifer = &quot;MM-171.3&quot;) %&gt;% 
  inner_join(de_mm_test, by = &quot;geneid&quot;) -&gt; de_mm
de_tm$table %&gt;% 
  rownames_to_column(&quot;geneid&quot;) %&gt;% 
  inner_join(eggnogs, by = &quot;geneid&quot;) %&gt;% 
  mutate(aquifer = &quot;TM-448.4&quot;) %&gt;%
  inner_join(de_tm_test, by = &quot;geneid&quot;) -&gt; de_tm

de_mm %&gt;% rbind(de_tm) %&gt;% as_tibble() -&gt; diffexp

remove(de_mm, de_tm, de_mm_test, de_tm_test, d, design.mat, fit)</code></pre>
<p>Fig. 6 and 7: please organise the y-axis according to function by grouping these categories, e.g., into “biofilm formation”, “motility”, “energy conservation”, “transporters”, “biosynthesis”… For the bar plot, the 11 most abundant taxonomic groups are selected (either phyla, classes or orders)</p>
<pre class="r"><code>cols &lt;- c(&quot;Other&quot; = &quot;#a6cee3&quot;, &quot;Unidentified&quot; = &quot;#d9d9d9&quot;,
          &quot;Acholeplasmatales&quot; = &quot;#969696&quot;, &quot;Bacillales&quot; = &quot;#b15928&quot;, &quot;Burkholderiales&quot; = &quot;#1f78b4&quot;,
          &quot;Campylobacterales&quot; = &quot;#33a02c&quot;, &quot;Desulfobacterales&quot; = &quot;#e31a1c&quot;,
          &quot;Methanobacteriales&quot; = &quot;#ff7f00&quot;,
          &quot;Nitrosomonadales&quot; = &quot;#fdbf6f&quot;, &quot;Rhizobiales&quot; = &quot;#cab2d6&quot;,
          &quot;Rhodocyclales&quot; = &quot;#6a3d9a&quot;, &quot;Trichomonadida&quot; = &quot;#ffff99&quot;)

i &lt;- c(&quot;Translation, ribosomal structure and biogenesis&quot;,&quot;Cell wall/membrane/envelope biogenesis&quot;,
       &quot;Energy production and conversion&quot;,&quot;Cell motility&quot;,&quot;Posttranslational modification, protein turnover, chaperones&quot;,&quot;Signal transduction mechanisms&quot;,
       &quot;Amino acid transport and metabolism&quot;,
       &quot;Replication, recombination and repair&quot;,&quot;Carbohydrate transport and metabolism&quot;,
       &quot;Intracellular trafficking, secretion, and vesicular transport&quot;)

seqcpm %&gt;%
  filter(geneid %in% diffexp$geneid) %&gt;%
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  group_by(order, sample) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = &quot;drop_last&quot;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order)) %&gt;%
  top_n(10, mean_relab) -&gt; t

ncbi %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  mutate(topphylum = if_else(is.na(order), &quot;Unidentified&quot;, topphylum)) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref</code></pre>
<pre class="r"><code>seqcpm %&gt;%
  inner_join(diffexp, by = &quot;geneid&quot;) %&gt;% filter(diffexp != 0 &amp; eggnog %in% i) %&gt;%
  inner_join(taxref, by = &#39;geneid&#39;) %&gt;%
  mutate(diffexp = if_else(diffexp == 1, &quot;Planktonic&quot;, &quot;Biofilm&quot;)) %&gt;%
  mutate(aquifer = paste(aquifer, diffexp)) %&gt;%
  group_by(topphylum, eggnog, aquifer) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  ggplot(aes(x = cpm, 
             y = fct_relevel(eggnog, i),
             fill = fct_relevel(topphylum, c(&#39;Other&#39;,&#39;Unidentified&#39;)) %&gt;% fct_rev()
             )
         ) +
  labs(x = expression(bold(&#39;Counts per million (&#39; %*% &#39;1000)&#39;)), y = &#39;&#39;) +
  facet_wrap(~ aquifer, nrow = 1, ncol = 4) +
  geom_col() +
  scale_fill_manual(values = cols) +
  scale_x_continuous(labels = c(expression(bold(&quot;0&quot;)), 
                                &quot;&quot;, 
                                expression(bold(&quot;100&quot;)),
                                &quot;&quot;,
                                expression(bold(&quot;200&quot;)), 
                                &quot;&quot;), 
                     n.breaks = 5, expand = c(0.004,0)) +  
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  theme_tidy(ratio = 3.4) + 
  theme(panel.border = element_blank(),
        strip.background = element_rect(size = 0,fill = &quot;#f0f0f0&quot;),
        axis.ticks.y = element_blank(),
        legend.position = c(1,0.72),
        strip.text = element_text(face = &quot;bold&quot;),
        axis.text.y = element_text(face = &quot;plain&quot;, size = 6, colour = &quot;black&quot;),
        axis.text.x = element_text(face = &quot;bold&quot;, size = 6),
        legend.spacing.x = unit(0.5, &quot;mm&quot;),
        legend.box.margin = margin(0,0,0,0),
        legend.box.background = element_rect(size = .7, colour = &quot;black&quot;),
        legend.key.size = unit(3.3, &quot;mm&quot;),
        axis.line.x.bottom = element_line(colour = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/plot%20significant%20DE%20eggnogs-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&#39;figures/fig_5.pdf&#39;, height = 10, width = 16, units = &#39;cm&#39;)
ggsave(&#39;figures/fig_5.png&#39;, height = 10, width = 16, units = &#39;cm&#39;)</code></pre>
<pre class="r"><code>ncbi %&gt;% 
  filter(kingdom == &quot;Fungi&quot;) %&gt;% 
  inner_join(seqcpm, by = &quot;geneid&quot;) %&gt;%
  mutate(rank = substr(rank, start = 1, stop = 1)) %&gt;%
  mutate(taxon = paste(taxon, &quot; (&quot;, rank, &quot;)&quot;, sep = &quot;&quot;)) %&gt;%
  group_by(sample, taxon) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  #mutate(rank = substr(rank, start = 1, stop = 1)) %&gt;%
  #mutate(taxon = paste(taxon, &quot; (&quot;, rank, &quot;)&quot;, sep = &quot;&quot;)) %&gt;%
  ggplot(aes(x = cpm, 
             y = fct_rev(fct_relevel(sample, c(&quot;P10152_S2&quot;,&quot;P10152_S3&quot;))),
             fill = taxon)) +
  labs(x = &#39;Counts per million&#39;, y = &#39;&#39;) +
  geom_col() +
  scale_fill_brewer(palette = &#39;Paired&#39;) +
  coord_cartesian(clip = &quot;off&quot;) +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE), expand = c(0.004,0)) +
  annotate(geom = &quot;text&quot;, x = 8.25e3, y = 1, hjust = 0, size = 6/.pt, label = &quot;TM planktonic&quot;) +
  annotate(geom = &quot;text&quot;, x = 8.25e3, y = 2.5, hjust = 0, size = 6/.pt, label = &quot;MM planktonic&quot;) +
  annotate(geom = &quot;text&quot;, x = 8.25e3, y = 4.5, hjust = 0, size = 6/.pt, label = &quot;TM biofilm&quot;) +
  annotate(geom = &quot;text&quot;, x = 8.25e3, y = 6.5, hjust = 0, size = 6/.pt, label = &quot;MM biofilm&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 0.55, yend = 1.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 1.55, yend = 3.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 3.55, yend = 5.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 5.55, yend = 7.45, lwd = 0.5, geom = &quot;segment&quot;) +
  theme_clean(ratio = 1.4) +
  theme(legend.spacing.x = unit(0.5, &quot;mm&quot;),
        panel.grid.major.y = element_blank(),
        legend.key.size = unit(3.5, &quot;mm&quot;),
        axis.line.x.bottom = element_line(colour = &quot;black&quot;)
  )</code></pre>
<p><img src="flowcells_files/figure-html/fungi-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_s4.pdf&quot;, height = 10, width = 10, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_s4.png&quot;, height = 10, width = 10, units = &quot;cm&quot;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
