---
title: "Flowcells Äspö HRL"
subtitle: "Biofilm formation in deep biosphere at Äspö HRL"
author: "George Westmeijer"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE}
library(tidyverse)
```


## Define theme and function(s)

```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
source("/Users/geweaa/Box Sync/Protocols/barplot.R")
```

## Read files


```{r input files, message=FALSE, warning=FALSE}
smd <- read_tsv('smd.tsv', col_types = cols(.default = col_character()))

# Important difference between NCBI and GTDB databases is within the Betaproteobacteria class
# Thiobacillus (GTDB) Gammaproteobacteria - Burkholderiales - Thiobacillaceae
# Thiobacillus (NCBI) Betaproteobacteria - Nitrosomonadales - Thiobacillaceae
# Sulfurimonas (GTDB) Campylobacterota - Campylobacteria - Campylobacterales - Sulfurimonadaceae
# Sulfurimonas (NCBI) Proteobacteria - Epsilonproteobacteria - Campylobacterales - Thiovulaceae

gtdb <- read_tsv("ASV-ncbi.tsv", col_types = cols(.default = col_character())) %>%
  mutate_if(grepl('uncultured',.), ~replace(., grepl('uncultured', .), NA)) %>%
  mutate(taxon = coalesce(genus,family,order,class,phylum,domain))

# Throughout the study, the taxonomy from NCBI is adopted
# ASVs are annotated using the GTDB database (release 207 - April 2022)
# NCBI taxonomy was assigned using database version late Jan 2023

read_tsv('ASV-table.tsv', col_types = cols(.default = col_character(), count = col_integer())) %>%
  # Filter out samples with less than 1000 reads
  group_by(sample) %>% filter(sum(count) > 1000) %>%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %>% ungroup() -> seqtab
```


## Rarefaction


```{r rarefaction}
seqtab %>%
  inner_join(smd, by = 'sample') %>%
  group_by(env, seqid) %>% summarise(count = sum(count), .groups = 'drop') %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('env') -> seqtab_matrix

min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, sample = ., step = 100, xlab = 'Sequencing depth (No. reads)', ylab = 'Diversity (No. ASVs)', label = TRUE)
```


## Alpha diversity using Shannon-Weaver index


```{r calculcate Shannon, message=FALSE, warning=FALSE}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>% 
  gather(seqid, shannon, -1) %>%
  rename(shannon = 1) %>%
  rownames_to_column('sample') %>%
  inner_join(smd, by = "sample") %>% as_tibble() -> adiv
```


```{r statistics alpha diversity}
stat <- adiv %>% 
  group_by(designation, nature, age) %>% 
  summarise(shannon = mean(shannon), .groups = "drop")

suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(stat$shannon)
# Preferably all the dots on the diagonal
car::qqPlot(stat$shannon)
# Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(shannon ~ env, data = stat)
# Run ANOVA (null hypothesis diversity is equal)
aov(shannon ~ designation/nature, data = stat) %>% summary()
# Post-hoc
aov(shannon ~ paste(designation,nature), data = stat) %>% TukeyHSD()
```


Plot alpha diversity


```{r plot adiv}
adiv %>%
  mutate(age = if_else(nature == "planktonic", "", age)) %>%
  mutate(env = paste(designation, nature)) %>%
  mutate(age = ifelse(sample %in% c("QL-1679-S75","QL-1679-S74","QL-1679-S76","QL-1679-S80"), "40/75 d",age)) %>%
  ggplot(aes(x = shannon, y = fct_relevel(env, c("TM-448.4 biofilm","TM-448.4 planktonic",
                                                 "MM-171.3 biofilm","MM-171.3 planktonic")))) +
  geom_boxplot(outlier.color = "white") +
  geom_jitter(width = 0.01, stroke = 0.8, size = 1, fill = "white", shape = 21, height = 0.1) +
  labs(x = "Alpha diversity (Shannon index)", y = "") +
  scale_x_continuous(limits = c(1,6), n.breaks = 4) +
  ggrepel::geom_label_repel(aes(label = age), color = "black",
                  box.padding = 0.4, label.padding = 0.1, 
                  max.overlaps = 10,
                  min.segment.length = 0.3,
                  size = 7/.pt) +
  theme_tidy(ratio = 1.4, fontsize = 7) +
  theme(plot.tag.position = c(.2,1.01),
        axis.ticks.y = element_blank(),
        panel.border = element_rect(),
        panel.grid.major.x = element_line(linetype = "dotted", color = "grey", size = 0.3),
        axis.text.y.left = element_text(face = "bold")
        ) -> p1
```


## Fig. 3b: NMDS on dataset DNA

```{r run nmds, results='hide'}
seqtab %>%
  group_by(seqid) %>% filter(count > 2) %>% ungroup() %>%
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") -> t

set.seed(999)
t %>%  vegan::metaMDS(trymax = 50, k = 2, autotransform = T) -> nmds.dna
```

Extract the scores

```{r nmds.scores}
vegan::scores(nmds.dna)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores.dna
```

Plot the ordination

```{r nmds plot, include = T}
nmds.scores.dna %>% select(1:3) %>% inner_join(smd, by = "sample") -> t

#getting the convex hull of each unique point set
find_hull <- function(t) t[chull(t$NMDS1, t$NMDS2), ]
hulls <- plyr::ddply(t, "nature", find_hull)

nmds.scores.dna %>%
  # Prepare data
  mutate(age = gsub("d.[1-2]"," d",age)) %>%
  mutate(age = ifelse(sample %in% c("QL-1679-S75","QL-1679-S74","QL-1679-S76","QL-1679-S80"), "40/75 d",age)) %>%
  mutate(age = ifelse(nature == "planktonic", substr(age,1,1), age)) %>%
  # Plot
  ggplot(aes(x = NMDS1, y = NMDS2, fill = nature)) +
  geom_polygon(data = hulls, alpha = .3) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  # Labelled according to groundwater type
  scale_color_manual(values = c(biofilm = "#08519c", planktonic = "#9ecae1"), guide = "none") +
  scale_fill_manual(values = c(biofilm = "#08519c", planktonic = "#9ecae1"), guide = "none") +
  annotate('text', x = Inf, y = Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds.dna$stress, digits = 2)),
           hjust = 1, vjust = 1, fontface = "bold"
           ) +
  annotate('text', x = 0.8, y = -0.5, size = 8/.pt, 
           label = "Biofilm", fontface = "bold", color = "#08519c"
           ) +
  annotate('text', x = -0.6, y = 0.2, size = 8/.pt, 
           label = "Planktonic", fontface = "bold", color = "#9ecae1"
           ) +
  ggrepel::geom_text_repel(aes(label = age), color = "black",
                  box.padding = 0.3, 
                  max.overlaps = 12,
                  min.segment.length = 0.5,
                  size = 7/.pt) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme_tidy(fontsize = 7) +
  theme(legend.spacing.x = unit(0.0, "mm"),
        legend.margin = margin(0,0,0,0),
        legend.position = c(0.1,0.1),
        legend.text = element_text(size = 7, face = "bold"),
        plot.tag.position = c(.02,1.01)
        ) -> p2
```


```{r Fig. 3}
library(patchwork)
p1 + p2 + plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(size = 8, color = "black", face = "bold"))
```

```{r export Fig. 3}
ggsave("figures/fig_3.pdf", width = 18, height = 12, units = "cm")
ggsave("figures/fig_3.png", width = 18, height = 12, units = "cm")
```


Statistics on beta diversity: PERMANOVA


```{r permanova}
seqtab %>%
  select(-relab) %>%
  inner_join(smd[,c("sample","nature")], by = "sample") %>% 
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") -> t

vegan::adonis2(t[,c(-1)] ~ nature, data = t)

seqtab %>%
  select(-relab) %>%
  inner_join(smd[,c("sample","nature")], by = "sample") %>% 
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") -> t
  
pairwiseAdonis::pairwise.adonis(t[,c(-1)], factors = t$env)
```

```{r differential abundance}
t <- seqtab %>% 
  select(-relab) %>%
  arrange(sample) %>%
  pivot_wider(names_from = "sample", values_from = "count", values_fill = 0) %>%
  column_to_rownames("seqid")

i <- smd %>% 
  filter(!is.na(age)) %>% 
  arrange(sample)

t <- ALDEx2::aldex(reads = t, conditions = i$nature)

t %>% filter(we.eBH <= 0.05) %>% 
  rownames_to_column("seqid") %>%
  inner_join(gtdb, by = "seqid") %>% View()
```




## Fig. S2:  Bar plot DNA phylum


```{r bar plot DNA phylum}
cols <- c("Unidentified" = "#A6CEE3",
          'Other' = '#1F78B4',
          'Alphaproteobacteria' = '#B2DF8A',
          'Betaproteobacteria' = '#33A02C',
          'Deltaproteobacteria' = '#FB9A99',
          'Epsilonproteobacteria' = '#E31A1C',
          'Gammaproteobacteria' = '#FDBF6F',
          'Bacteroidetes' = '#FF7F00', 
          'Chloroflexi' = '#CAB2D6',
          "Omnitrophota" = '#6A3D9A',
          'Patescibacteria' = '#FFFF99', 
          'Spirochaetes' = '#B15928')

i <- c("QK1636-S12","QK1636-S24","QK1636-S36", # MM before
       "QL-1679-S77","QL-1679-S72", # MM 20 d
       "QL-1679-S71","QL-1679-S78", # MM 40 d
       "QL-1679-S73","QL-1679-S79", # MM 75 d
       "QL-1679-S83","QL-1679-S86","QL-1679-S88", # MM after
       "QK1636-S04","QK1636-S16","QK1636-S28", # TM before
       "QL-1679-S81","QL-1679-S82", # TM 20 d
       "QL-1679-S74","QL-1679-S80", # TM 40 d
       "QL-1679-S75","QL-1679-S76",  # TM 75 d
       "QL-1679-S90","QL-1679-S92","QL-1679-S94") # TM after

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(gtdb, by = "seqid") %>%
  mutate(phylum = if_else(phylum == "Proteobacteria", class, phylum)) %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum) & phylum != "uncultured bacterium") %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  mutate(phylum = if_else(phylum == "Proteobacteria", class, phylum)) %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other",
                                             "Alphaproteobacteria","Betaproteobacteria",
                                             "Deltaproteobacteria","Epsilonproteobacteria",
                                             "Gammaproteobacteria")
                                )
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Phylum") +
  geom_col() +
  scale_fill_manual(values = cols) +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  5.45, y = 101, yend = 101) +
  annotate("segment", x =  5.55, xend =  7.45, y = 101, yend = 101) +
  annotate("segment", x =  7.55, xend =  9.45, y = 101, yend = 101) +
  annotate("segment", x =  9.55, xend = 12.45, y = 101, yend = 101) +
  annotate("segment", x = 12.55, xend = 15.45, y = 101, yend = 101) +
  annotate("segment", x = 15.55, xend = 17.45, y = 101, yend = 101) +
  annotate("segment", x = 17.55, xend = 21.45, y = 101, yend = 101) +
  annotate("segment", x = 21.55, xend = 24.45, y = 101, yend = 101) +
  annotate("segment", x = 12.50, xend = 12.50, y = 100.75, yend = 105) +
  annotate('text', x =    2, y = 103, size = 7 / .pt, label = "MM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  4.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  6.5, y = 103, size = 7 / .pt, label = "40 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  8.5, y = 103, size = 7 / .pt, label = "75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  11,  y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  14, y = 103, size = 7 / .pt, label = "TM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  16.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  19.5, y = 103, size = 7 / .pt, label = "40 + 75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  23, y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        ) -> p1
```


```{r export barplot DNA phylum}
ggsave("figures/fig_s3.pdf", height = 10, width = 16, units = "cm")
ggsave("figures/fig_s3.png", height = 10, width = 16, units = "cm")
```


```{r barplot DNA order}
cols <- c("Unidentified" = "#a6cee3",
          "Other" = "#1f78b4",
          "Hyphomicrobiales" = "#B2DF8A", # Alpha
          "Burkholderiales" = "#33A02C", # Beta
          "Nitrosomonadales" = "#FB9A99", # Beta
          "Desulfobacterales" = "#E31A1C", # Delta
          "Desulfovibrionales" = "#FDBF6F", # Delta
          "Campylobacterales" = "#FF7F00", # Epsilon
          "Methylococcales" = "#CAB2D6", # Gamma
          "Pseudomonadales" = "#6A3D9A", # Gamma
          "Anaerolineales" = "#FFFF99", 
          "Patescibacteria" = "#B15928")

seqtab %>%
  inner_join(gtdb, by = "seqid") %>%
  mutate(order = if_else(phylum == "Patescibacteria", phylum, order)) %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  mutate(order = if_else(phylum == "Patescibacteria", phylum, order)) %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other","Hyphomicrobiales",
                                             "Burkholderiales","Nitrosomonadales",
                                             "Desulfobacterales","Desulfovibrionales",
                                             "Campylobacterales","Methylococcales","Pseudomonadales"))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Order") +
  geom_col() +
  scale_fill_manual(values = cols) +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  5.45, y = 101, yend = 101) +
  annotate("segment", x =  5.55, xend =  7.45, y = 101, yend = 101) +
  annotate("segment", x =  7.55, xend =  9.45, y = 101, yend = 101) +
  annotate("segment", x =  9.55, xend = 12.45, y = 101, yend = 101) +
  annotate("segment", x = 12.55, xend = 15.45, y = 101, yend = 101) +
  annotate("segment", x = 15.55, xend = 17.45, y = 101, yend = 101) +
  annotate("segment", x = 17.55, xend = 21.45, y = 101, yend = 101) +
  annotate("segment", x = 21.55, xend = 24.45, y = 101, yend = 101) +
  annotate("segment", x = 12.50, xend = 12.50, y = 100.75, yend = 105) +
  annotate('text', x =    2, y = 103, size = 7 / .pt, label = "MM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  4.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  6.5, y = 103, size = 7 / .pt, label = "40 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  8.5, y = 103, size = 7 / .pt, label = "75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  11,  y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  14, y = 103, size = 7 / .pt, label = "TM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  16.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  19.5, y = 103, size = 7 / .pt, label = "40 + 75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  23, y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  theme_barplot(fontsize = 7, ratio = 0.7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        ) #-> p1
```
```{r export Fig. 4}
ggsave("figures/fig_4.pdf", height = 10, width = 18, units = "cm")
ggsave("figures/fig_4.png", height = 10, width = 18, units = "cm")
```


```{r barplot DNA genus}
cols <- c("Unidentified" = "#A6CEE3",
          "Other" = "#1F78B4",
          "Filomicrobium" = "#B2DF8A", # Hyphomicrobiales (Alpha)
          "Hoeflea" = "#33A02C", # Burkholderiales (Beta)
          "Ralstonia" = "#33A02C", # Burkholderiales (Beta)
          "Methylotenera" = "#FB9A99", # Nitrosomonadales (Beta)
          "Thiobacillus" = "#FB9A99", # Nitrosomonadales (Beta)
          "Desulfopila" = "#E31A1C", # Desulfobacterales (Delta)
          "DB1" = "#E31A1C", # Desulfobacterales (Delta)
          "UBA2262" = "#E31A1C", # Desulfobacterales (Delta)
          "Methylotuvimicrobium" = "#CAB2D6", # Methylococcales (Gamma)
          "JAHITL01" = "#B15928") # Patescibacteria     

i <- c("QL-1679-S77","QL-1679-S72", # MM 20 d
       "QL-1679-S71","QL-1679-S78", # MM 40 d
       "QL-1679-S73","QL-1679-S79", # MM 75 d
       "QL-1679-S81","QL-1679-S82", # TM 20 d
       "QL-1679-S74","QL-1679-S80", # TM 40 d
       "QL-1679-S75","QL-1679-S76")  # TM 75 d


biofilm <- smd %>% filter(nature == "biofilm" & !is.na(age)) %>% pull(sample)
          
seqtab %>%
  filter(sample %in% biofilm) %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(genus, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(genus) & !grepl(pattern = "uncultured", x = genus)) %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(genus, topphylum = genus), by = "genus") %>%
  mutate(topphylum = if_else(is.na(genus), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% biofilm) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = topphylum)
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Genus") +
  geom_col() +
  scale_fill_brewer(palette = "Paired") +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  5.45, y = 101, yend = 101) +
  annotate("segment", x =  5.55, xend =  7.45, y = 101, yend = 101) +
  annotate("segment", x =  7.55, xend =  9.45, y = 101, yend = 101) +
  annotate("segment", x =  9.55, xend = 12.45, y = 101, yend = 101) +
  annotate("segment", x = 12.55, xend = 15.45, y = 101, yend = 101) +
  annotate("segment", x = 15.55, xend = 17.45, y = 101, yend = 101) +
  annotate("segment", x = 17.55, xend = 21.45, y = 101, yend = 101) +
  annotate("segment", x = 21.55, xend = 24.45, y = 101, yend = 101) +
  annotate("segment", x = 12.50, xend = 12.50, y = 100.75, yend = 105) +
  annotate('text', x =    2, y = 103, size = 7 / .pt, label = "MM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  4.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  6.5, y = 103, size = 7 / .pt, label = "40 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  8.5, y = 103, size = 7 / .pt, label = "75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  11,  y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  14, y = 103, size = 7 / .pt, label = "TM before", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  16.5, y = 103, size = 7 / .pt, label = "20 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  19.5, y = 103, size = 7 / .pt, label = "40 + 75 d", hjust = 0.5, fontface = "bold") +
  annotate('text', x =  23, y = 103, size = 7 / .pt, label = "after", hjust = 0.5, fontface = "bold") +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 0.5, fill = NA)
        ) #-> p2
```


```{r Fig. 4}
library(patchwork)
p1 + p2 + plot_layout(nrow = 2)
```


```{r export Fig. 4}
ggsave("figures/fig_s3.pdf", height = 18, width = 18, units = "cm")
ggsave("figures/fig_s3.png", height = 18, width = 18, units = "cm")
```


## Load gene counts, taxonomy and sample metadata


```{r read data, message=FALSE, warning=FALSE}
read_tsv('final.contigs.tsv', col_types = cols(.default = col_character(),
                                               count = col_integer()
                                               )
         )  -> rnatab

# Read taxonomy (Prokka)
read_tsv('prokka.tsv', col_types = cols(.default = col_character(),
                                        length_bp = col_integer()
                                        )
         ) %>% 
  rename(geneid = locus_tag) -> prokka

# Taxonomy (NCBI)
emapper <- read_tsv("emapper-annotations.tsv", col_types = cols(.default = col_character()))
ncbi <- read_tsv('ncbi-taxonomy.tsv', col_types = cols(.default = col_character()))
```


## Fig. 5a: NMDS on Bray-Curtis dissimilarities (RNA)


Transform the genecounts



```{r run NMDS on RNA, results='hide'}
rnatab %>%
  select(geneid, sample, count) %>% spread(geneid, count, fill = 0) %>%
  column_to_rownames("sample") -> t

set.seed(999)
t %>%  vegan::metaMDS(trymax = 20, k = 2, autotransform = T) -> nmds
```

Extract the scores

```{r nmds.scores RNA}
vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores
```

Plot the ordination

```{r NMDS plot RNA, include = T}
nmds.scores %>% select(1:3) %>% inner_join(smd, by = "sample") -> df

#getting the convex hull of each unique point set
find_hull <- function(df) df[chull(df$NMDS1, df$NMDS2), ]
hulls <- plyr::ddply(df, "nature", find_hull)

nmds.scores %>%
  ggplot(aes(x = NMDS1, y = NMDS2, fill = nature)) +
  geom_polygon(data = hulls,alpha = .3) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  # Labelled according to groundwater type
  scale_color_manual(values = c(biofilm = "#08519c", planktonic = "#9ecae1"), guide = "none") +
  scale_fill_manual(values = c(biofilm = "#08519c", planktonic = "#9ecae1"), guide = "none") +
  annotate('text', x = Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = 1.1, vjust = -1, fontface = "bold",
           ) +
  annotate('text', x = -0.25, y = -0.15, size = 7/.pt, 
           label = "Biofilm", fontface = "bold", color = "#08519c"
           ) +
  annotate('text', x = 0.25, y = 0.2, size = 7/.pt, 
           label = "Planktonic", fontface = "bold", color = "#9ecae1"
           ) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme_tidy(fontsize = 7) +
  theme(legend.spacing.x = unit(0.0, "mm"),
        legend.box.spacing = unit(-0.5,"cm"),
        plot.margin = margin(0,0,0,0)) -> p1
```


## Fig. 5b: Barplot mRNA

Transform absolute counts to counts per million (cpm) which is the relative abundance within each sample.


```{r calculate cpm from raw counts}
rnatab %>%
  filter(geneid %in% ncbi$geneid) %>%
  spread(sample, count, fill = 0) %>%
  column_to_rownames('geneid') %>%
  edgeR::DGEList(group = na.omit(smd$treat)) %>% edgeR::cpm() %>%
  as.data.frame() %>%
  rownames_to_column('geneid') %>%
  gather(sample, cpm , -1) %>% filter(cpm > 0) %>% 
  as_tibble() -> seqcpm
```


```{r select 11 most abundant orders RNA}
seqcpm %>%
  # Only use transcripts annotated on phylum level
  inner_join(ncbi, by = "geneid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(12, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r barplot mRNA order level}
cols <- c("Unidentified" = "#a6cee3",
          "Other" = "#1f78b4",
          "Hyphomicrobiales" = "#B2DF8A", # Alpha
          "Burkholderiales" = "#33A02C", # Beta
          "Nitrosomonadales" = "#FB9A99", # Beta
          "Rhodocyclales" = "#A50026", # Beta
          "Desulfobacterales" = "#E31A1C", # Delta
          "Campylobacterales" = "#FF7F00", # Epsilon
          "Enterobacterales" = "#FDBF6F", # Gamma
          "Acholeplasmatales" = "#00441B", # Tenericutes
          "Bacillales" = "#CAB2D6", # Bacillota
          "Eubacteriales" = "#6A3D9A", # Bacillota
          "Methanobacteriales" = "#FFFF99", # Euryarchaeota
          "Trichomonadida" = "#B15928") # Parabasalia


seqcpm %>%
  inner_join(taxref, by = "geneid") %>%
  # Combine water type and origin
  inner_join(smd, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  # Call plot
  ggplot(aes(x = fct_relevel(sample, c('P4055_101','P4055_102','P10152_S2','P10152_S3',
                                       'P4056_101','P4056_102','P10152_S1','P10152_S4')), 
             y = cpm, 
             fill = fct_relevel(topphylum, c('Unidentified','Other',
                                             "Hyphomicrobiales",
                                             "Burkholderiales","Nitrosomonadales","Rhodocyclales",
                                             "Desulfobacterales",
                                             "Campylobacterales",
                                             "Enterobacterales",
                                             "Acholeplasmatales","Bacillales","Eubacteriales"))
             )
         ) +
  labs(y = expression(bold('Counts per million (' %*% '1000)')), x = '', fill = "Order") +
  scale_y_continuous(labels = c(expression(bold("0")), 
                                expression(bold("250")), 
                                expression(bold("500")), 
                                expression(bold("750")), 
                                expression(bold("1000"))), 
                     ) +
  geom_col() +
  scale_fill_manual(values = cols) +
  annotate(geom = "segment", x = .55, xend = 2.45, y = 1.01e6, yend = 1.01e6) +
  annotate(geom = "segment", x = 2.55, xend = 4.45, y = 1.01e6, yend = 1.01e6) +
  annotate(geom = "segment", x = 4.55, xend = 6.45, y = 1.01e6, yend = 1.01e6) +
  annotate(geom = "segment", x = 6.55, xend = 8.45, y = 1.01e6, yend = 1.01e6) +
  annotate('text', x = 1.5, y = 1.03e6, size = 7 / .pt, label = "MM planktonic", hjust = 0.5) +
  annotate('text', x = 3.5, y = 1.03e6, size = 7 / .pt, label = "MM biofilm",hjust = 0.5) +
  annotate('text', x = 5.5, y = 1.03e6, size = 7 / .pt, label = "TM planktonic",hjust = 0.5) +
  annotate('text', x = 7.5, y = 1.03e6, size = 7 / .pt, label = "TM biofilm",hjust = 0.5) +
  theme_barplot(ratio = 1, fontsize = 7) + 
  guides(fill = guide_legend(nrow = 5)) +
  theme(legend.key.size = unit(3, "mm"),
        panel.border = element_rect(size = 0.5, fill = NA),
        legend.box.spacing = unit(-2,"cm"),
        plot.margin = margin(0,0,0,4),
        axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.ticks.y = element_line(),
        legend.text = element_text(size = 7), legend.title = element_text(size = 7)
        ) -> p2
```


```{r Fig. 5}
library(patchwork)
p1 + p2 + plot_annotation(tag_levels = "a") & theme(plot.tag = element_text(size = 8, color = "black", face = "bold"))
```

```{r export Fig. 5}
ggsave("figures/fig_5x.pdf", height = 12, width = 18, units = "cm")
ggsave("figures/fig_5x.png", height = 12, width = 18, units = "cm")
```


```{r PERMANOVA on Bray-Curtis dissimilarities RNA}
set.seed(999)

rnatab %>%
  inner_join(smd[,c("sample","nature")], by = "sample") %>% 
  spread(geneid, count, fill = 0) %>%
  column_to_rownames("sample") -> t

vegan::adonis(t[,c(-1)] ~ nature, data = t)
```


## Fig. 5: Differential gene expression using EdgeR


```{r differential expression}
# A Planktonic MM
# B Biofilm MM
# C Planktonic TM
# D Biofilm TM

library(edgeR)
rnatab %>% spread(sample, count, fill = 0) %>% column_to_rownames('geneid') -> d
DGEList(d, group = smd$treat %>% na.omit()) -> d

# Filter based on cpm
keep <- filterByExpr(d)
d <- d[keep, , keep.lib.sizes = F]
# TTM normalization to correct for compositional bias
calcNormFactors(d) -> d

# Design matrix based on experimental design
design.mat <- model.matrix(~ 0 + smd$treat)
rownames(design.mat) <- colnames(d)
colnames(design.mat) <- levels(d$samples$group)

# Estimate gene expression dispersion
d %>% estimateDisp(design = design.mat, robust = F) -> d
# Fit a GLM and run the test
fit <- glmFit(d, design = design.mat)

# Genes which baseline differences between planktonic and biofilm
i <- makeContrasts(mm = A-B, tm = C-D, levels = design.mat)

de_mm <- glmLRT(fit, contrast = i[, "mm"])
de_tm <- glmLRT(fit, contrast = i[, "tm"])
```


```{r extract data DE analysis}
de_mm_test <- decideTestsDGE(de_mm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")
de_tm_test <- decideTestsDGE(de_tm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")

de_mm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "MM-171.3") %>% 
  inner_join(de_mm_test, by = "geneid") -> de_mm
de_tm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "TM-448.4") %>%
  inner_join(de_tm_test, by = "geneid") -> de_tm

de_mm %>% rbind(de_tm) %>% as_tibble() -> diffexp

remove(de_mm, de_tm, de_mm_test, de_tm_test, d, design.mat, fit)
```


Fig. 6 and 7: please organise the y-axis according to function by grouping these categories, e.g., into "biofilm formation", "motility", "energy conservation", "transporters", "biosynthesis"... For the bar plot, the 11 most abundant taxonomic groups are selected (either phyla, classes or orders)


```{r prepare plot DE eggnogs}
cols <- c("Unidentified" = "#a6cee3",
          "Other" = "#1f78b4",
          "Hyphomicrobiales" = "#B2DF8A", # Alpha
          "Burkholderiales" = "#33A02C", # Beta
          "Nitrosomonadales" = "#FB9A99", # Beta
          "Rhodocyclales" = "#A50026", # Beta
          "Desulfobacterales" = "#E31A1C", # Delta
          "Campylobacterales" = "#FF7F00", # Epsilon
          "Acholeplasmatales" = "#00441B", # Firmicutes
          "Bacillales" = "#CAB2D6", # Firmicutes
          "Methanobacteriales" = "#FFFF99",
          "Trichomonadida" = "#B15928")

i <- c("Energy production and conversion", # C
       "Translation, ribosomal structure and biogenesis", # J
       "Inorganic ion transport and metabolism", # P
       "Cell motility", # N
       "Amino acid transport and metabolism", # E
       "Transcription", # K
       "Carbohydrate transport and metabolism", # G
       "Intracellular trafficking, secretion, and vesicular transport", # U
       "Signal transduction mechanisms", # T
       "Cell wall/membrane/envelope biogenesis") # M

seqcpm %>%
  filter(geneid %in% diffexp$geneid) %>%
  inner_join(cogs, by = "geneid") %>% filter(COG_ref %in% i) %>%
  inner_join(ncbi, by = "geneid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(10, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r plot significant DE eggnogs}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>% 
  inner_join(cogs, by = "geneid") %>%
  rename(reference = COG_ref) %>%
  filter(diffexp != 0 & reference %in% i) %>%
  inner_join(taxref, by = 'geneid') %>%
  mutate(diffexp = if_else(diffexp == 1, "planktonic", "biofilm")) %>%
  mutate(aquifer = paste(aquifer, diffexp)) %>%
  group_by(topphylum, reference, aquifer) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = fct_relevel(reference, i),
             fill = fct_relevel(topphylum, rev(names(cols))))
         ) +
  labs(x = expression(bold('Counts per million (' %*% '1000)')), y = '', fill = "Order") +
  facet_wrap(~ aquifer, nrow = 1, ncol = 4) +
  geom_col() +
  scale_fill_manual(values = cols) +
  scale_x_continuous(labels = c(expression(bold("0")), 
                                "", 
                                expression(bold("100")),
                                "",
                                expression(bold("200")), 
                                ""), 
                     n.breaks = 5, expand = c(0.004,0)) +  
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  theme_tidy(ratio = 3.4) + 
  theme(panel.border = element_blank(),
        strip.background = element_rect(size = 0,fill = "#f0f0f0"),
        axis.ticks.y = element_blank(),
        legend.position = c(1,0.72),
        strip.text = element_text(face = "bold"),
        axis.text.y = element_text(face = "plain", size = 6, colour = "black"),
        axis.text.x = element_text(face = "bold", size = 6),
        legend.spacing.x = unit(0.5, "mm"),
        legend.box.margin = margin(0,0,0,0), legend.title = element_text(size = 6),
        legend.box.background = element_rect(size = .7, colour = "black"),
        legend.key.size = unit(3.3, "mm"),
        axis.line.x.bottom = element_line(colour = "black"))
```


```{r export plot DE eggnogs}
ggsave('figures/fig_6x.pdf', height = 10, width = 16, units = 'cm')
ggsave('figures/fig_6x.png', height = 10, width = 16, units = 'cm')
```


```{r fungi}
ls <- list(sample = "P4056_102", taxon = "Fungi (kingdom)", cpm = 0)

ncbi %>% 
  filter(kingdom == "Fungi") %>% 
  inner_join(seqcpm, by = "geneid") %>%
  mutate(taxon = paste(taxon, " (", rank, ")", sep = "")) %>%
  group_by(sample, taxon) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  rbind(ls) %>%
  ggplot(aes(x = fct_relevel(sample, c('P4055_101','P4055_102','P10152_S2','P10152_S3',
                                       'P4056_101','P4056_102','P10152_S1','P10152_S4')), 
             y = cpm, 
             fill = taxon
             )
         ) +
  labs(y = 'Counts per million', x = '', fill = "Taxon") +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  scale_y_continuous(n.breaks = 4, labels = c("0","3,000","6,000","0")) +
  annotate(geom = "text", x = 1.5, y = 8.4e3, size = 7/.pt, label = "MM planktonic") +
  annotate(geom = "text", x = 3.5, y = 8.4e3, size = 7/.pt, label = "MM biofilm") +
  annotate(geom = "text", x = 5.5, y = 8.4e3, size = 7/.pt, label = "TM planktonic") +
  annotate(geom = "text", x = 7.5, y = 8.4e3, size = 7/.pt, label = "TM biofilm") +
  annotate(x = 0.55, xend = 2.45, y = 8.2e3, yend = 8.2e3, geom = "segment") +
  annotate(x = 2.55, xend = 4.45, y = 8.2e3, yend = 8.2e3, geom = "segment") +
  annotate(x = 4.55, xend = 6.45, y = 8.2e3, yend = 8.2e3, geom = "segment") +
  annotate(x = 6.55, xend = 8.45, y = 8.2e3, yend = 8.2e3, geom = "segment") +
  theme_barplot(fontsize = 7) + 
  theme(legend.key.size = unit(3, "mm"),
        axis.text.y = element_text(face = "bold"),
        panel.grid.major.y = element_line(linetype = "dotted", colour = "grey", size = 0.5),
        legend.position = "right",
        panel.border = element_rect(fill = NA),
        plot.margin = margin(0,0,0,4),
        axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.ticks.y = element_line(),
        legend.text = element_text(size = 7), legend.title = element_text(size = 7)
  )
```


```{r export barplot fungi}
ggsave("figures/fig_s4.pdf", height = 8, width = 14, units = "cm")
ggsave("figures/fig_s4.png", height = 8, width = 14, units = "cm")
```


```{r prepare plot eggnogs}
cols <- c("Unidentified" = "#a6cee3",
          "Other" = "#1f78b4",
          "Hyphomicrobiales" = "#B2DF8A", # Alpha
          "Burkholderiales" = "#33A02C", # Beta
          "Nitrosomonadales" = "#FB9A99", # Beta
          "Rhodocyclales" = "#A50026", # Beta
          "Desulfobacterales" = "#E31A1C", # Delta
          "Campylobacterales" = "#FF7F00", # Epsilon
          "Oceanospirillales" = "#00441B", # Gamma
          "Bacillales" = "#CAB2D6", # Firmicutes
          "Methanobacteriales" = "#FFFF99",
          "Trichomonadida" = "#B15928")

seqcpm %>%
  inner_join(ncbi, by = "geneid") %>%
  inner_join(cogs, by = "geneid") %>%
  filter(COG_ref %in% i) %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(10, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r plot eggnogs orders}
seqcpm %>%
  inner_join(smd, by = "sample") %>%
  inner_join(taxref, by = 'geneid') %>%
  inner_join(cogs, by = "geneid") %>% filter(COG_ref %in% i) %>%
  mutate(aquifer = paste(designation, tolower(nature))) %>%
  group_by(topphylum, COG_ref, aquifer) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = fct_relevel(COG_ref, i),
             fill = fct_relevel(topphylum, names(cols)) %>% fct_rev())
         ) +
  labs(x = expression(bold('Counts per million (' %*% '1000)')), y = '', fill = "Order") +
  facet_wrap(~ aquifer, nrow = 1, ncol = 4) +
  geom_col() +
  scale_fill_manual(values = cols) +
  scale_x_continuous(breaks = c(0,5e4,1e5,1.5e5,2e5),
                     labels = c(expression(bold("0")),
                                expression(bold("50")),
                                expression(bold("100")),
                                expression(bold("150")),
                                expression(bold("200"))), 
                    expand = c(0.004,0)) +  
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  theme_tidy(ratio = 3.4) + 
  theme(panel.border = element_blank(),
        legend.title = element_text(size = 6),
        strip.background = element_rect(size = 0,fill = "#f0f0f0"),
        axis.ticks.y = element_blank(),
        legend.position = c(1,0.72),
        strip.text = element_text(face = "bold"),
        axis.text.y = element_text(face = "plain", size = 6, colour = "black"),
        axis.text.x = element_text(face = "bold", size = 6),
        legend.spacing.x = unit(0.5, "mm"),
        legend.box.margin = margin(0,0,0,0),
        legend.box.background = element_rect(size = .7, colour = "black"),
        legend.key.size = unit(3.3, "mm"),
        axis.line.x.bottom = element_line(colour = "black"))
```

```{r export plot eggnogs}
ggsave('figures/fig_s6x.pdf', height = 10, width = 16, units = 'cm')
ggsave('figures/fig_s6x.png', height = 10, width = 16, units = 'cm')
```


```{r Redox}
kos <- list(
  red = c("K00394", # aprA; adenylylsulfate reductase, subunit A [EC:1.8.99.2]
          "K00395", # aprB; adenylylsulfate reductase, subunit B [EC:1.8.99.2]
          "K00957", # cysD; sulfate adenylyltransferase subunit 2 [EC:2.7.7.4]
          "K00958", # sat, met3; sulfate adenylyltransferase [EC:2.7.7.4]
          "K11180", # dsrA; dissimilatory sulfite reductase alpha subunit [EC:1.8.99.5]
          "K11181", # dsrB; dissimilatory sulfite reductase beta subunit [EC:1.8.99.5]
          "K16950", # asrA; anaerobic sulfite reductase subunit A
          "K16951", # asrB; anaerobic sulfite reductase subunit B
          "K08352", # phsA, psrA; thiosulfate reductase / polysulfide reductase chain A [EC:1.8.5.5]
          "K08354",  # phsC; thiosulfate reductase cytochrome b subunit
          "K00380", # sulfite reductase (NADPH) flavoprotein alpha-component [EC:1.8.1.2]
          "K00860"  # adenylylsulfate kinase [EC:2.7.1.25]
         ), 
  ox = c("K17222", # soxA; L-cysteine S-thiosulfotransferase [EC:2.8.5.2]
         "K21307", # soeA; sulfite dehydrogenase (quinone) subunit SoeA [EC:1.8.5.6]
         "K21308", # soeB; sulfite dehydrogenase (quinone) subunit SoeB [EC:1.8.5.6]
         "K21309", # soeC; sulfite dehydrogenase (quinone) subunit SoeC
         "K00386", # sorB; sulfite dehydrogenase (cytochrome) subunit B [EC:1.8.2.1]
         "K05301", # sorA; sulfite dehydrogenase (cytochrome) subunit A [EC:1.8.2.1]
         "K17229", # fccB; sulfide dehydrogenase [flavocytochrome c] flavoprotein chain [EC:1.8.2.3]
         "K17218",  # sqr; sulfide:quinone oxidoreductase [EC:1.8.5.4]
         "K17226" # sulfur-oxidizing protein SoxY
         )
         )


# Reduction
sulfur <- emapper %>%
  select(geneid, pathway = KEGG_ko, module = KEGG_Module) %>%
  mutate(redox = ifelse(grepl(
    pattern = c("K00394|K00395|K00958|K11180|K11181|K16950|K16951|K08352|K08354|K00380|K00860|K00957|K13811|K00955|K00956|K00390|K00381|K00392"), pathway), "S red. / disp.",NA)
    ) %>%
  mutate(redox = ifelse(grepl(
    pattern = c("K17222|K17223|K17224|K17225|K21307|K21308|K21309|K00386|K05301|K17229|K17218|K17226|K00957|K00958|K22622|K17227|K16937"), pathway), "S oxidation",redox)
    ) %>%
  mutate(redox = ifelse(module == "M00595","S oxidation",redox)) %>%
  mutate(redox = ifelse(module %in% c("M00176","M00596"),"S red. / disp.",redox)) %>%
  filter(!is.na(redox)) %>%
  inner_join(seqcpm, by = "geneid") %>%
  group_by(sample, redox) %>% summarise(cpm = sum(cpm), .groups = "drop")



prokka %>%
  filter(EC_number %in% c("2.7.7.4","1.8.99.5") | 
           product %in% c("Adenylylsulfate reductase subunit alpha",
                          "Adenylylsulfate reductase subunit beta",
                          "Anaerobic sulfite reductase subunit A",
                          "Anaerobic sulfite reductase subunit B") | 
           COG %in% c("COG0243","COG4117","COG0243")) %>% 
  select(geneid, product) %>% 
  mutate(redox = "S red. / disp.") -> sulfur

# Oxidation
prokka %>% 
  filter(EC_number %in% c("2.7.7.4","2.8.5.2","1.8.2.3") |
           product %in% c("Sulfide-quinone reductase") |
           gene %in% c("soeA","soeB","soeC","sqr") |
           COG %in% c("COG3302")) %>% 
  select(geneid, product) %>% 
  mutate(redox = "S oxidation") %>%
  rbind(sulfur) %>%
  inner_join(seqcpm, by = "geneid") %>% 
  inner_join(smd, by = "sample") %>%
  group_by(sample, redox) %>%
  summarise(cpm = sum(cpm), .groups = "drop") -> sulfur

sulfur %>%
  inner_join(smd, by = "sample") %>%
  group_by(env, redox) %>%
  summarise(mean = sum(cpm) / 2, sd = sd(cpm), .groups = "drop") %>%
  ggplot(aes(x = env, 
             y = mean, 
             fill = redox, 
             group = redox)) +
  geom_errorbar(aes(ymin = mean-(sd/2), ymax = mean+sd), 
                width = 0.2, colour = "black",
                position = position_dodge(0.5)) +
  geom_col(position = "dodge", width = .5) +
  labs(y = expression(bold('Counts per million (' %*% '1000)')), x = '') +
  scale_fill_manual(values = c("S oxidation" = "#d9d9d9", 
                               "S red. / disp." = "#525252")
                    ) +
  scale_y_continuous(n.breaks = 4,
                     limits = c(0,30000),
                     labels = c("0","10","20","30")) +
  scale_x_discrete(labels = c("MM planktonic","MM biofilm",
                              "TM planktonic","TM biofilm")) +
  theme_clean(ratio = 1, fontsize = 7) +
  theme(axis.ticks.y = element_line(),
        panel.border = element_rect(fill = NA),
        panel.grid.major.y = element_line(linetype = "dotted"),
        legend.text = element_text(size = 7),
        legend.position = "right"
        )
```

```{r export redox}
ggsave('figures/fig_s5.pdf', height = 10, width = 12, units = 'cm')
ggsave('figures/fig_s5.png', height = 10, width = 12, units = 'cm')
```


```{r emapper}
cols <- c("Unidentified" = "#a6cee3",
          "Other" = "#1f78b4",
          "Hyphomicrobiales" = "#fb9a99", # Alpha
          "Burkholderiales" = "#e31a1c", # Beta
          "Nitrosomonadales" = "#fdbf6f", # Beta
          "Rhodocyclales" = "#ffff99", # Beta
          "Desulfobacterales" = "#ff7f00", # Delta
          "Campylobacterales" = "#b2df8a", # Epsilon
          "Acholeplasmatales" = "#00441b", # Firmicutes
          "Bacillales" = "#cab2d6", # Firmicutes
          "Methanobacteriales" = "#bf812d",
          "Trichomonadida" = "#b15928")

i <- c("Energy production and conversion", # C
       "Translation, ribosomal structure and biogenesis", # J
       "Transcription", # K
       "Cell motility", # N
       "Amino acid transport and metabolism", # E
       "Inorganic ion transport and metabolism", # P
       "Carbohydrate transport and metabolism", # G
       "Intracellular trafficking, secretion, and vesicular transport", # U
       "Signal transduction mechanisms", # T
       "Cell wall/membrane/envelope biogenesis" # M
       )

# J: Translation, ribosomal structure and biogenesis
# C: Energy production and conversion
# N: Cell motility
# O: Posttranslational modification, protein turnover, chaperones
# E: Amino acid transport and metabolism
# L: Replication, recombination and repair
# G: Carbohydrate transport and metabolism
# U: Intracellular trafficking, secretion, and vesicular transport
# T: Signal transduction mechanisms
# M: Cell wall/membrane/envelope biogenesis

seqcpm %>%
  # Only use genes that are differentially expressed and have a COG annotation
  filter(geneid %in% diffexp$geneid & geneid %in% cogs$geneid) %>%
  inner_join(ncbi, by = "geneid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(10, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r plot significant DE eggnogs}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>%
  inner_join(cogs, by = "geneid") %>% filter(COG_ref %in% i) %>%
  inner_join(taxref, by = 'geneid') %>%
  mutate(diffexp = if_else(diffexp == 1, "planktonic", "biofilm")) %>%
  mutate(aquifer = paste(aquifer, diffexp)) %>%
  group_by(topphylum, COG_ref, aquifer) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = fct_relevel(COG_ref, i),
             fill = fct_relevel(topphylum, c('Unidentified','Other',
                                             "Hyphomicrobiales",
                                             "Burkholderiales","Nitrosomonadales","Rhodocyclales",
                                             "Desulfobacterales",
                                             "Campylobacterales")) %>% fct_rev()
             )
         ) +
  labs(x = expression(bold('Counts per million (' %*% '1000)')), y = '', fill = "Order") +
  facet_wrap(~ aquifer, nrow = 1, ncol = 4) +
  geom_col() +
  scale_fill_manual(values = cols) +
  scale_x_continuous(labels = c(expression(bold("0")), 
                                "", 
                                expression(bold("100")),
                                "",
                                expression(bold("200")), 
                                ""), 
                     n.breaks = 5, expand = c(0.004,0)) +  
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  theme_tidy(ratio = 3.4) + 
  theme(panel.border = element_blank(),
        strip.background = element_rect(size = 0,fill = "#f0f0f0"),
        axis.ticks.y = element_blank(),
        legend.position = c(1,0.72),
        strip.text = element_text(face = "bold"),
        axis.text.y = element_text(face = "plain", size = 6, colour = "black"),
        axis.text.x = element_text(face = "bold", size = 6),
        legend.spacing.x = unit(0.5, "mm"),
        legend.box.margin = margin(0,0,0,0), legend.title = element_text(size = 6),
        legend.box.background = element_rect(size = .7, colour = "black"),
        legend.key.size = unit(3.3, "mm"),
        axis.line.x.bottom = element_line(colour = "black"))
```




```{r}
coord <- function(x, y) {
  distance <- (x - 89)/0.173
  if (y <= 391) {depth <- 0}
  else {depth <- (y - 391)/0.1646}
  return(paste("Distance is: ",round(distance,3)," m",", depth is: ",round(depth,3)," m", sep = ""))
}
i <- list(borehole = c("SA1229A-1","KA3385A-1R"), depth = c(-171.26, -448.35), distance = c(1100,1600))

data.frame(i) %>% ggplot(aes(x = distance, y = depth)) +
  geom_point(size = 2) +
  annotate("segment", x = 405, xend = 445, y = -115, yend = 0, linetype = "dashed") +
  annotate("segment", x = 428, xend = 486, y = -115, yend = 0, linetype = "dashed") +
  annotate("segment", x = 1156, xend = 1358, y = 0, yend = -711, linetype = "dashed") +
  annotate("segment", x = 1156, xend = 1428, y = 0, yend = -687, linetype = "dashed") +
  annotate("segment", x = 1254, xend = 1358, y = -535, yend = 0, linetype = "dashed") +
  annotate("segment", x = 1428, xend = 1584, y = -352, yend = 0, linetype = "dashed") +
  annotate("segment", x = 688, xend = 723, y = -231, yend = 0, linetype = "dashed") +
  annotate("segment", x = 688, xend = 769, y = -231, yend = 0, linetype = "dashed") +
  annotate("segment", x = 740, xend = 792, y = -225, yend = 0, linetype = "dashed") +
  annotate("segment", x = 855, xend = 908, y = 0, yend = -231, linetype = "dashed") +
  annotate("segment", x = 867, xend = 948, y = 0, yend = -225, linetype = "dashed") +
  # Land / sea segments: #f3e2d0 (land), 8da3ae (water)
  annotate("segment", x = 0, xend = 202, y = 30, yend = 30, linewidth = 10, colour = "#f3e2d0") +
  annotate("segment", x = 202, xend = 335, y = 30, yend = 30, linewidth = 10, colour = "#8da3ae") +
  annotate("segment", x = 335, xend = 711, y = 30, yend = 30, linewidth = 10, colour = "#f3e2d0") +
  annotate("segment", x = 711, xend = 855, y = 30, yend = 30, linewidth = 10, colour = "#8da3ae") +
  annotate("segment", x = 855, xend = 942, y = 30, yend = 30, linewidth = 10, colour = "#f3e2d0") +
  annotate("segment", x = 942, xend = 977, y = 30, yend = 30, linewidth = 10, colour = "#8da3ae") +
  annotate("segment", x = 977, xend = 1029, y = 30, yend = 30, linewidth = 10, colour = "#f3e2d0") +
  annotate("segment", x = 1029, xend = 1092, y = 30, yend = 30, linewidth = 10, colour = "#8da3ae") +
  annotate("segment", x = 1092, xend = 1900, y = 30, yend = 30, linewidth = 10, colour = "#f3e2d0") +
  # Labels for the islands
  annotate("text", x = 523, y = 0, label = "Bockholmen", size = 8/.pt) +
  annotate("text", x = 930, y = 0, label = "Borholmen", size = 8/.pt) +
  annotate("text", x = 1496, y = 0, label = "Äspö", size = 8/.pt) +
  # Lines for tunnel distance
  annotate("segment", x = 52, xend = 52, y = -12, yend = -22) + # 100 m
  annotate("segment", x = 243, xend = 243, y = -43, yend = -53) + # 300 m
  annotate("segment", x = 618, xend = 618, y = -97, yend = -107) + # 700 m
  annotate("segment", x = 815, xend = 815, y = -128, yend = -138) + # 900 m
  annotate("segment", x = 994, xend = 994, y = -152, yend = -162) + # 1100 m
  annotate("segment", x = 1168, xend = 1168, y = -176, yend = -186) + # 1300 m
  annotate("segment", x = 1364, xend = 1364, y = -207, yend = -217) + # 1500 m
  annotate("segment", x = 1653, xend = 1663, y = -261, yend = -261) + # 1900 m horizontal
  annotate("segment", x = 1366, xend = 1376, y = -316, yend = -316) + # 2400 m horizontal
  annotate("segment", x = 1661, xend = 1671, y = -383, yend = -383) + # 2900 m horizontal
  annotate("segment", x = 1572, xend = 1572, y = -413, yend = -423) + # 3100 m
  annotate("segment", x = 1728, xend = 1728, y = -456, yend = -466) + # 3500 m
  # Labels for tunnel distance
  annotate("text", x = 52, y = -30, label = "100", size = 7/.pt) + # 100 m
  annotate("text", x = 243, y = -61, label = "300", size = 7/.pt) + # 300 m
  annotate("text", x = 618, y = -115, label = "700", size = 7/.pt) + # 700 m
  annotate("text", x = 815, y = -146, label = "900", size = 7/.pt) + # 900 m
  annotate("text", x = 994, y = -170, label = "1100", size = 7/.pt) + # 1100 m
  annotate("text", x = 1168, y = -194, label = "1300", size = 7/.pt) + # 1300 m
  annotate("text", x = 1364, y = -225, label = "1500", size = 7/.pt) + # 1500 m
  annotate("text", x = 1671, y = -261, label = "1900", size = 7/.pt) + # 1900 m horizontal
  annotate("text", x = 1358, y = -316, label = "2400", size = 7/.pt) + # 2400 m horizontal
  annotate("text", x = 1679, y = -383, label = "2900", size = 7/.pt) + # 2900 m horizontal
  annotate("text", x = 1572, y = -431, label = "3100", size = 7/.pt) + # 3100 m
  annotate("text", x = 1728, y = -474, label = "3600", size = 7/.pt) + # 3500 m
  # Draw tunnel
  scale_x_continuous(expand = c(0,002),
                     breaks = c(0,500,1000,1500),
                     labels = c("0","500","1,000","1,500")) +
  scale_y_continuous(breaks = seq(0,-700, by = -100), 
                     expand = c(0,002)) +
  labs(x = "Distance (m)", y = "Depth (m)") +
  theme(
    panel.grid = element_blank(),
    panel.background = element_blank(),
    panel.border = element_rect(fill = NA, linewidth = 1),
    axis.title = element_text(size = 8, colour = "black"),
    axis.text = element_text(size = 8, colour = "black")
  )
```
```{r}
ggsave(filename = "äspö_draft.pdf", width = 14, height = 10, units = "cm")
```


```{r plot sweden}
sweden <- sf::st_read("../Cultures/didactic-spork/data/SWE_adm/SWE_adm0.shp")
baltic <- sf::st_read("../../Downloads/World_Seas_IHO_v3/World_Seas_IHO_v3.shp") %>% filter(NAME %in% c("Baltic Sea","Gulf of Bothnia"))


ggplot() + 
  geom_sf(data = baltic, colour = "#637e8b", fill = "#8da3ae") +
  geom_sf(data = sweden, aes(fill = var1), colour = "#e3bc93", fill = "#f3e2d0") +
  ggspatial::annotation_scale(location = "br") +
  theme_void()
```
```{r}
ggsave(filename = "sweden.png", width = 6, height = 10, units = "cm")
```
