---
title: "flowcells_dna"
author: "George"
date: "20/12/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE}
library(tidyverse)
```

## Define theme and function(s)

```{r set theme}
theme_tidy <- function() {
  theme(
    axis.title.x = element_text(colour = "black", size = 8),
    axis.title.y = element_text(colour = "black", size = 8),
    axis.text.x = element_text(colour = "black", size = 8),
    axis.text.y = element_text(colour = "black", size = 8),
    axis.ticks.y = elelement_blank(),
    legend.text = element_text(colour = "black", size = 8),
    strip.background = element_blank(),
    strip.placement = "outside",
    strip.text = element_text(colour = "black", size = 8, hjust = 0.5),
    panel.grid = element_blank(),
    legend.position = "bottom",
    aspect.ratio = 1.0,
    plot.background = element_blank(),
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", size = 0.5, fill = NA),
    legend.title = element_blank(),
    legend.box.background = element_blank(),
    legend.key = element_blank()
  )
}

theme_barplot <- function() {
  theme(
    axis.title.x = element_text(colour = "black", size = 8),
    axis.title.y = element_text(colour = "black", size = 8),
    axis.text.x = element_text(colour = "black", size = 8),
    axis.text.y = element_text(colour = "black", size = 8, margin = margin(0,-5,0,0, unit = 'mm')),
    legend.text = element_text(colour = "black", size = 8),
    strip.background = element_blank(),
    panel.grid = element_blank(),
    legend.position = "bottom",
    aspect.ratio = 0.7,
    strip.text = element_text(colour = "black", size = 8, hjust = 0.5),
    plot.background = element_blank(),
    panel.background = element_blank(),
    legend.title = element_blank(),
    legend.box.background = element_blank(),
    legend.key = element_blank(),
    legend.spacing.x = unit(0.5, 'mm'),
    panel.border = element_blank(),
    legend.key.size = unit(4.5, "mm"),
    legend.box.spacing = unit(1, 'mm'),
    axis.ticks.y = element_blank()
  )
}
```


```{r barplot function}
barplot <- function(asvs = seqtab, meta = smd, taxonomy = tax, taxlevel = "phylum", n = 11) {
  if (taxlevel == "phylum") {
    taxonomy <- taxonomy %>% mutate(phylum = if_else(phylum == "Proteobacteria", class, phylum))
  asvs %>%
  inner_join(meta, by = 'sample') %>%
  inner_join(taxonomy, by = "seqid") %>%
  group_by(phylum, env) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum) & phylum != "uncultured bacterium") %>%
  top_n(11, mean_relab) -> t

taxonomy %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  replace_na(list("topphylum" = "Other")) -> taxref

asvs %>%
  inner_join(taxref, by = "seqid") %>% 
  inner_join(meta, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %>% 
  summarise(relab = sum(relab), .groups = 'drop') -> asvs.phylum
  return(asvs.phylum)
  }
  else if (taxlevel == "class") {
  asvs %>%
  inner_join(meta, by = 'sample') %>%
  inner_join(taxonomy, by = "seqid") %>%
  group_by(class, env) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(class) & class != "uncultured bacterium") %>%
  top_n(11, mean_relab) -> t

taxonomy %>%
  left_join(t %>% transmute(class, topphylum = class), by = "class") %>%
  replace_na(list("topphylum" = "Other")) -> taxref

asvs %>%
  inner_join(taxref, by = "seqid") %>% 
  inner_join(meta, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %>% 
  summarise(relab = sum(relab), .groups = 'drop') -> asvs.class
  return(asvs.class)
  }
  else {
  asvs %>%
  inner_join(meta, by = 'sample') %>%
  inner_join(taxonomy, by = "seqid") %>%
  group_by(order, env) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & !grepl(pattern = "uncultured", x = order)) %>%
  top_n(n, mean_relab) -> t

taxonomy %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) -> taxref

asvs %>%
  inner_join(taxref, by = "seqid") %>% 
  inner_join(meta, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %>% 
  summarise(relab = sum(relab), .groups = 'drop') -> asvs.order
  return(asvs.order)
  }
}
```

## Read files

```{r input files, message=FALSE, warning=FALSE}
read_tsv('smd.tsv', col_types = cols(.default = col_character())) %>%
  mutate(env = if_else(nature == "Planktonic", paste(nature, desig), paste(nature, desig, age))) -> smd

read_tsv("taxonomy.tsv", col_types = cols(.default = col_character())) -> tax

read_tsv('seqtab.tsv', col_types = cols(.default = col_character(), count = col_integer())) %>%
  # Filter out samples with less than 1000 reads
  group_by(sample) %>% filter(sum(count) > 1000) %>%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %>% ungroup() %>%
  # Average the relab within an environment by dividing by the total
  inner_join(smd, by = 'sample') %>%
  group_by(env) %>% mutate(relab = relab / sum(relab)) %>% ungroup() %>%
  select(seqid, sample, count, relab) -> seqtab
```

## Rarefaction

```{r rarefaction}
seqtab %>%
  inner_join(smd, by = 'sample') %>% mutate(nature = paste(nature, desig)) %>%
  group_by(nature, seqid) %>% summarise(count = sum(count), .groups = 'drop') %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('nature') -> seqtab_matrix

min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, sample = ., step = 100, xlab = 'Sequencing depth', ylab = 'Diversity (# ASVs)', label = TRUE)
```

## Alpha diversity using Shannon-Weaver index

```{r calculcate Shannon, message=FALSE, warning=FALSE}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>% gather(seqid, shannon, -1) %>%
  rename(shannon = 1) %>%
  rownames_to_column('sample') %>%
  inner_join(smd, by = "sample") %>%
  group_by(nature, borehole, filter, age) %>% mutate(meandiv = mean(shannon)) %>% ungroup() %>%
  arrange(env) -> adiv
```

Plot alpha diversity

```{r plot adiv}
adiv %>%
  mutate(env = paste(nature, desig)) %>%
  ggplot(aes(x = shannon, y = fct_relevel(env, c('Biofilm TM-448.4','Planktonic TM-448.4')))) + 
  geom_boxplot() +
  geom_jitter(width = 0.01, alpha = 0.4, height = 0.1) +
  labs(x = "Alpha diversity (Shannon index)", y = "") +
  theme_tidy()
```

```{r export adiv plot, message=FALSE, warning=FALSE}
ggsave("figures/adiv_dna.pdf", width = 12, height = 10, units = "cm")
```

```{r statistics alpha diversity}
suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(adiv$meandiv)
# Preferably all the dots on the diagonal
qqPlot(adiv$meandiv)
# Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(meandiv ~ paste(nature, desig), data = adiv)
# Run ANOVA (null hypothesis diversity is equal)
aov(meandiv ~ paste(nature, desig), data = adiv) %>% summary()
# Post-hoc
aov(meandiv ~ paste(nature, desig), data = adiv) %>% TukeyHSD()
```

## Bar plot DNA

```{r bar plot DNA phylum}
barplot() %>%
  # Call the plot
  ggplot(aes(x = fct_rev(env), y = relab, fill = fct_relevel(topphylum, c("Other")))) +
  labs(x = '', y = 'Relative abundance') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_flip() + 
  scale_y_continuous(trans = 'reverse', labels = c('1.0','0.75','0.50','0.25','0')) +
  theme_barplot()
```

```{r export barplot DNA phylum}
ggsave("figures/barplot_phylum.pdf", height = 12, width = 16, units = "cm")
```


```{r bar plot DNA class}
barplot(taxlevel = "class") %>%
  # Call the plot
  ggplot(aes(x = fct_rev(env), y = relab, fill = fct_relevel(topphylum, c("Other")))) +
  labs(x = '', y = 'Relative abundance') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_flip() + 
  scale_y_continuous(trans = 'reverse', labels = c('1.0','0.75','0.50','0.25','0')) +
  theme_barplot()
```
```{r export barplot DNA class}
ggsave("figures/barplot_class.pdf", height = 12, width = 16, units = "cm")
```


```{r bar plot DNA order}
barplot(taxlevel = "order") %>%
  mutate(topphylum = gsub("Candidatus", "Ca.", topphylum)) %>%
  mutate(topphylum = gsub("Marine group B", "", topphylum)) %>%
  mutate(topphylum = gsub("[()]", "", topphylum)) %>%
  # Call the plot
  ggplot(aes(x = fct_rev(env), y = relab, fill = fct_relevel(topphylum, c("Other")))) +
  labs(x = '', y = 'Relative abundance') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_flip() + 
  scale_y_continuous(trans = 'reverse', labels = c('1.0','0.75','0.50','0.25','0')) +
  theme_barplot()
```

```{r export barplot DNA order}
ggsave("figures/barplot_order.pdf", height = 12, width = 16, units = "cm")
```


## Differential abundance and heatmap


```{r ALDEx2 analysis for differential ASV abundance, message=FALSE, warning=FALSE}
seqtab %>%
  # Filter out very low abundant ASVs
  group_by(seqid) %>% filter(count > 10) %>% ungroup() %>%
  select(-relab) %>% spread(sample, count, fill = 0) %>%
  column_to_rownames('seqid') %>%
  ALDEx2::aldex(conditions = smd %>% arrange(sample) %>% pull(nature)) %>%
  rownames_to_column("seqid") %>%
  inner_join(tax) -> aldex
```


```{r pheatmap}
aldex %>%
  slice_max(rab.win.Biofilm, n = 25) %>%
  # Mutate taxonomies that only say 'uncultured'
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured bacterium'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured organism'))) %>%
  # Create a label based on the highest taxonomic level
  mutate(label = coalesce(genus, family, order, class, phylum, domain)) %>%
  # Add a unique ASV labl
  mutate(label = paste("ASV", sprintf('%02d', row_number()), label)) %>%
  select(label, seqid) %>%
  # Join with seqtab to obtain abundance per sample
  inner_join(seqtab, by = 'seqid') %>%
  # Join with metadata to add categories
  inner_join(smd, by = 'sample') %>%
  select(label, env, relab) %>%
  # Sum abundance within each grouping
  group_by(env, label) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  # Transform abundance
  mutate(relab = sqrt(relab)) %>%
  spread(env, relab, fill = 0) %>% column_to_rownames('label') %>%
  # Convert to a matrix
  as.matrix() -> heatmap_matrix

library(RColorBrewer)
pheatmap::pheatmap(heatmap_matrix,
                   filename = 'figures/heatmap_d.pdf',
                   color = colorRampPalette(brewer.pal(8, "Blues"))(7),
                   legend_breaks = c(0, 0.1, 0.2, 0.3, max(heatmap_matrix)),
                   legend_labels = c('0', '0.1', '0.2', '0.3', 'Abundance'),
                   breaks = c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35),
                   border_color = '#d9d9d9', fontsize = 8,
                   cellwidth = 15, cellheight = 15,
                   scale = 'none', 
                   cluster_rows = T, cluster_cols = F, 
                   treeheight_row = 20,
                   clustering_distance_rows = 'euclidean',
                   legend = T)
```

## PCA on Hellinger-transformed dataset DNA

Statistics on beta diversity: PERMANOVA

```{r permanova}
smd %>% mutate(nature = paste(nature, desig)) %>% arrange(sample) -> df

seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  pairwiseAdonis::pairwise.adonis(factors = df$nature, sim.method = 'euclidean')
```

Transform the genecounts

```{r run PCA}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  rownames_to_column('sample') %>%
  gather(seqid, hellinger, -1) %>%
  filter(hellinger > 0) %>%
  mutate(hellinger = as.numeric(hellinger)) %>%
  spread(seqid, hellinger, fill = 0) %>%
  column_to_rownames('sample') %>% vegan::rda() -> pca
```


```{r Extract PCA values}
pca.samples <- pca$CA$u %>% data.frame() %>% tibble::rownames_to_column('sample')
pca.geneid    <- pca$CA$v %>% data.frame() %>% tibble::rownames_to_column('seqid')
pca.eigs    <- pca$CA$eig %>% data.frame() %>% tibble::rownames_to_column('pc') %>%
  rename(eigval = 2) %>%
  mutate(propexpl = eigval/sum(eigval))
```


```{r Plot PCA}
pca.samples %>%
  inner_join(smd, by = 'sample') %>%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = desig), size = 5) +
  # Labelled according to groundwater type
  scale_color_grey() +
  xlab(sprintf("PC1 (%2.1f%% explained)", pca.eigs[1,3] * 100)) +
  ylab(sprintf("PC2 (%2.1f%% explained)", pca.eigs[2,3] * 100)) +
  theme_tidy() +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme(legend.position = c(0.85, 0.25),
        legend.box.background = element_rect(color = 'black', size = 0.25),
        legend.background = element_blank())
```


```{r export PCA}
ggsave("figures/pca_dna.pdf", width = 12, height = 12, units = "cm")
```


## Load gene counts, taxonomy and sample metadata


```{r read data, message=FALSE, warning=FALSE}
read_tsv('final.contigs.tsv') %>%
  select(-Start, -Length, -End, -Strand, -Chr) %>%
  rename(geneid = Geneid) %>%
  gather(sample, count, -1) %>% filter(count > 0) %>%
  mutate(sample = gsub('.pe.bowtie2.sorted.bam', '', sample)) %>%
  mutate(count = as.integer(count)) -> rnatab

# Read taxonomy (Prokka)
read_tsv('prokka.tsv') %>% rename(geneid = locus_tag) -> prokka
read_tsv('eggnogs.tsv', col_types = cols(.default = col_character())) -> eggnogs

# Read sample metadata
read_tsv('smd_metat.tsv') %>% 
  arrange(sample) %>% 
  mutate(treat = factor(treat, levels = c('A','B','C','D'))) %>%
  mutate(ref = paste(designation, nature)) -> smd_metat

# Taxonomy (NCBI)
read_tsv('taxa.tsv', col_types = cols(.default = col_character())) -> ncbi
```


## PCA on Hellinger-transformed dataset RNA

Transform the genecounts

```{r run PCA RNA}
rnatab %>%
  # Vegan transformation
  spread(geneid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  vegan::rda() -> pca
```


```{r Extract PCA RNA values}
pca.samples <- pca$CA$u %>% data.frame() %>% tibble::rownames_to_column('sample')
pca.geneid    <- pca$CA$v %>% data.frame() %>% tibble::rownames_to_column('geneid')
pca.eigs    <- pca$CA$eig %>% data.frame() %>% tibble::rownames_to_column('pc') %>%
  rename(eigval = 2) %>%
  mutate(propexpl = eigval/sum(eigval))
```


```{r Plot PCA RNA}
pca.samples %>%
  inner_join(smd_metat, by = 'sample') %>%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 5) +
  scale_color_grey() +
  xlab(sprintf("PC1 (%2.1f%% explained)", pca.eigs[1,3] * 100)) +
  ylab(sprintf("PC2 (%2.1f%% explained)", pca.eigs[2,3] * 100)) +
  theme_tidy() +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme(legend.position = c(0.85, 0.25), 
        legend.title = element_blank(), 
        aspect.ratio = 1.2,
        legend.box.background = element_rect(colour = "black", size = 0.5))
```


```{r export PCA RNA}
ggsave("figures/fig4a.pdf", width = 10, height = 10, units = "cm")
```


## Differential gene expression using EdgeR


```{r differential expression}
library(edgeR)
rnatab %>% spread(sample, count, fill = 0) %>% column_to_rownames('geneid') -> d
DGEList(d, group = smd_metat$treat) -> d

# Filter based on cpm
keep <- filterByExpr(d)
d <- d[keep, , keep.lib.sizes = F]
# TTM normalization to correct for compositional bias
calcNormFactors(d) -> d

# Design matrix based on experimental design
design.mat <- model.matrix(~ 0 + smd_metat$treat)
rownames(design.mat) <- colnames(d)
colnames(design.mat) <- levels(d$samples$group)

# Estimate gene expression dispersion
d %>% estimateDisp(design = design.mat, robust = F) -> d
# Fit a GLM and run the test
fit <- glmFit(d, design = design.mat)

# Genes which baseline differences between planktonic and biofilm
i <- makeContrasts(mm = A-B, tm = C-D, levels = design.mat)

de_mm <- glmLRT(fit, contrast = i[, "mm"])
de_tm <- glmLRT(fit, contrast = i[, "tm"])
```


```{r extract data DE analysis}
de_mm_test <- decideTestsDGE(de_mm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")
de_tm_test <- decideTestsDGE(de_tm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")

de_mm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "MM-171.3") %>% 
  inner_join(de_mm_test, by = "geneid") -> de_mm
de_tm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "TM-448.4") %>%
  inner_join(de_tm_test, by = "geneid") -> de_tm

de_mm %>% rbind(de_tm) %>% as_tibble() -> diffexp

remove(de_mm, de_tm, de_mm_test, de_tm_test)
```


```{r violin plot DE significant}
diffexp %>%
  # Select significant DE genes
  filter(diffexp != 0) %>%
  # Set minimum data points
  group_by(eggnog, aquifer) %>% add_tally() %>% filter(n > 2) %>% ungroup() %>%
  # Call plot
  ggplot(aes(x = logFC, y = eggnog)) +
  labs(x = 'Log fold-change', y = '') + 
  geom_violin() +
  facet_wrap(~ aquifer, scales = "free_x") +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  theme_tidy() + theme(aspect.ratio = 3.0)
```


```{r export violin plot}
ggsave('figures/fig6.pdf', height = 14, width = 18, units = 'cm')
```

For the bar plot, the 11 most abundant taxonomic groups are selected (either phyla, classes or orders)

```{r calculate cpm from raw counts}
rnatab %>%
  filter(geneid %in% ncbi$geneid) %>%
  spread(sample, count, fill = 0) %>%
  column_to_rownames('geneid') %>%
  edgeR::DGEList(group = smd_metat$treat) %>% edgeR::cpm() %>%
  as.data.frame() %>%
  rownames_to_column('geneid') %>%
  gather(sample, cpm , -1) %>% filter(cpm > 0) %>% 
  as_tibble() -> seqcpm
```


```{r select 11 most abundant DE orders}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>% filter(diffexp != 0) %>%
  inner_join(ncbi, by = 'geneid') %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm)) %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(11, mean_relab) -> t
```


```{r add topphylum to taxonomy, message=FALSE, warning=FALSE}
# Add column to taxonomy whether or not order is part of selection
ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) %>%
  select(geneid, topphylum) -> taxref
```


```{r bar plot significant DE transcripts eggnog}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>% filter(diffexp != 0) %>%
  inner_join(taxref, by = 'geneid') %>%
  mutate(diffexp = if_else(diffexp == 1, "Planktonic", "Biofilm")) %>%
  mutate(aquifer = paste(aquifer, diffexp)) %>%
  group_by(topphylum, eggnog, aquifer) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = eggnog,
             fill = fct_relevel(topphylum, c('Other')))) +
  labs(x = 'Counts per million', y = '') +
  facet_wrap(~ aquifer) +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() +
  theme(panel.border = element_blank(), axis.ticks.y = element_blank())
```

```{r export barplot edger significant DE}
ggsave('figures/fig7.pdf', height = 20, width = 22, units = 'cm')
```

## Barplot mRNA

```{r select 11 most abundant phyla RNA}
ncbi <- ncbi %>% mutate(phylum = if_else(superkingdom == 'Eukaryota', superkingdom, phylum))

seqcpm %>%
  inner_join(ncbi, by = "geneid") %>%
  mutate(phylum = if_else(phylum == "Proteobacteria", class, phylum)) %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm)) %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(11, mean_relab) -> t

ncbi %>%
  mutate(phylum = if_else(phylum == "Proteobacteria", class, phylum)) %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r barplot mRNA phylum level}
smd_metat %>% mutate(ref = paste(nature, designation, sep = '\n')) -> smd_metat

seqcpm %>%
  left_join(taxref, by = "geneid") %>%
  # Combine water type and origin
  inner_join(smd_metat, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(ref, topphylum) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  replace_na(list(topphylum = 'Other')) %>%
  # Call plot
  ggplot(aes(x = fct_relevel(ref, c('Biofilm\nTM-448.4','Planktonic\nTM-448.4')), 
             y = cpm, 
             fill = fct_relevel(topphylum, c('Other')))) +
  labs(x = '', y = 'Counts per million') + 
  scale_y_continuous(labels = c("1.0e+6", "7.5e+5", "5.0e+5", "2.5e+5", "0.0"), trans = 'reverse') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_flip() + theme_barplot()
```


```{r export barplot mRNA phylum}
ggsave("figures/fig4b.pdf", height = 14, width = 14, units = "cm")
```


```{r barplot biofilm order level eggnog}
seqcpm %>%
  filter(sample %in% smd_metat[smd_metat$nature == 'Biofilm',]$sample) %>%
  inner_join(ncbi, by = "geneid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm)) %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(11, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) %>%
  inner_join(eggnogs, by = "geneid") -> taxref

seqcpm %>%
  inner_join(taxref, by = 'geneid') %>% 
  inner_join(smd_metat, by = 'sample') %>% filter(nature == "Biofilm") %>%
  group_by(topphylum, eggnog, ref) %>% summarise(cpm = sum(cpm), .groups = 'drop') %>%
  # Call plot
  ggplot(aes(x = cpm, 
             y = eggnog,
             fill = fct_relevel(topphylum, c('Other')))) +
  labs(x = 'Counts per million', y = '') +
  facet_wrap(~ ref) +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() + 
  theme(panel.border = element_blank(), axis.ticks.y = element_blank())
```


```{r explot barplot eggnog}
ggsave("figures/fig5.pdf", height = 14, width = 24, units = "cm")
```

