<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="George Westmeijer" />


<title>Flowcells Äspö HRL</title>

<script src="flowcells_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="flowcells_files/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="flowcells_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="flowcells_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="flowcells_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="flowcells_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="flowcells_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="flowcells_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="flowcells_files/navigation-1.1/tabsets.js"></script>
<script src="flowcells_files/navigation-1.1/codefolding.js"></script>
<link href="flowcells_files/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="flowcells_files/highlightjs-9.12.0/highlight.js"></script>
<link href="flowcells_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="flowcells_files/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Flowcells Äspö HRL</h1>
<h3 class="subtitle">Biofilm formation in deep biosphere at Äspö HRL</h3>
<h4 class="author">George Westmeijer</h4>

</div>


<pre class="r"><code>library(tidyverse)</code></pre>
<div id="define-theme-and-functions" class="section level2" number="0.1">
<h2><span class="header-section-number">0.1</span> Define theme and function(s)</h2>
<pre class="r"><code>theme_tidy &lt;- function() {
  theme(
    axis.title.x = element_text(colour = &quot;black&quot;, size = 8),
    axis.title.y = element_text(colour = &quot;black&quot;, size = 8),
    axis.text.x = element_text(colour = &quot;black&quot;, size = 8),
    axis.text.y = element_text(colour = &quot;black&quot;, size = 8),
    axis.ticks.y = element_blank(),
    legend.text = element_text(colour = &quot;black&quot;, size = 8),
    strip.background = element_blank(),
    strip.placement = &quot;outside&quot;,
    strip.text = element_text(colour = &quot;black&quot;, size = 8, hjust = 0.5),
    panel.grid = element_blank(),
    legend.position = &quot;bottom&quot;,
    aspect.ratio = 1.0,
    panel.background = element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size = 0.5, fill = NA),
    legend.title = element_blank(),
    legend.box.background = element_blank(),
    legend.key = element_blank()
  )
}

theme_barplot &lt;- function() {
  theme(
    axis.title.x = element_text(colour = &quot;black&quot;, size = 8),
    axis.title.y = element_text(colour = &quot;black&quot;, size = 8),
    axis.text.x = element_text(colour = &quot;black&quot;, size = 8),
    axis.text.y = element_text(colour = &quot;black&quot;, size = 8, margin = margin(0,-3,0,0, unit = &#39;mm&#39;)),
    legend.text = element_text(colour = &quot;black&quot;, size = 8),
    strip.background = element_blank(),
    panel.grid = element_blank(),
    legend.position = &quot;bottom&quot;,
    aspect.ratio = 0.7,
    strip.text = element_text(colour = &quot;black&quot;, size = 8, hjust = 0.5),
    panel.background = element_blank(),
    legend.title = element_blank(),
    legend.box.background = element_blank(),
    legend.key = element_blank(),
    legend.spacing.x = unit(0.5, &#39;mm&#39;),
    panel.border = element_blank(),
    legend.key.size = unit(4.5, &quot;mm&quot;),
    legend.box.spacing = unit(1, &#39;mm&#39;),
    axis.ticks.y = element_blank()
  )
}</code></pre>
<pre class="r"><code>barplot &lt;- function(asvs = seqtab, meta = smd, taxonomy = tax, taxlevel = &quot;phylum&quot;, n = 11) {
  if (taxlevel == &quot;phylum&quot;) {
    taxonomy &lt;- taxonomy %&gt;% mutate(phylum = if_else(phylum == &quot;Proteobacteria&quot;, class, phylum))
  asvs %&gt;%
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  inner_join(taxonomy, by = &quot;seqid&quot;) %&gt;%
  group_by(phylum, env) %&gt;%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = &#39;drop_last&#39;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(phylum) &amp; phylum != &quot;uncultured bacterium&quot;) %&gt;%
  top_n(n, mean_relab) -&gt; t

taxonomy %&gt;%
  left_join(t %&gt;% transmute(phylum, topphylum = phylum), by = &quot;phylum&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref

asvs %&gt;%
  inner_join(taxref, by = &quot;seqid&quot;) %&gt;% 
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %&gt;% 
  summarise(relab = sum(relab), .groups = &#39;drop&#39;) -&gt; asvs.phylum
  return(asvs.phylum)
  }
  else if (taxlevel == &quot;class&quot;) {
  asvs %&gt;%
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  inner_join(taxonomy, by = &quot;seqid&quot;) %&gt;%
  group_by(class, env) %&gt;%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = &#39;drop_last&#39;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(class) &amp; class != &quot;uncultured bacterium&quot;) %&gt;%
  top_n(n, mean_relab) -&gt; t

taxonomy %&gt;%
  left_join(t %&gt;% transmute(class, topphylum = class), by = &quot;class&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref

asvs %&gt;%
  inner_join(taxref, by = &quot;seqid&quot;) %&gt;% 
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %&gt;% 
  summarise(relab = sum(relab), .groups = &#39;drop&#39;) -&gt; asvs.class
  return(asvs.class)
  }
  else if (taxlevel == &quot;family&quot;) {
  asvs %&gt;%
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  inner_join(taxonomy, by = &quot;seqid&quot;) %&gt;%
  group_by(family, env) %&gt;%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = &#39;drop_last&#39;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(family) &amp; family != &quot;uncultured bacterium&quot;) %&gt;%
  top_n(n, mean_relab) -&gt; t

taxonomy %&gt;%
  left_join(t %&gt;% transmute(family, topphylum = family), by = &quot;family&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref

asvs %&gt;%
  inner_join(taxref, by = &quot;seqid&quot;) %&gt;% 
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %&gt;% 
  summarise(relab = sum(relab), .groups = &#39;drop&#39;) -&gt; asvs.family
  return(asvs.family)
  }
  else if (taxlevel == &quot;genus&quot;) {
  asvs %&gt;%
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  inner_join(taxonomy, by = &quot;seqid&quot;) %&gt;%
  group_by(genus, env) %&gt;%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = &#39;drop_last&#39;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(genus) &amp; genus != &quot;uncultured bacterium&quot; &amp; genus != &quot;uncultured&quot;) %&gt;%
  top_n(n, mean_relab) -&gt; t

taxonomy %&gt;%
  left_join(t %&gt;% transmute(genus, topphylum = genus), by = &quot;genus&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref

asvs %&gt;%
  inner_join(taxref, by = &quot;seqid&quot;) %&gt;% 
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %&gt;% 
  summarise(relab = sum(relab), .groups = &#39;drop&#39;) -&gt; asvs.genus
  return(asvs.genus)
  }
  else {
  asvs %&gt;%
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  inner_join(taxonomy, by = &quot;seqid&quot;) %&gt;%
  group_by(order, env) %&gt;%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = &#39;drop_last&#39;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order) &amp; !grepl(pattern = &quot;uncultured&quot;, x = order)) %&gt;%
  top_n(n, mean_relab) -&gt; t

taxonomy %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref

asvs %&gt;%
  inner_join(taxref, by = &quot;seqid&quot;) %&gt;% 
  inner_join(meta, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(topphylum, env) %&gt;% 
  summarise(relab = sum(relab), .groups = &#39;drop&#39;) -&gt; asvs.order
  return(asvs.order)
  }
}</code></pre>
</div>
<div id="read-files" class="section level2" number="0.2">
<h2><span class="header-section-number">0.2</span> Read files</h2>
<pre class="r"><code>read_tsv(&#39;smd.tsv&#39;, col_types = cols(.default = col_character())) %&gt;%
  mutate(env = if_else(nature == &quot;Planktonic&quot;, 
                       paste(nature, desig), 
                       paste(nature, desig, age))) %&gt;%
  group_by(env) %&gt;% add_tally() %&gt;% ungroup() -&gt; smd

read_tsv(&quot;taxonomy.tsv&quot;, col_types = cols(.default = col_character())) -&gt; tax

read_tsv(&#39;seqtab.tsv&#39;, col_types = cols(.default = col_character(), count = col_integer())) %&gt;%
  # Filter out samples with less than 1000 reads
  group_by(sample) %&gt;% filter(sum(count) &gt; 1000) %&gt;%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %&gt;% ungroup() %&gt;%
  # Average the relab within an environment by dividing by the total
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  group_by(env) %&gt;% mutate(relab = relab / sum(relab)) %&gt;% ungroup() %&gt;%
  select(seqid, sample, count, relab) -&gt; seqtab</code></pre>
</div>
<div id="rarefaction" class="section level2" number="0.3">
<h2><span class="header-section-number">0.3</span> Rarefaction</h2>
<pre class="r"><code>seqtab %&gt;%
  inner_join(smd, by = &#39;sample&#39;) %&gt;% mutate(nature = paste(nature, desig)) %&gt;%
  group_by(nature, seqid) %&gt;% summarise(count = sum(count), .groups = &#39;drop&#39;) %&gt;%
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&#39;nature&#39;) -&gt; seqtab_matrix

min(rowSums(seqtab_matrix)) %&gt;%
vegan::rarecurve(seqtab_matrix, sample = ., step = 100, xlab = &#39;Sequencing depth&#39;, ylab = &#39;Diversity (# ASVs)&#39;, label = TRUE)</code></pre>
<p><img src="flowcells_files/figure-html/rarefaction-1.png" width="672" /></p>
</div>
<div id="alpha-diversity-using-shannon-weaver-index" class="section level2" number="0.4">
<h2><span class="header-section-number">0.4</span> Alpha diversity using Shannon-Weaver index</h2>
<pre class="r"><code>seqtab %&gt;%
  select(-relab) %&gt;%
  spread(seqid, count, fill = 0) %&gt;% column_to_rownames(&#39;sample&#39;) %&gt;%
  vegan::diversity() %&gt;% as.data.frame() %&gt;% 
  gather(seqid, shannon, -1) %&gt;%
  rename(shannon = 1) %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  inner_join(smd, by = &quot;sample&quot;) -&gt; adiv</code></pre>
<pre class="r"><code>suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(adiv$shannon)</code></pre>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  adiv$shannon
## W = 0.98661, p-value = 0.9331</code></pre>
<pre class="r"><code># Preferably all the dots on the diagonal
qqPlot(adiv$shannon)</code></pre>
<p><img src="flowcells_files/figure-html/statistics%20alpha%20diversity-1.png" width="672" /></p>
<pre><code>## [1] 29  3</code></pre>
<pre class="r"><code># Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(shannon ~ paste(nature, desig), data = adiv)</code></pre>
<pre><code>## Warning in leveneTest.default(y = y, group = group, ...): group coerced to
## factor.</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["F value"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"1.938663","3":"0.1431852","_rn_":"group"},{"1":"32","2":"NA","3":"NA","_rn_":""}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># Run ANOVA (null hypothesis diversity is equal)
aov(shannon ~ paste(nature, desig), data = adiv) %&gt;% summary()</code></pre>
<pre><code>##                      Df Sum Sq Mean Sq F value Pr(&gt;F)
## paste(nature, desig)  3  3.689  1.2296   1.772  0.172
## Residuals            32 22.200  0.6938</code></pre>
<pre class="r"><code># Post-hoc
aov(shannon ~ paste(nature, desig), data = adiv) %&gt;% TukeyHSD()</code></pre>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = shannon ~ paste(nature, desig), data = adiv)
## 
## $`paste(nature, desig)`
##                                                diff        lwr       upr
## Biofilm TM-448.4-Biofilm MM-171.3       -0.33095113 -1.6338446 0.9719423
## Planktonic MM-171.3-Biofilm MM-171.3     0.50600036 -0.6223385 1.6343392
## Planktonic TM-448.4-Biofilm MM-171.3     0.45637041 -0.6719684 1.5847092
## Planktonic MM-171.3-Biofilm TM-448.4     0.83695149 -0.2913873 1.9652903
## Planktonic TM-448.4-Biofilm TM-448.4     0.78732154 -0.3410173 1.9156604
## Planktonic TM-448.4-Planktonic MM-171.3 -0.04962994 -0.9709147 0.8716548
##                                             p adj
## Biofilm TM-448.4-Biofilm MM-171.3       0.9008344
## Planktonic MM-171.3-Biofilm MM-171.3    0.6220715
## Planktonic TM-448.4-Biofilm MM-171.3    0.6945850
## Planktonic MM-171.3-Biofilm TM-448.4    0.2056575
## Planktonic TM-448.4-Biofilm TM-448.4    0.2521671
## Planktonic TM-448.4-Planktonic MM-171.3 0.9988672</code></pre>
<p>Plot alpha diversity</p>
<pre class="r"><code>return.n &lt;- function(x) {return(data.frame(label = paste(&#39;n =&#39;, length(x)), y = max(x) + 0.65))}


adiv %&gt;%
  mutate(env = paste(nature, desig)) %&gt;%
  mutate(env = factor(env, levels = 
                        c(&quot;Planktonic MM-171.3&quot;, &quot;Biofilm MM-171.3&quot;, 
                          &quot;Planktonic TM-448.4&quot;, &quot;Biofilm TM-448.4&quot;))) %&gt;%
  ggplot(aes(x = shannon, y = fct_rev(env))) + 
  geom_boxplot() +
  geom_jitter(width = 0.01, alpha = 0.7, height = 0.1, color = &quot;black&quot;) +
  stat_summary(fun.data = return.n, geom = &quot;text&quot;,
               size = 2.75,
               fun = median, 
               position = position_dodge(width = 0.75)) +
  labs(x = &quot;Alpha diversity (Shannon index)&quot;, y = &quot;&quot;) +
  theme_tidy() +
  scale_x_continuous(limits = c(1.5,6.5)) +
  theme(panel.border = element_blank(),
        axis.line.x = element_line(),
        axis.line.y.left = element_line()) -&gt; plot.adiv</code></pre>
<pre class="r"><code>ggsave(&quot;figures/fig_1a.pdf&quot;, width = 10, height = 8, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_1a.png&quot;, width = 10, height = 8, units = &quot;cm&quot;)</code></pre>
</div>
<div id="fig.-1b-pca-on-hellinger-transformed-dataset-dna" class="section level2" number="0.5">
<h2><span class="header-section-number">0.5</span> Fig. 1b: PCA on Hellinger-transformed dataset DNA</h2>
<p>Statistics on beta diversity: PERMANOVA</p>
<pre class="r"><code>smd %&gt;% mutate(nature = paste(nature, desig)) %&gt;% arrange(sample) -&gt; df

seqtab %&gt;%
  select(-relab) %&gt;%
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) %&gt;%
  vegan::decostand(method = &#39;hellinger&#39;) %&gt;%
  pairwiseAdonis::pairwise.adonis(factors = df$nature, sim.method = &#39;euclidean&#39;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["pairs"],"name":[1],"type":["fct"],"align":["left"]},{"label":["Df"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["SumsOfSqs"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F.Model"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["R2"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["p.value"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["p.adjusted"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["sig"],"name":[8],"type":["fct"],"align":["left"]}],"data":[{"1":"Biofilm TM-448.4 vs Planktonic TM-448.4","2":"1","3":"2.4777947","4":"4.615922","5":"0.22390081","6":"0.001","7":"0.006","8":"*"},{"1":"Biofilm TM-448.4 vs Biofilm MM-171.3","2":"1","3":"0.5322095","4":"0.937775","5":"0.08573727","6":"0.431","7":"1.000","8":""},{"1":"Biofilm TM-448.4 vs Planktonic MM-171.3","2":"1","3":"2.3896169","4":"4.066235","5":"0.20264064","6":"0.004","7":"0.024","8":"."},{"1":"Planktonic TM-448.4 vs Biofilm MM-171.3","2":"1","3":"2.6879645","4":"4.882617","5":"0.23381252","6":"0.001","7":"0.006","8":"*"},{"1":"Planktonic TM-448.4 vs Planktonic MM-171.3","2":"1","3":"3.7455374","4":"6.573316","5":"0.23005087","6":"0.001","7":"0.006","8":"*"},{"1":"Biofilm MM-171.3 vs Planktonic MM-171.3","2":"1","3":"2.8630115","4":"4.760600","5":"0.22930936","6":"0.001","7":"0.006","8":"*"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Hellinger transform the absolute counts</p>
<pre class="r"><code>seqtab %&gt;%
  select(-relab) %&gt;%
  spread(seqid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) %&gt;%
  vegan::decostand(method = &#39;hellinger&#39;) %&gt;%
  rownames_to_column(&#39;sample&#39;) %&gt;%
  gather(seqid, hellinger, -1) %&gt;%
  filter(hellinger &gt; 0) %&gt;%
  mutate(hellinger = as.numeric(hellinger)) %&gt;%
  spread(seqid, hellinger, fill = 0) %&gt;%
  column_to_rownames(&#39;sample&#39;) %&gt;% 
  vegan::rda() -&gt; pca</code></pre>
<pre class="r"><code>pca.samples &lt;- pca$CA$u %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;sample&#39;)
pca.geneid    &lt;- pca$CA$v %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;seqid&#39;)
pca.eigs    &lt;- pca$CA$eig %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;pc&#39;) %&gt;%
  rename(eigval = 2) %&gt;%
  mutate(propexpl = eigval/sum(eigval))</code></pre>
<pre class="r"><code>pca.samples %&gt;%
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  mutate(age = if_else(nature == &quot;Planktonic&quot;, &quot;&quot;, age)) %&gt;%
  mutate(age = gsub(&#39;d&#39;, &#39; d&#39;, age)) %&gt;%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = desig), size = 3) +
  # Labelled according to groundwater type
  scale_color_grey() + 
  geom_text(aes(label = age), size = 2.5, hjust = -0.5) +
  xlab(sprintf(&quot;PC1 (%2.1f%% explained)&quot;, pca.eigs[1,3] * 100)) +
  ylab(sprintf(&quot;PC2 (%2.1f%% explained)&quot;, pca.eigs[2,3] * 100)) +
  theme_tidy() +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;) +
  geom_hline(yintercept = 0, linetype = &#39;dotted&#39;) +
  theme(legend.text = element_text(size = 8, colour = &quot;black&quot;),
        legend.spacing.x = unit(0.0, &quot;mm&quot;)) -&gt; plot.pca</code></pre>
<pre class="r"><code>cowplot::plot_grid(plot.adiv, 
                   plot.pca,
                   labels = c(&quot;a)&quot;, &quot;b)&quot;), label_size = 10, 
                   label_fontface = &quot;plain&quot;,
                   label_x = c(0.33, 0.11), label_y = 0.96)</code></pre>
<p><img src="flowcells_files/figure-html/cowplot%20fig.%201-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_1.pdf&quot;, width = 18, height = 12, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_1.png&quot;, width = 18, height = 12, units = &quot;cm&quot;)</code></pre>
</div>
<div id="bar-plot-dna" class="section level2" number="0.6">
<h2><span class="header-section-number">0.6</span> Bar plot DNA</h2>
<pre class="r"><code>cols &lt;- c(&#39;Other&#39; = &#39;#A6CEE3&#39;, &#39;Actinobacteria&#39; = &#39;#569EA4&#39;, &#39;Alphaproteobacteria&#39; = &#39;#AADB84&#39;,
&#39;Bacteroidetes&#39; = &#39;#52AF43&#39;, &#39;Chloroflexi&#39; = &#39;#F88A89&#39;,
&#39;Deltaproteobacteria&#39; = &#39;#E73233&#39;, &#39;Epsilonproteobacteria&#39; = &#39;#fdbf6f&#39;,
&#39;Firmicutes&#39; = &#39;#fccde5&#39;, &#39;Gammaproteobacteria&#39; = &#39;#bc80bd&#39;, &quot;Omnitrophicaeota&quot; = &#39;#3D3F99&#39;,
&#39;Patescibacteria&#39; = &#39;#ffff99&#39;, &#39;Spirochaetes&#39; = &#39;#b15928&#39;)

barplot() %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;SAR324 clade(Marine group B)&quot;, &quot;SAR324 clade&quot;, topphylum)) %&gt;%
  mutate(env = gsub(&#39;Planktonic &#39;, &#39;&#39;, env)) %&gt;%
  mutate(env = gsub(&#39;Biofilm &#39;, &#39;&#39;, env)) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance&#39;) +
  geom_col() +
  scale_fill_manual(values = cols) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;1.0&#39;,&#39;0.75&#39;,&#39;0.50&#39;,&#39;0.25&#39;,&#39;0&#39;)) +
  theme_barplot() +
  guides(fill = guide_legend(nrow = 3))</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20DNA%20phylum-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_2_phylum.pdf&quot;, height = 12, width = 14, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code>my.pal &lt;- colorRampPalette(RColorBrewer::brewer.pal(12, &quot;Paired&quot;))
paired.pal &lt;- my.pal(15)

barplot(taxlevel = &quot;order&quot;, n = 14) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;SAR324 clade(Marine group B)&quot;, &quot;SAR324 clade&quot;, topphylum)) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;Candidatus Falkowbacteria&quot;, &quot;Ca. Falkowbacteria&quot;, topphylum)) %&gt;%
  mutate(env = gsub(&#39;Planktonic &#39;, &#39;&#39;, env)) %&gt;%
  mutate(env = gsub(&#39;Biofilm &#39;, &#39;&#39;, env)) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance&#39;) +
  geom_col(width = 0.9) +
  scale_fill_manual(values = paired.pal) +
  annotate(&#39;text&#39;, x = 5, y = 0.58, size = 2.5, label = &quot;25% Thiobacillus&quot;) +
  annotate(&#39;text&#39;, x = 4, y = 0.5, size = 2.5, label = &quot;74% Thiobacillus&quot;) +
  annotate(&#39;text&#39;, x = 3, y = 0.51, size = 2.5, label = &quot;72% Thiobacillus&quot;) +
  annotate(&#39;text&#39;, x = 2, y = 0.56, size = 2.5, label = &quot;43% Thiobacillus&quot;) +
  annotate(&#39;text&#39;, x = 1, y = 0.58, size = 2.5, label = &quot;37% Thiobacillus&quot;) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;1.0&#39;,&#39;0.75&#39;,&#39;0.50&#39;,&#39;0.25&#39;,&#39;0&#39;)) +
  theme_barplot() + 
  guides(fill = guide_legend(nrow = 4)) + 
  theme(legend.text = element_text(size = 7))</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20DNA%20order-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_2.pdf&quot;, height = 12, width = 14, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code>my.pal &lt;- colorRampPalette(RColorBrewer::brewer.pal(12, &quot;Paired&quot;))
paired.pal &lt;- my.pal(15)

barplot(taxlevel = &quot;genus&quot;, n = 14) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;uncultured delta proteobacterium Sva0853&quot;, &quot;Sva0853 clade&quot;, topphylum)) %&gt;%
  mutate(env = gsub(&#39;Planktonic &#39;, &#39;&#39;, env)) %&gt;%
  mutate(env = gsub(&#39;Biofilm &#39;, &#39;&#39;, env)) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c(&#39;TM-448.4 75d&#39;, &#39;TM-448.4 40d&#39;, &#39;TM-448.4 20d&#39;,
                                    &#39;MM-171.3 75d&#39;, &#39;MM-171.3 40d&#39;, &#39;MM-171.3 20d&#39;,
                                    &#39;TM-448.4&#39;, &#39;MM-171.3&#39;)),
             y = relab, 
             fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance&#39;) +
  geom_col() +
  scale_fill_manual(values = paired.pal) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;1.0&#39;,&#39;0.75&#39;,&#39;0.50&#39;,&#39;0.25&#39;,&#39;0&#39;)) +
  theme_barplot() + guides(fill = guide_legend(nrow = 4)) + theme(legend.text = element_text(size = 7))</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20DNA%20genus-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/barplot_genus.pdf&quot;, height = 12, width = 14, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code>barplot(taxlevel = &quot;class&quot;) %&gt;%
  # Call the plot
  ggplot(aes(x = fct_rev(env), y = relab, fill = fct_relevel(topphylum, c(&quot;Other&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Relative abundance&#39;) +
  geom_col() +
  scale_fill_brewer(palette = &#39;Paired&#39;) +
  coord_flip() + 
  scale_y_continuous(trans = &#39;reverse&#39;, labels = c(&#39;1.0&#39;,&#39;0.75&#39;,&#39;0.50&#39;,&#39;0.25&#39;,&#39;0&#39;)) +
  theme_barplot()</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20DNA%20class-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/barplot_class.pdf&quot;, height = 12, width = 14, units = &quot;cm&quot;)</code></pre>
</div>
<div id="differential-abundance-and-heatmap" class="section level2" number="0.7">
<h2><span class="header-section-number">0.7</span> Differential abundance and heatmap</h2>
<pre class="r"><code>seqtab %&gt;%
  # Filter out very low abundant ASVs
  group_by(seqid) %&gt;% filter(count &gt; 10) %&gt;% ungroup() %&gt;%
  select(-relab) %&gt;% spread(sample, count, fill = 0) %&gt;%
  column_to_rownames(&#39;seqid&#39;) %&gt;%
  ALDEx2::aldex(conditions = smd %&gt;% arrange(sample) %&gt;% pull(nature)) %&gt;%
  rownames_to_column(&quot;seqid&quot;) %&gt;%
  inner_join(tax) -&gt; aldex</code></pre>
<pre class="r"><code>aldex %&gt;%
  slice_max(rab.win.Biofilm, n = 25) %&gt;%
  # Mutate taxonomies that only say &#39;uncultured&#39;
  mutate(across(c(class, order, family, genus, species), ~na_if(., &#39;uncultured bacterium&#39;))) %&gt;%
  mutate(across(c(class, order, family, genus, species), ~na_if(., &#39;uncultured&#39;))) %&gt;%
  mutate(across(c(class, order, family, genus, species), ~na_if(., &#39;uncultured organism&#39;))) %&gt;%
  # Create a label based on the highest taxonomic level
  mutate(label = coalesce(genus, family, order, class, phylum, domain)) %&gt;%
  # Add a unique ASV labl
  mutate(label = paste(&quot;ASV&quot;, sprintf(&#39;%02d&#39;, row_number()), label)) %&gt;%
  select(label, seqid) %&gt;%
  # Join with seqtab to obtain abundance per sample
  inner_join(seqtab, by = &#39;seqid&#39;) %&gt;%
  # Join with metadata to add categories
  inner_join(smd, by = &#39;sample&#39;) %&gt;%
  select(label, env, relab) %&gt;%
  # Sum abundance within each grouping
  group_by(env, label) %&gt;% summarise(relab = sum(relab), .groups = &quot;drop&quot;) %&gt;%
  # Transform abundance
  mutate(relab = sqrt(relab)) %&gt;%
  spread(env, relab, fill = 0) %&gt;% column_to_rownames(&#39;label&#39;) %&gt;%
  # Convert to a matrix
  as.matrix() -&gt; heatmap_matrix

library(RColorBrewer)
pheatmap::pheatmap(heatmap_matrix,
                   filename = &#39;figures/fig_3.pdf&#39;,
                   color = colorRampPalette(brewer.pal(8, &quot;Blues&quot;))(7),
                   legend_breaks = c(0, 0.1, 0.2, 0.3, max(heatmap_matrix)),
                   legend_labels = c(&#39;0&#39;, &#39;0.1&#39;, &#39;0.2&#39;, &#39;0.3&#39;, &#39;Abundance&#39;),
                   breaks = c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35),
                   border_color = &#39;#d9d9d9&#39;, fontsize = 8,
                   cellwidth = 15, cellheight = 15,
                   scale = &#39;none&#39;, 
                   cluster_rows = T, cluster_cols = F, 
                   treeheight_row = 20,
                   clustering_distance_rows = &#39;euclidean&#39;,
                   legend = T)</code></pre>
</div>
<div id="load-gene-counts-taxonomy-and-sample-metadata" class="section level2" number="0.8">
<h2><span class="header-section-number">0.8</span> Load gene counts, taxonomy and sample metadata</h2>
<pre class="r"><code>read_tsv(&#39;final.contigs.tsv&#39;) %&gt;%
  select(-Start, -Length, -End, -Strand, -Chr) %&gt;%
  rename(geneid = Geneid) %&gt;%
  gather(sample, count, -1) %&gt;% filter(count &gt; 0) %&gt;%
  mutate(sample = gsub(&#39;.pe.bowtie2.sorted.bam&#39;, &#39;&#39;, sample)) %&gt;%
  mutate(count = as.integer(count)) -&gt; rnatab

# Read taxonomy (Prokka)
read_tsv(&#39;prokka.tsv&#39;) %&gt;% rename(geneid = locus_tag) -&gt; prokka
read_tsv(&#39;eggnogs.tsv&#39;, col_types = cols(.default = col_character())) -&gt; eggnogs

# Read sample metadata
read_tsv(&#39;smd_metat.tsv&#39;) %&gt;% 
  arrange(sample) %&gt;% 
  mutate(treat = factor(treat, levels = c(&#39;A&#39;,&#39;B&#39;,&#39;C&#39;,&#39;D&#39;))) %&gt;%
  mutate(ref = paste(designation, nature)) -&gt; smd_metat

# Taxonomy (NCBI)
read_tsv(&#39;taxa.tsv&#39;, col_types = cols(.default = col_character())) -&gt; ncbi</code></pre>
</div>
<div id="fig.-4a-pca-on-hellinger-transformed-dataset" class="section level2" number="0.9">
<h2><span class="header-section-number">0.9</span> Fig. 4a: PCA on Hellinger-transformed dataset</h2>
<p>Transform the genecounts</p>
<pre class="r"><code>rnatab %&gt;%
  # Vegan transformation
  spread(geneid, count, fill = 0) %&gt;%
  column_to_rownames(&quot;sample&quot;) %&gt;%
  vegan::decostand(method = &#39;hellinger&#39;) %&gt;%
  vegan::rda() -&gt; pca</code></pre>
<pre class="r"><code>pca.samples &lt;- pca$CA$u %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;sample&#39;)
pca.geneid    &lt;- pca$CA$v %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;geneid&#39;)
pca.eigs    &lt;- pca$CA$eig %&gt;% data.frame() %&gt;% tibble::rownames_to_column(&#39;pc&#39;) %&gt;%
  rename(eigval = 2) %&gt;%
  mutate(propexpl = eigval/sum(eigval))</code></pre>
<pre class="r"><code>pca.samples %&gt;%
  inner_join(smd_metat, by = &#39;sample&#39;) %&gt;%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  scale_color_grey() +
  xlab(sprintf(&quot;PC1 (%2.1f%% explained)&quot;, pca.eigs[1,3] * 100)) +
  ylab(sprintf(&quot;PC2 (%2.1f%% explained)&quot;, pca.eigs[2,3] * 100)) +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;) +
  geom_hline(yintercept = 0, linetype = &#39;dotted&#39;) +
  theme_tidy() +
  theme(legend.position = c(0.80, 0.30),
        legend.text = element_text(size = 8, colour = &quot;black&quot;),
        legend.box.background = element_rect(color = &#39;black&#39;, size = 0.25),
        legend.background = element_blank()) -&gt; plot.pca</code></pre>
<pre class="r"><code>ggsave(&quot;figures/fig_4a.pdf&quot;, width = 10, height = 10, units = &quot;cm&quot;)</code></pre>
</div>
<div id="fig.-4b-barplot-mrna" class="section level2" number="0.10">
<h2><span class="header-section-number">0.10</span> Fig. 4b: Barplot mRNA</h2>
<p>Transform absolute counts to counts per million (cpm) which is the relative abundance within each sample.</p>
<pre class="r"><code>rnatab %&gt;%
  filter(geneid %in% ncbi$geneid) %&gt;%
  spread(sample, count, fill = 0) %&gt;%
  column_to_rownames(&#39;geneid&#39;) %&gt;%
  edgeR::DGEList(group = smd_metat$treat) %&gt;% edgeR::cpm() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&#39;geneid&#39;) %&gt;%
  gather(sample, cpm , -1) %&gt;% filter(cpm &gt; 0) %&gt;% 
  as_tibble() -&gt; seqcpm</code></pre>
<pre class="r"><code>seqcpm %&gt;%
  # Only use transcripts annotated on phylum level
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  inner_join(smd_metat, by = &quot;sample&quot;) %&gt;%
  group_by(phylum, ref) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = &quot;drop_last&quot;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(phylum)) %&gt;%
  top_n(6, mean_relab) -&gt; t

ncbi %&gt;%
  left_join(t %&gt;% transmute(phylum, topphylum = phylum), by = &quot;phylum&quot;) %&gt;%
  mutate(topphylum = if_else(is.na(phylum), &quot;Unidentified&quot;, topphylum)) %&gt;%
  mutate(topphylum = if_else(topphylum == &quot;Proteobacteria&quot;, class, topphylum)) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) -&gt; taxref</code></pre>
<pre class="r"><code>smd_metat %&gt;% mutate(ref = paste(nature, designation, sep = &#39;\n&#39;)) -&gt; smd_metat
cols &lt;- c(&#39;Other&#39; = &#39;#A6CEE3&#39;, &#39;Unidentified&#39; = &#39;#4992C2&#39;,
&#39;Alphaproteob.&#39; = &#39;#AADB84&#39;,
&#39;Bacteroidetes&#39; = &#39;#52AF43&#39;, &#39;Betaproteob.&#39; = &#39;#8A9D5B&#39;,
&#39;Deltaproteob.&#39; = &#39;#E73233&#39;, &#39;Epsilonproteob.&#39; = &#39;#fdbf6f&#39;,
&#39;Eukaryota&#39; = &#39;#FE870D&#39;, &#39;Euryarchaeota&#39; = &#39;#E19B78&#39;,
&#39;Firmicutes&#39; = &#39;#fccde5&#39;, &#39;Gammaproteob.&#39; = &#39;#bc80bd&#39;,
&#39;Tenericutes&#39; = &#39;#C7B699&#39;)

seqcpm %&gt;%
  inner_join(taxref, by = &quot;geneid&quot;) %&gt;%
  # Combine water type and origin
  inner_join(smd_metat, by = &#39;sample&#39;) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(ref, topphylum) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  mutate(topphylum = gsub(&quot;proteobacteria&quot;, &quot;proteob.&quot;, topphylum)) %&gt;%
  # Call plot
  ggplot(aes(x = fct_relevel(ref, c(&#39;Biofilm\nTM-448.4&#39;,&#39;Planktonic\nTM-448.4&#39;)), 
             y = cpm, 
             fill = fct_relevel(topphylum, c(&#39;Other&#39;, &#39;Unidentified&#39;)))) +
  labs(x = &#39;&#39;, y = &#39;Counts per million&#39;) + 
  scale_y_continuous(labels = c(&quot;1e+6&quot;, &quot;7.5e+5&quot;, &quot;5.0e+5&quot;, &quot;2.5e+5&quot;, &quot;0.0&quot;), trans = &#39;reverse&#39;) +
  geom_col(width = 0.7) +
  scale_fill_manual(values = cols) +
  coord_flip() + 
  annotate(&#39;text&#39;, x = 1, y = 700000, size = 2.5, label = &quot;65% Thiobacillus&quot;) +
  theme_barplot() + 
  guides(fill = guide_legend(nrow = 4)) + 
  theme(legend.key.size = unit(4, &#39;mm&#39;)) -&gt; plot.phylum</code></pre>
<pre class="r"><code>cowplot::plot_grid(plot.pca, 
                   plot.phylum, 
                   rel_widths = c(1.0, 1.2),
                   labels = c(&quot;a)&quot;, &quot;b)&quot;), label_size = 10, 
                   label_fontface = &quot;plain&quot;,
                   label_x = c(0.13, 0.19), label_y = 0.90)</code></pre>
<p><img src="flowcells_files/figure-html/cowplot%20fig.%204-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_4.pdf&quot;, height = 12, width = 18, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_4.png&quot;, height = 12, width = 18, units = &quot;cm&quot;)</code></pre>
<p>Barplot superkingdom</p>
<pre class="r"><code>seqcpm %&gt;%
  inner_join(taxref, by = &quot;geneid&quot;) %&gt;%
  # Combine water type and origin
  inner_join(smd_metat, by = &#39;sample&#39;) %&gt;%
  #mutate(superkingdom = if_else(!is.na(superkingdom) &amp; superkingdom == &quot;Eukaryota&quot; &amp; genus == &quot;Trichomonas&quot;, genus, superkingdom)) %&gt;%
  mutate(superkingdom = if_else(superkingdom == &quot;Eukaryota&quot;, kingdom, superkingdom)) %&gt;%
  mutate(superkingdom = if_else(species == &quot;Trichomonas vaginalis&quot; &amp; !is.na(species), species, superkingdom)) %&gt;%
  # Summarize in order to have the sum for each category and topphylum
  group_by(ref, superkingdom) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  replace_na(list(superkingdom = &quot;Unidentified&quot;)) %&gt;%
  # Call plot
  ggplot(aes(x = fct_relevel(ref, c(&#39;Biofilm\nTM-448.4&#39;,&#39;Planktonic\nTM-448.4&#39;)), 
             y = cpm, 
             fill = fct_relevel(superkingdom, c(&quot;Unidentified&quot;)))) +
  labs(x = &#39;&#39;, y = &#39;Counts per million&#39;) + 
  scale_y_continuous(labels = c(&quot;1.0e+6&quot;, &quot;7.5e+5&quot;, &quot;5.0e+5&quot;, &quot;2.5e+5&quot;, &quot;0.0&quot;), trans = &#39;reverse&#39;) +
  geom_col(width = 0.8) + 
  scale_fill_brewer(palette = &quot;Paired&quot;) +
  coord_flip() + theme_barplot() + guides(fill = guide_legend(nrow = 1)) + 
  theme(aspect.ratio = 0.6, legend.key.size = unit(4, &#39;mm&#39;))</code></pre>
<p><img src="flowcells_files/figure-html/barplot%20superkingdom-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/barplot_superkingdom.pdf&quot;, height = 10, width = 14, units = &quot;cm&quot;)</code></pre>
</div>
<div id="differential-gene-expression-using-edger" class="section level2" number="0.11">
<h2><span class="header-section-number">0.11</span> Differential gene expression using EdgeR</h2>
<pre class="r"><code>library(edgeR)
rnatab %&gt;% spread(sample, count, fill = 0) %&gt;% column_to_rownames(&#39;geneid&#39;) -&gt; d
DGEList(d, group = smd_metat$treat) -&gt; d

# Filter based on cpm
keep &lt;- filterByExpr(d)
d &lt;- d[keep, , keep.lib.sizes = F]
# TTM normalization to correct for compositional bias
calcNormFactors(d) -&gt; d

# Design matrix based on experimental design
design.mat &lt;- model.matrix(~ 0 + smd_metat$treat)
rownames(design.mat) &lt;- colnames(d)
colnames(design.mat) &lt;- levels(d$samples$group)

# Estimate gene expression dispersion
d %&gt;% estimateDisp(design = design.mat, robust = F) -&gt; d
# Fit a GLM and run the test
fit &lt;- glmFit(d, design = design.mat)

# Genes which baseline differences between planktonic and biofilm
i &lt;- makeContrasts(mm = A-B, tm = C-D, levels = design.mat)

de_mm &lt;- glmLRT(fit, contrast = i[, &quot;mm&quot;])
de_tm &lt;- glmLRT(fit, contrast = i[, &quot;tm&quot;])</code></pre>
<pre class="r"><code>de_mm_test &lt;- decideTestsDGE(de_mm) %&gt;% data.frame() %&gt;% rename(diffexp = 1) %&gt;% rownames_to_column(&quot;geneid&quot;)
de_tm_test &lt;- decideTestsDGE(de_tm) %&gt;% data.frame() %&gt;% rename(diffexp = 1) %&gt;% rownames_to_column(&quot;geneid&quot;)

de_mm$table %&gt;% 
  rownames_to_column(&quot;geneid&quot;) %&gt;% 
  inner_join(eggnogs, by = &quot;geneid&quot;) %&gt;% 
  mutate(aquifer = &quot;MM-171.3&quot;) %&gt;% 
  inner_join(de_mm_test, by = &quot;geneid&quot;) -&gt; de_mm
de_tm$table %&gt;% 
  rownames_to_column(&quot;geneid&quot;) %&gt;% 
  inner_join(eggnogs, by = &quot;geneid&quot;) %&gt;% 
  mutate(aquifer = &quot;TM-448.4&quot;) %&gt;%
  inner_join(de_tm_test, by = &quot;geneid&quot;) -&gt; de_tm

de_mm %&gt;% rbind(de_tm) %&gt;% as_tibble() -&gt; diffexp

remove(de_mm, de_tm, de_mm_test, de_tm_test)</code></pre>
<p>Fig. 6 and 7: please organise the y-axis according to function by grouping these categories, e.g., into “biofilm formation”, “motility”, “energy conservation”, “transporters”, “biosynthesis”…</p>
<pre class="r"><code>diffexp %&gt;%
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  # Select significant DE genes
  filter(diffexp != 0) %&gt;%
  # Set minimum data points
  group_by(eggnog, aquifer) %&gt;% add_tally() %&gt;% filter(n &gt; 2) %&gt;% ungroup() %&gt;%
  # Call plot
  ggplot(aes(x = logFC, y = eggnog)) +
  labs(x = &#39;Log fold-change&#39;, y = &#39;&#39;) + 
  geom_violin() +
  facet_wrap(~ aquifer, scales = &quot;free_x&quot;) +
  geom_vline(xintercept = 0, linetype = &#39;dotted&#39;) +
  theme_tidy() + 
  theme(aspect.ratio = 3.0, 
        axis.text.y = element_text(size = 6),
        panel.border = element_blank(),
        axis.line.y.left = element_line(),
        axis.line.x.bottom = element_line()) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 35))</code></pre>
<p><img src="flowcells_files/figure-html/violin%20plot%20DE%20significant-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&#39;figures/fig_6.pdf&#39;, height = 14, width = 18, units = &#39;cm&#39;)</code></pre>
<p>For the bar plot, the 11 most abundant taxonomic groups are selected (either phyla, classes or orders)</p>
<pre class="r"><code>seqcpm %&gt;%
  inner_join(diffexp, by = &quot;geneid&quot;) %&gt;% filter(diffexp != 0) %&gt;%
  inner_join(ncbi, by = &#39;geneid&#39;) %&gt;%
  group_by(order, sample) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm)) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order)) %&gt;%
  top_n(11, mean_relab) -&gt; t</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;order&#39;. You can override using the
## `.groups` argument.</code></pre>
<pre class="r"><code># Add column to taxonomy whether or not order is part of selection
ncbi %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) %&gt;%
  select(geneid, topphylum) -&gt; taxref</code></pre>
<pre class="r"><code>seqcpm %&gt;%
  inner_join(diffexp, by = &quot;geneid&quot;) %&gt;% filter(diffexp != 0) %&gt;%
  inner_join(taxref, by = &#39;geneid&#39;) %&gt;%
  mutate(diffexp = if_else(diffexp == 1, &quot;Planktonic&quot;, &quot;Biofilm&quot;)) %&gt;%
  mutate(aquifer = paste(aquifer, diffexp)) %&gt;%
  group_by(topphylum, eggnog, aquifer) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  ggplot(aes(x = cpm, 
             y = eggnog,
             fill = fct_relevel(topphylum, c(&#39;Other&#39;)))) +
  labs(x = &#39;Counts per million&#39;, y = &#39;&#39;) +
  facet_wrap(~ aquifer) +
  geom_col() +
  scale_fill_brewer(palette = &#39;Paired&#39;) +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() +
  theme(panel.border = element_blank(), axis.ticks.y = element_blank())</code></pre>
<p><img src="flowcells_files/figure-html/bar%20plot%20significant%20DE%20transcripts%20eggnog-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&#39;figures/fig7.pdf&#39;, height = 20, width = 22, units = &#39;cm&#39;)</code></pre>
<pre class="r"><code>seqcpm %&gt;%
  filter(sample %in% smd_metat[smd_metat$nature == &#39;Biofilm&#39;,]$sample) %&gt;%
  inner_join(ncbi, by = &quot;geneid&quot;) %&gt;%
  group_by(order, sample) %&gt;%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = &quot;drop_last&quot;) %&gt;%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  filter(!is.na(order)) %&gt;%
  top_n(11, mean_relab) -&gt; t

ncbi %&gt;%
  left_join(t %&gt;% transmute(order, topphylum = order), by = &quot;order&quot;) %&gt;%
  replace_na(list(&quot;topphylum&quot; = &quot;Other&quot;)) %&gt;%
  inner_join(eggnogs, by = &quot;geneid&quot;) -&gt; taxref

seqcpm %&gt;%
  inner_join(taxref, by = &#39;geneid&#39;) %&gt;% 
  inner_join(smd_metat, by = &#39;sample&#39;) %&gt;% filter(nature == &quot;Biofilm&quot;) %&gt;%
  group_by(topphylum, eggnog, ref) %&gt;% summarise(cpm = sum(cpm), .groups = &#39;drop&#39;) %&gt;%
  # Call plot
  ggplot(aes(x = cpm, 
             y = eggnog,
             fill = fct_relevel(topphylum, c(&#39;Other&#39;)))) +
  labs(x = &#39;Counts per million&#39;, y = &#39;&#39;) +
  facet_wrap(~ ref, labeller = label_wrap_gen(width = 40)) +
  geom_col() +
  scale_fill_brewer(palette = &#39;Paired&#39;) +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() + 
  scale_y_discrete(labels = function(x) str_wrap(x, width = 35)) +
  theme(panel.border = element_blank(),
        legend.spacing.x = unit(0.5, &quot;mm&quot;),
        legend.key.size = unit(4.0, &quot;mm&quot;),
        axis.text.y = element_text(size = 6.0),
        axis.ticks.y.left = element_line(),
        axis.line.x.bottom = element_line(colour = &quot;black&quot;),
        axis.line.y.left = element_line(colour = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/barplot%20biofilm%20order%20level%20eggnog-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fig_5.pdf&quot;, height = 12, width = 22, units = &quot;cm&quot;)
ggsave(&quot;figures/fig_5.png&quot;, height = 12, width = 22, units = &quot;cm&quot;)</code></pre>
<pre class="r"><code>ncbi %&gt;% 
  filter(kingdom == &quot;Fungi&quot;) %&gt;% 
  inner_join(seqcpm, by = &quot;geneid&quot;) %&gt;%
  group_by(sample, taxon) %&gt;% summarise(cpm = sum(cpm), .groups = &quot;drop&quot;) %&gt;%
  ggplot(aes(x = cpm, 
             y = fct_rev(fct_relevel(sample, c(&quot;P10152_S2&quot;,&quot;P10152_S3&quot;))),
             fill = taxon)) +
  labs(x = &#39;Counts per million&#39;, y = &#39;&#39;) +
  geom_col() +
  scale_fill_brewer(palette = &#39;Paired&#39;) +
  coord_cartesian(clip = &quot;off&quot;) +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() +
  geom_text(aes(label = &quot;TM planktonic&quot;, x = 8.25e3, y = 1), hjust = 0, size = 2.0) +
  geom_text(aes(label = &quot;MM planktonic&quot;, x = 8.25e3, y = 2.5), hjust = 0, size = 2.0) +
  geom_text(aes(label = &quot;TM biofilm&quot;, x = 8.25e3, y = 4.5), hjust = 0, size = 2.0) +
  geom_text(aes(label = &quot;MM biofilm&quot;, x = 8.25e3, y = 6.5), hjust = 0, size = 2.0) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 0.55, yend = 1.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 1.55, yend = 3.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 3.55, yend = 5.45, lwd = 0.5, geom = &quot;segment&quot;) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 5.55, yend = 7.45, lwd = 0.5, geom = &quot;segment&quot;) +
  theme(panel.border = element_blank(),
        legend.spacing.x = unit(0.5, &quot;mm&quot;),
        legend.key.size = unit(4.0, &quot;mm&quot;),
        axis.text.y = element_text(size = 6.0),
        axis.ticks.y.left = element_line(),
        axis.line.x.bottom = element_line(colour = &quot;black&quot;),
        axis.line.y.left = element_line(colour = &quot;black&quot;))</code></pre>
<p><img src="flowcells_files/figure-html/fungi-1.png" width="672" /></p>
<pre class="r"><code>ggsave(&quot;figures/fungi.pdf&quot;, height = 12, width = 14, units = &quot;cm&quot;)
ggsave(&quot;figures/fungi.png&quot;, height = 12, width = 14, units = &quot;cm&quot;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
