---
title: "Flowcells Äspö HRL"
subtitle: "Biofilm formation in deep biosphere at Äspö HRL"
author: "George Westmeijer"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE}
library(tidyverse)
```


## Define theme and function(s)

```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
source("/Users/geweaa/Box Sync/Protocols/barplot.R")
```

## Read files


```{r input files, message=FALSE, warning=FALSE}
read_tsv('smd.tsv', col_types = cols(.default = col_character(),
                                     env = col_factor(levels = 
                                                        c("Planktonic MM-171.3","Biofilm MM-171.3",
                                                          "Planktonic TM-448.4","Biofilm TM-448.4")),
                                     treat = col_factor(levels = c(
                                       "A","B","C","D","0.1","0.2"
                                     ))
                                     )
         ) -> smd

read_tsv("taxonomy.tsv", col_types = cols(.default = col_character())) -> tax

read_tsv('seqtab.tsv', col_types = cols(.default = col_character(), count = col_integer())) %>%
  # Filter out samples with less than 1000 reads
  group_by(sample) %>% filter(sum(count) > 1000) %>%
  # Calculate relative abundance within sample
  mutate(relab = count / sum(count)) %>% ungroup() %>%
  # Average the relative abundance within an environment by dividing by the total
  inner_join(smd, by = 'sample') %>%
  mutate(env = if_else(nature == "Planktonic", as.vector(env), paste(env,age))) %>%
  group_by(env) %>% mutate(relab = relab / sum(relab)) %>% ungroup() %>%
  select(sample, seqid, count, relab) -> seqtab
```


## Rarefaction


```{r rarefaction}
seqtab %>%
  inner_join(smd, by = 'sample') %>% mutate(nature = paste(nature, desig)) %>%
  group_by(nature, seqid) %>% summarise(count = sum(count), .groups = 'drop') %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('nature') -> seqtab_matrix

min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, sample = ., step = 100, xlab = 'Sequencing depth', ylab = 'Diversity (# ASVs)', label = TRUE)
```


## Alpha diversity using Shannon-Weaver index


```{r calculcate Shannon, message=FALSE, warning=FALSE}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>% 
  gather(seqid, shannon, -1) %>%
  rename(shannon = 1) %>%
  rownames_to_column('sample') %>%
  inner_join(smd, by = "sample") %>% as_tibble() -> adiv
```


```{r statistics alpha diversity}
suppressPackageStartupMessages(library(car))
# Normality of the data (null hypothesis data is normally distributed)
shapiro.test(adiv$shannon)
# Preferably all the dots on the diagonal
qqPlot(adiv$shannon)
# Homogeneity of variance using Levene (null hypothesis variance is equal)
leveneTest(shannon ~ paste(nature, desig), data = adiv)
# Run ANOVA (null hypothesis diversity is equal)
aov(shannon ~ paste(nature, desig), data = adiv) %>% summary()
# Post-hoc
aov(shannon ~ paste(nature, desig), data = adiv) %>% TukeyHSD()
```


Plot alpha diversity


```{r plot adiv}
return.n <- function(x) {return(data.frame(label = paste('n =', length(x)), y = max(x) + 0.65))}

adiv %>%
  ggplot(aes(x = shannon, y = fct_rev(env))) + 
  geom_boxplot() +
  geom_jitter(width = 0.01, stroke = 0.8, size = 1, fill = "white", shape = 21, height = 0.1) +
  #stat_summary(fun.data = return.n, geom = "text",
               #size = 2.75,
               #fun = median, 
               #position = position_dodge(width = 0.75)) +
  labs(x = "Alpha diversity (Shannon index)", y = "") +
  scale_x_continuous(limits = c(1,6), expand = c(0.002,0)) +
  theme_clean(ratio = 1.6) +
  theme(plot.tag.position = c(.2,.99),
        axis.text.y.left = element_text(face = "plain"))  -> a
```


## Fig. 1b: PCA on Hellinger-transformed dataset DNA


Hellinger transform the absolute counts


```{r run PCA}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  rownames_to_column('sample') %>%
  gather(seqid, hellinger, -1) %>%
  filter(hellinger > 0) %>%
  mutate(hellinger = as.numeric(hellinger)) %>%
  spread(seqid, hellinger, fill = 0) %>%
  column_to_rownames('sample') %>% 
  vegan::rda() -> pca
```


```{r Extract PCA values}
pca.samples <- pca$CA$u %>% data.frame() %>% tibble::rownames_to_column('sample')
pca.geneid    <- pca$CA$v %>% data.frame() %>% tibble::rownames_to_column('seqid')
pca.eigs    <- pca$CA$eig %>% data.frame() %>% tibble::rownames_to_column('pc') %>%
  rename(eigval = 2) %>%
  mutate(propexpl = eigval/sum(eigval))
```


```{r Plot PCA}
library(ggrepel)

pca.samples %>%
  inner_join(smd, by = 'sample') %>%
  mutate(age = if_else(nature == "Planktonic", "", age)) %>%
  mutate(age = gsub('d', ' d', age)) %>%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  # Labelled according to groundwater type
  scale_color_grey() + 
  geom_text_repel(aes(label = age), color = "black",
                  box.padding = 0.3, 
                  max.overlaps = 12,
                  min.segment.length = 0.5,
                  size = 6/.pt) +
  #geom_text(aes(label = age), size = 6/.pt, hjust = -0.5) +
  xlab(sprintf("PC1 (%2.1f%% explained)", pca.eigs[1,3] * 100)) +
  ylab(sprintf("PC2 (%2.1f%% explained)", pca.eigs[2,3] * 100)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme_tidy() +
  theme(legend.spacing.x = unit(0.0, "mm"),
        plot.tag.position = c(.02,.99)) -> b
```


```{r Fig. 1}
library(patchwork)
a + b + plot_annotation(tag_levels = "a", tag_suffix = ")") &
  theme(plot.tag = element_text(size = 8, color = "black"))
```

```{r export Fig. 1}
ggsave("figures/fig_1.pdf", width = 16, height = 10, units = "cm")
ggsave("figures/fig_1.png", width = 16, height = 10, units = "cm")
```


Statistics on beta diversity: PERMANOVA


```{r permanova}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  pairwiseAdonis::pairwise.adonis(factors = smd$env, sim.method = 'euclidean')
```



## Bar plot DNA


```{r bar plot DNA phylum}
cols <- c('Other' = '#A6CEE3', 'Actinobacteria' = '#569EA4', 'Alphaproteobacteria' = '#AADB84',
'Bacteroidetes' = '#52AF43', 'Chloroflexi' = '#F88A89',
'Deltaproteobacteria' = '#E73233', 'Epsilonproteobacteria' = '#fdbf6f',
'Firmicutes' = '#fccde5', 'Gammaproteobacteria' = '#bc80bd', "Omnitrophicaeota" = '#3D3F99',
'Patescibacteria' = '#ffff99', 'Spirochaetes' = '#b15928')

barplot() %>%
  mutate(topphylum = if_else(topphylum == "SAR324 clade(Marine group B)", "SAR324 clade", topphylum)) %>%
  mutate(env =  gsub(" after", "", env)) %>%
  mutate(env =  gsub(" before", "", env)) %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c('TM-448.4 75d', 'TM-448.4 40d', 'TM-448.4 20d',
                                    'MM-171.3 75d', 'MM-171.3 40d', 'MM-171.3 20d',
                                    'TM-448.4', 'MM-171.3')),
             y = relab, 
             fill = fct_relevel(topphylum, c("Other")))) +
  labs(x = '', y = 'Relative abundance (%)') +
  geom_col(width = 0.9) +
  scale_fill_manual(values = cols) +
  annotate(geom = "segment", x = .55, xend = 6.45, y = -.01, yend = -.01) +
  annotate(geom = "segment", x = 6.55, xend = 8.45, y = -.01, yend = -.01) +
  annotate('text', x = 3.5, y = -0.03, size = 6 / .pt, label = "Biofilm", angle = 90, hjust = 0.5) +
  annotate('text', x = 7.5, y = -0.03, size = 6 / .pt, label = "Planktonic", angle = 90, hjust = 0.5) +
  coord_flip() + 
  scale_y_continuous(trans = 'reverse', labels = c('100','75','50','25','0')) +
  theme_barplot() + 
  guides(fill = guide_legend(nrow = 3))
```


```{r export barplot DNA phylum}
ggsave("figures/fig_sx.pdf", height = 10, width = 14, units = "cm")
ggsave("figures/fig_sx.png", height = 10, width = 14, units = "cm")
```


```{r lollipop}
seqtab %>%
  inner_join(smd, by = "sample") %>%
  mutate(env = ifelse(nature == "Planktonic", designation, paste(designation,age))) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(env, genus) %>% summarise(relab = sum(relab) * 100, .groups = "drop_last") %>%
  filter(!is.na(genus) & genus != "uncultured bacterium" & genus != "uncultured") %>%
  slice_max(order_by = relab, n = 1) %>% ungroup() %>%
  mutate(hjust = if_else(relab >= 70, 1.07, -0.09)) %>%
  mutate(relab = round(relab, digits = 2)) %>%
  mutate(genus = paste(genus," (",relab,"%)", sep = "")) %>%
  ggplot(aes(x = fct_relevel(env, c('TM-448.4 75d', 'TM-448.4 40d', 'TM-448.4 20d',
                                    'MM-171.3 75d', 'MM-171.3 40d', 'MM-171.3 20d',
                                    'TM-448.4', 'MM-171.3')), 
             y = relab)) +
  geom_segment(aes(xend = env, y = 0, yend = relab), color = "black", size = 0.1) +
  geom_point(stroke = 0.8, size = 1, fill = "white", shape = 21) +
  geom_label(aes(label = genus, hjust = hjust), 
            color = "black", fontface = "plain", size = 2, label.size = 0.1, label.padding = unit(0.10, "lines")) +
  coord_flip() +
  scale_y_continuous(expand = c(0.002,0), limits = c(0,100), breaks = c(0,50,100), labels = c("0","50","100")) +
  labs(x = "", y = "Relative abundance (%)") +
  theme_clean(ratio = 1.8) +
  theme(panel.grid.major.x = element_line(colour = "grey", linetype = "dotted", size = 0.4),
        panel.grid.major.y = element_blank(),
        plot.margin = margin(4,8,4,0),
        plot.tag.position = c(.1,1.02)) -> a
```


```{r bar plot DNA order}
#my.pal <- colorRampPalette(RColorBrewer::brewer.pal(12, "Paired"))
#paired.pal <- my.pal(15)

barplot(taxlevel = "order", n = 11) %>%
  mutate(topphylum = if_else(topphylum == "SAR324 clade(Marine group B)", "SAR324 clade", topphylum)) %>%
  mutate(topphylum = if_else(topphylum == "Candidatus Falkowbacteria", "Ca. Falkowbacteria", topphylum)) %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(env, c('TM-448.4 75d', 'TM-448.4 40d', 'TM-448.4 20d',
                                    'MM-171.3 75d', 'MM-171.3 40d', 'MM-171.3 20d',
                                    'TM-448.4', 'MM-171.3')),
             y = relab, 
             fill = fct_relevel(topphylum, c("Other")))) +
  labs(x = '', y = 'Relative abundance (%)') +
  geom_col(width = 0.9) +
  scale_fill_brewer(palette = "Paired") +
  #scale_fill_manual(values = paired.pal) +
  #annotate('text', x = 5, y = 0.58, size = 6 / .pt, label = "25% Thiobacillus") +
  #annotate('text', x = 4, y = 0.5, size = 6 / .pt, label = "74% Thiobacillus") +
  #annotate('text', x = 3, y = 0.51, size = 6 / .pt, label = "72% Thiobacillus") +
  #annotate('text', x = 2, y = 0.56, size = 6 / .pt, label = "43% Thiobacillus") +
  #annotate('text', x = 1, y = 0.58, size = 6 / .pt, label = "37% Thiobacillus") +
  annotate(geom = "segment", x = .55, xend = 6.45, y = -.01, yend = -.01) +
  annotate(geom = "segment", x = 6.55, xend = 8.45, y = -.01, yend = -.01) +
  annotate('text', x = 3.5, y = -0.03, size = 6 / .pt, label = "Biofilm", angle = 90, hjust = 0.5) +
  annotate('text', x = 7.5, y = -0.03, size = 6 / .pt, label = "Planktonic", angle = 90, hjust = 0.5) +
  coord_flip() + 
  scale_y_continuous(trans = 'reverse', labels = c('100','75','50','25','0')) + 
  theme_barplot(ratio = 1) + 
  theme(legend.key.size = unit(3.5, "mm"), 
        axis.text.y = element_blank(),
        plot.margin = margin(4,0,4,0),
        plot.tag.position = c(.1,1.02)) +
  guides(fill = guide_legend(nrow = 4)) -> b
```


```{r Fig. 2}
library(patchwork)
a + b +
  plot_layout(widths = c(1,1.8)) +
  plot_annotation(tag_levels = "a", tag_suffix = ")") &
  theme(plot.tag = element_text(size = 8, color = "black"))
```


```{r export Fig. 2}
ggsave("figures/fig_2.pdf", height = 10, width = 14, units = "cm")
ggsave("figures/fig_2.png", height = 10, width = 14, units = "cm")
```


## Differential abundance and heatmap


```{r ALDEx2 analysis for differential ASV abundance, message=FALSE, warning=FALSE}
seqtab %>%
  # Filter out very low abundant ASVs
  group_by(seqid) %>% filter(count > 10) %>% ungroup() %>%
  select(-relab) %>% spread(sample, count, fill = 0) %>%
  column_to_rownames('seqid') %>%
  ALDEx2::aldex(conditions = smd %>% arrange(sample) %>% pull(nature)) %>%
  rownames_to_column("seqid") %>%
  inner_join(tax) -> aldex
```


```{r pheatmap}
aldex %>%
  slice_max(rab.win.Biofilm, n = 25) %>%
  # Mutate taxonomies that only say 'uncultured'
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured bacterium'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured'))) %>%
  mutate(across(c(class, order, family, genus, species), ~na_if(., 'uncultured organism'))) %>%
  # Create a label based on the highest taxonomic level
  mutate(label = coalesce(genus, family, order, class, phylum, domain)) %>%
  # Add a unique ASV labl
  mutate(label = paste("ASV", sprintf('%02d', row_number()), label)) %>%
  select(label, seqid) %>%
  # Join with seqtab to obtain abundance per sample
  inner_join(seqtab, by = 'seqid') %>%
  # Join with metadata to add categories
  inner_join(smd, by = 'sample') %>%
  select(label, env, relab) %>%
  # Sum abundance within each grouping
  group_by(env, label) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  # Transform abundance
  mutate(relab = sqrt(relab)) %>%
  spread(env, relab, fill = 0) %>% column_to_rownames('label') %>%
  # Convert to a matrix
  as.matrix() -> heatmap_matrix

library(RColorBrewer)
pheatmap::pheatmap(heatmap_matrix,
                   filename = 'figures/fig_3.pdf',
                   color = colorRampPalette(brewer.pal(8, "Blues"))(7),
                   legend_breaks = c(0, 0.1, 0.2, 0.3, max(heatmap_matrix)),
                   legend_labels = c('0', '0.1', '0.2', '0.3', 'Abundance'),
                   breaks = c(0, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35),
                   border_color = '#d9d9d9', fontsize = 8,
                   cellwidth = 15, cellheight = 15,
                   scale = 'none', 
                   cluster_rows = T, cluster_cols = F, 
                   treeheight_row = 20,
                   clustering_distance_rows = 'euclidean',
                   legend = T)
```


## Load gene counts, taxonomy and sample metadata


```{r read data, message=FALSE, warning=FALSE}
read_tsv('final.contigs.tsv') %>%
  select(-Start, -Length, -End, -Strand, -Chr) %>%
  rename(geneid = Geneid) %>%
  gather(sample, count, -1) %>% filter(count > 0) %>%
  mutate(sample = gsub('.pe.bowtie2.sorted.bam', '', sample)) %>%
  mutate(count = as.integer(count)) -> rnatab

# Read taxonomy (Prokka)
read_tsv('prokka.tsv') %>% rename(geneid = locus_tag) -> prokka
read_tsv('eggnogs.tsv', col_types = cols(.default = col_character())) -> eggnogs

# Read sample metadata
smd %>% filter(is.na(age)) %>% select(-age) -> smd

# Taxonomy (NCBI)
read_tsv('taxa.tsv', col_types = cols(.default = col_character())) -> ncbi
```


## Fig. 4a: PCA on Hellinger-transformed dataset


Transform the genecounts


```{r run PCA RNA}
rnatab %>%
  # Vegan transformation
  spread(geneid, count, fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::decostand(method = 'hellinger') %>%
  vegan::rda() -> pca
```


```{r Extract PCA RNA values}
pca.samples <- pca$CA$u %>% data.frame() %>% tibble::rownames_to_column('sample')
pca.geneid    <- pca$CA$v %>% data.frame() %>% tibble::rownames_to_column('geneid')
pca.eigs    <- pca$CA$eig %>% data.frame() %>% tibble::rownames_to_column('pc') %>%
  rename(eigval = 2) %>%
  mutate(propexpl = eigval/sum(eigval))
```


```{r Plot PCA RNA}
pca.samples %>%
  inner_join(smd, by = 'sample') %>%
  ggplot(aes(x = PC1, y = PC2)) +
  # Points for samples, coloured by nature
  geom_point(aes(colour = nature, shape = designation), size = 3) +
  scale_color_grey() +
  xlab(sprintf("PC1 (%2.1f%% explained)", pca.eigs[1,3] * 100)) +
  ylab(sprintf("PC2 (%2.1f%% explained)", pca.eigs[2,3] * 100)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  theme_tidy() +
  theme(legend.spacing.x = unit(-0.5, "mm")) -> a
```

## Fig. 4b: Barplot mRNA

Transform absolute counts to counts per million (cpm) which is the relative abundance within each sample.


```{r calculate cpm from raw counts}
rnatab %>%
  filter(geneid %in% ncbi$geneid) %>%
  spread(sample, count, fill = 0) %>%
  column_to_rownames('geneid') %>%
  edgeR::DGEList(group = smd$treat) %>% edgeR::cpm() %>%
  as.data.frame() %>%
  rownames_to_column('geneid') %>%
  gather(sample, cpm , -1) %>% filter(cpm > 0) %>% 
  as_tibble() -> seqcpm
```


```{r select 11 most abundant phyla RNA}
seqcpm %>%
  # Only use transcripts annotated on phylum level
  inner_join(ncbi, by = "geneid") %>%
  inner_join(smd, by = "sample") %>%
  group_by(phylum, env) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(6, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  mutate(topphylum = if_else(topphylum == "Proteobacteria", class, topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r barplot mRNA phylum level}
cols <- c('Other' = '#A6CEE3', 'Unidentified' = '#4992C2',
'Alphaproteob.' = '#AADB84',
'Bacteroidetes' = '#52AF43', 'Betaproteob.' = '#8A9D5B',
'Deltaproteob.' = '#E73233', 'Epsilonproteob.' = '#fdbf6f',
'Eukaryota' = '#FE870D', 'Euryarchaeota' = '#E19B78',
'Firmicutes' = '#fccde5', 'Gammaproteob.' = '#bc80bd',
'Tenericutes' = '#C7B699')

seqcpm %>%
  inner_join(taxref, by = "geneid") %>%
  # Combine water type and origin
  inner_join(smd, by = 'sample') %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(env, topphylum) %>% summarise(cpm = sum(cpm) / 2, .groups = "drop") %>%
  mutate(topphylum = gsub("proteobacteria", "proteob.", topphylum)) %>%
  # Call plot
  ggplot(aes(x = fct_rev(env), 
             y = cpm, 
             fill = fct_relevel(topphylum, c('Other', 'Unidentified')))) +
  labs(x = '', y = 'Counts per million') + 
  scale_y_continuous(labels = c(expression(10^6), "", expression(5 %*% 10^5), "", "0"), 
                     trans = 'reverse', expand = c(0.002,0)) +
  scale_x_discrete(labels = c("Biofilm\nTM-448.4","Planktonic\nTM-448.4",
                              "Biofilm\nMM-171.3","Planktonic\nMM-171.3")) +
  geom_col() +
  scale_fill_manual(values = cols) +
  coord_flip() + 
  annotate('text', x = 1, y = 350000, size = 2, label = "41% Thiobacillaceae") +
  annotate('text', x = 3, y = 350000, size = 2, label = "34% Desulfobacteraceae") +
  theme_barplot(ratio = 1) + 
  guides(fill = guide_legend(nrow = 4)) +
  theme(legend.key.size = unit(3.5, "mm"),
        axis.line.x = element_line())-> b
```


```{r Fig. 4}
library(patchwork)
a + b + plot_annotation(tag_levels = "a", tag_suffix = ")") &
  theme(plot.tag = element_text(size = 8, color = "black"))
```

```{r export Fig. 4}
ggsave("figures/fig_4.pdf", height = 10, width = 16, units = "cm")
ggsave("figures/fig_4.png", height = 10, width = 16, units = "cm")
```


Barplot superkingdom


```{r barplot superkingdom}
seqcpm %>%
  inner_join(taxref, by = "geneid") %>%
  # Combine water type and origin
  inner_join(smd_metat, by = 'sample') %>%
  #mutate(superkingdom = if_else(!is.na(superkingdom) & superkingdom == "Eukaryota" & genus == "Trichomonas", genus, superkingdom)) %>%
  mutate(superkingdom = if_else(superkingdom == "Eukaryota", kingdom, superkingdom)) %>%
  mutate(superkingdom = if_else(species == "Trichomonas vaginalis" & !is.na(species), species, superkingdom)) %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(ref, superkingdom) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  replace_na(list(superkingdom = "Unidentified")) %>%
  # Call plot
  ggplot(aes(x = fct_relevel(ref, c('Biofilm\nTM-448.4','Planktonic\nTM-448.4')), 
             y = cpm, 
             fill = fct_relevel(superkingdom, c("Unidentified")))) +
  labs(x = '', y = 'Counts per million') + 
  scale_y_continuous(labels = c("1.0e+6", "7.5e+5", "5.0e+5", "2.5e+5", "0.0"), trans = 'reverse') +
  geom_col(width = 0.8) + 
  scale_fill_brewer(palette = "Paired") +
  coord_flip() + theme_barplot() + guides(fill = guide_legend(nrow = 1)) + 
  theme(aspect.ratio = 0.6, legend.key.size = unit(4, 'mm'))
```


```{r export barplot mRNA superkingdom}
ggsave("figures/barplot_superkingdom.pdf", height = 10, width = 14, units = "cm")
```


```{r barplot biofilm order level eggnog}
i <- c("Translation, ribosomal structure and biogenesis","Cell wall/membrane/envelope biogenesis",
       "Energy production and conversion","Signal transduction mechanisms",
       "Cell motility","Amino acid transport and metabolism",
       "Replication, recombination and repair","Carbohydrate transport and metabolism",
       "Lipid transport and metabolism","Nucleotide transport and metabolism",
       "Defense mechanisms","RNA processing and modification")

seqcpm %>%
  filter(sample %in% smd[smd$nature == 'Biofilm',]$sample) %>%
  inner_join(ncbi, by = "geneid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm), .groups = "drop_last") %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(11, mean_relab) -> t

ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) %>%
  inner_join(eggnogs, by = "geneid") -> taxref

seqcpm %>%
  inner_join(taxref, by = 'geneid') %>% 
  inner_join(smd, by = 'sample') %>% filter(nature == "Biofilm") %>%
  filter(eggnog %in% i) %>%
  group_by(topphylum, eggnog, env) %>% summarise(cpm = sum(cpm), .groups = 'drop') %>%
  # Call plot
  ggplot(aes(x = cpm, 
             y = fct_relevel(eggnog, i),
             fill = fct_relevel(topphylum, c('Other')))) +
  labs(x = 'Counts per million', y = '') +
  facet_wrap(~ env, labeller = label_wrap_gen(width = 40)) +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  #scale_x_continuous(labels = function(x) format(x, scientific = TRUE), n.breaks = 6) +
  scale_x_continuous(labels = c(0, expression(0.5 %*% 10^5), expression(10^5),
                                expression(1.5 %*% 10^5),""), n.breaks = 6, expand = c(0.002,0)) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 35)) +
  theme_tidy(ratio = 2.0) + 
  theme(panel.border = element_blank(),
        legend.position = c(.93,.70),
        strip.text = element_text(face = "bold"),
        axis.text.y = element_text(face = "plain", size = 6),
        axis.text.x = element_text(face = "bold"),
        legend.spacing.x = unit(0.5, "mm"),
        legend.key.size = unit(3.5, "mm"),
        axis.ticks.y.left = element_line(),
        axis.line.x.bottom = element_line(colour = "black"),
        axis.line.y.left = element_line(colour = "black"))
```


```{r export barplot eggnog}
ggsave("figures/fig_5.pdf", height = 10, width = 14, units = "cm")
ggsave("figures/fig_5.png", height = 10, width = 14, units = "cm")
```


## Differential gene expression using EdgeR


```{r differential expression}
library(edgeR)
rnatab %>% spread(sample, count, fill = 0) %>% column_to_rownames('geneid') -> d
DGEList(d, group = smd_metat$treat) -> d

# Filter based on cpm
keep <- filterByExpr(d)
d <- d[keep, , keep.lib.sizes = F]
# TTM normalization to correct for compositional bias
calcNormFactors(d) -> d

# Design matrix based on experimental design
design.mat <- model.matrix(~ 0 + smd_metat$treat)
rownames(design.mat) <- colnames(d)
colnames(design.mat) <- levels(d$samples$group)

# Estimate gene expression dispersion
d %>% estimateDisp(design = design.mat, robust = F) -> d
# Fit a GLM and run the test
fit <- glmFit(d, design = design.mat)

# Genes which baseline differences between planktonic and biofilm
i <- makeContrasts(mm = A-B, tm = C-D, levels = design.mat)

de_mm <- glmLRT(fit, contrast = i[, "mm"])
de_tm <- glmLRT(fit, contrast = i[, "tm"])
```


```{r extract data DE analysis}
de_mm_test <- decideTestsDGE(de_mm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")
de_tm_test <- decideTestsDGE(de_tm) %>% data.frame() %>% rename(diffexp = 1) %>% rownames_to_column("geneid")

de_mm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "MM-171.3") %>% 
  inner_join(de_mm_test, by = "geneid") -> de_mm
de_tm$table %>% 
  rownames_to_column("geneid") %>% 
  inner_join(eggnogs, by = "geneid") %>% 
  mutate(aquifer = "TM-448.4") %>%
  inner_join(de_tm_test, by = "geneid") -> de_tm

de_mm %>% rbind(de_tm) %>% as_tibble() -> diffexp

remove(de_mm, de_tm, de_mm_test, de_tm_test)
```


Fig. 6 and 7: please organise the y-axis according to function by grouping these categories, e.g., into "biofilm formation", "motility", "energy conservation", "transporters", "biosynthesis"...


```{r violin plot DE significant}
diffexp %>%
  inner_join(ncbi, by = "geneid") %>%
  # Select significant DE genes
  filter(diffexp != 0) %>%
  # Set minimum data points
  group_by(eggnog, aquifer) %>% add_tally() %>% filter(n > 2) %>% ungroup() %>%
  # Call plot
  ggplot(aes(x = logFC, y = eggnog)) +
  labs(x = 'Log fold-change', y = '') + 
  geom_violin() +
  facet_wrap(~ aquifer, scales = "free_x") +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  theme_tidy() + 
  theme(aspect.ratio = 3.0, 
        axis.text.y = element_text(size = 6),
        panel.border = element_blank(),
        axis.line.y.left = element_line(),
        axis.line.x.bottom = element_line()) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 35))
```


```{r export violin plot}
ggsave('figures/fig_6.pdf', height = 14, width = 18, units = 'cm')
```

For the bar plot, the 11 most abundant taxonomic groups are selected (either phyla, classes or orders)


```{r select 11 most abundant DE orders}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>% filter(diffexp != 0) %>%
  inner_join(ncbi, by = 'geneid') %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a category
  summarise(cpm = sum(cpm)) %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(cpm), .groups = 'drop') %>%
  filter(!is.na(order)) %>%
  top_n(11, mean_relab) -> t
```


```{r add topphylum to taxonomy, message=FALSE, warning=FALSE}
# Add column to taxonomy whether or not order is part of selection
ncbi %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  replace_na(list("topphylum" = "Other")) %>%
  select(geneid, topphylum) -> taxref
```


```{r bar plot significant DE transcripts eggnog}
seqcpm %>%
  inner_join(diffexp, by = "geneid") %>% filter(diffexp != 0) %>%
  inner_join(taxref, by = 'geneid') %>%
  mutate(diffexp = if_else(diffexp == 1, "Planktonic", "Biofilm")) %>%
  mutate(aquifer = paste(aquifer, diffexp)) %>%
  group_by(topphylum, eggnog, aquifer) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = eggnog,
             fill = fct_relevel(topphylum, c('Other')))) +
  labs(x = 'Counts per million', y = '') +
  facet_wrap(~ aquifer) +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() +
  theme(panel.border = element_blank(), axis.ticks.y = element_blank())
```


```{r export barplot edger significant DE}
ggsave('figures/fig7.pdf', height = 20, width = 22, units = 'cm')
```


```{r fungi}
ncbi %>% 
  filter(kingdom == "Fungi") %>% 
  inner_join(seqcpm, by = "geneid") %>%
  group_by(sample, taxon) %>% summarise(cpm = sum(cpm), .groups = "drop") %>%
  ggplot(aes(x = cpm, 
             y = fct_rev(fct_relevel(sample, c("P10152_S2","P10152_S3"))),
             fill = taxon)) +
  labs(x = 'Counts per million', y = '') +
  geom_col() +
  scale_fill_brewer(palette = 'Paired') +
  coord_cartesian(clip = "off") +
  scale_x_continuous(labels = function(x) format(x, scientific = TRUE)) +
  theme_tidy() +
  geom_text(aes(label = "TM planktonic", x = 8.25e3, y = 1), hjust = 0, size = 2.0) +
  geom_text(aes(label = "MM planktonic", x = 8.25e3, y = 2.5), hjust = 0, size = 2.0) +
  geom_text(aes(label = "TM biofilm", x = 8.25e3, y = 4.5), hjust = 0, size = 2.0) +
  geom_text(aes(label = "MM biofilm", x = 8.25e3, y = 6.5), hjust = 0, size = 2.0) +
  annotate(x = 8.2e3, xend = 8.2e3, y = 0.55, yend = 1.45, lwd = 0.5, geom = "segment") +
  annotate(x = 8.2e3, xend = 8.2e3, y = 1.55, yend = 3.45, lwd = 0.5, geom = "segment") +
  annotate(x = 8.2e3, xend = 8.2e3, y = 3.55, yend = 5.45, lwd = 0.5, geom = "segment") +
  annotate(x = 8.2e3, xend = 8.2e3, y = 5.55, yend = 7.45, lwd = 0.5, geom = "segment") +
  theme(panel.border = element_blank(),
        legend.spacing.x = unit(0.5, "mm"),
        legend.key.size = unit(4.0, "mm"),
        axis.text.y = element_text(size = 6.0),
        axis.ticks.y.left = element_line(),
        axis.line.x.bottom = element_line(colour = "black"),
        axis.line.y.left = element_line(colour = "black"))
```


```{r export barplot fungi}
ggsave("figures/fungi.pdf", height = 12, width = 14, units = "cm")
ggsave("figures/fungi.png", height = 12, width = 14, units = "cm")
```


```{r Venn}
library(ggvenn)

seqtab %>% 
  inner_join(smd, by = "sample") %>% 
  group_by(nature,seqid) %>% 
  summarise(count = sum(count)) %>% 
  mutate(relab = count / sum(count)) %>% ungroup() -> i

i %>% 
  filter(seqid %in% biofilm & seqid %in% planktonic) %>% 
  filter(nature == "Planktonic") %>% pull(relab) %>% sum()

seqtab %>% inner_join(smd, by = "sample") %>% 
  filter(nature == "Planktonic") %>%
  pull(seqid) %>% unique() -> planktonic

seqtab %>% inner_join(smd, by = "sample") %>% 
  filter(nature == "Biofilm") %>%
  pull(seqid) %>% unique() -> biofilm

seqtab %>%
  select(seqid) %>%
  distinct() %>%
  mutate(Biofilm = if_else(seqid %in% biofilm, TRUE,FALSE)) %>%
  mutate(Planktonic = if_else(seqid %in% planktonic, TRUE,FALSE)) %>%
  ggplot(aes(A = Biofilm, B = Planktonic)) +
  geom_venn(
    show_percentage = FALSE,
    fill_color = c("#2171b5","#c6dbef"),
    fill_alpha = .7,
    set_name_size = 6/.pt, text_size = 6/.pt,
    stroke_size = .5) + 
  annotate(geom = "text", x = -0.5, y = 0, label = "(84%)", size = 6/.pt) +
  annotate(geom = "text", x = 1, y = 0, label = "(80%)", size = 6/.pt) +
  theme_void() + 
  theme(aspect.ratio = 1,
                       plot.background = element_rect(fill = "white", size = 0),
                       plot.margin = margin(0,0,0,0))
```

```{r export Venn}
ggsave("figures/venn.png", height = 6, width = 8, units = "cm")
```

